diff --git a/src/xplist.c b/src/xplist.c
--- a/src/xplist.c
+++ b/src/xplist.c
@@ -406,13 +406,13 @@ static void parse_date(const char *strval, struct TM *btime)
 static int num_digits_i(int64_t i)
 {
     int n;
     int64_t po10;
     n=1;
     if (i < 0) {
-        i = -i;
+        i = (i == INT64_MIN) ? INT64_MAX : -i;
         n++;
     }
     po10=10;
     while (i>=po10) {
         n++;
         if (po10 > PO10i_LIMIT) break;
