diff --git a/tests/spng_read_fuzzer.cc b/tests/spng_read_fuzzer.cc
index d6f40cc..6b1a319 100644
--- a/tests/spng_read_fuzzer.cc
+++ b/tests/spng_read_fuzzer.cc
@@ -1,8 +1,6 @@
+#define SPNG_UNTESTED
 #include "../src/spng.h"
 
-int spng_set_chunk_limits(spng_ctx *ctx, size_t chunk_size, size_t cache_size);
-int spng_get_chunk_limits(spng_ctx *ctx, size_t *chunk_size, size_t *cache_size);
-
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size)
 {
     unsigned char *out = NULL;
