diff --git a/prog/fuzzing/flipdetect_fuzzer.cc b/prog/fuzzing/flipdetect_fuzzer.cc
index 3eb4a85..bad7a6e 100644
--- a/prog/fuzzing/flipdetect_fuzzer.cc
+++ b/prog/fuzzing/flipdetect_fuzzer.cc
@@ -9,7 +9,7 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	PIX *pixs_payload = pixReadMemSpix(data, size);
 	if(pixs_payload == NULL) return 0;
 	
-	l_float32 minupconf, minratio, conf1, conf2, upconf1, leftconf1, &upconf2, &leftconf2;
+	l_float32 minupconf, minratio, conf1 upconf1, leftconf1;
 	PIX *pix_pointer_payload, *return_pix;
 	
 	pix_pointer_payload = pixCopy(NULL, pixs_payload);
@@ -17,11 +17,7 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	pixDestroy(&pix_pointer_payload);
 
 	pix_pointer_payload = pixCopy(NULL, pixs_payload);
-	pixMirrorDetectDwa(pix_pointer_payload, &conf2, 0, 0);
-	pixDestroy(&pix_pointer_payload);
-
-	pix_pointer_payload = pixCopy(NULL, pixs_payload);
-	return_pix = pixOrientCorrect(pixs, minupconf, minratio, NULL, NULL, NULL, 1);
+	return_pix = pixOrientCorrect(pix_pointer_payload, minupconf, minratio, NULL, NULL, NULL, 1);
 	pixDestroy(&pix_pointer_payload);	
 	pixDestroy(&return_pix);
 
@@ -29,10 +25,6 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	pixOrientDetect(pix_pointer_payload, &upconf1, &leftconf1, 0, 0);
 	pixDestroy(&pix_pointer_payload);
 
-	pix_pointer_payload = pixCopy(NULL, pixs_payload);
-	pixOrientDetectDwa(pix1, &upconf2, &leftconf2, 0, 0);
-	pixDestroy(&pix_pointer_payload);
-
 	pixDestroy(&pixs_payload);
 	return 0;
 }
diff --git a/prog/fuzzing/morph_fuzzer.cc b/prog/fuzzing/morph_fuzzer.cc
index e82d0f3..f405555 100644
--- a/prog/fuzzing/morph_fuzzer.cc
+++ b/prog/fuzzing/morph_fuzzer.cc
@@ -10,41 +10,35 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
     if(pixs_payload == NULL) return 0;
 
     PIX *pix, *pix1, *pix_copy1, *pix_copy2, *pix_copy3, *pix_copy4;
+    BOX *box1;
     SEL *sel;
 
-    pix = pixRead("../test8.jpg");
-    pix1 = pixCreate(size, size, 1);
+    pix = pixRead("../feyn-fract.tif");
+    box1 = boxCreate(507, 65, 60, 36);
+    pix1 = pixClipRectangle(pix, box1, NULL);
     sel = selCreateFromPix(pix1, 6, 6, "plus_sign");
     pix_copy1 = pixCopy(NULL, pixs_payload);
     pixCloseGeneralized(pix_copy1, pix, sel);
+    boxDestroy(&box1);
     pixDestroy(&pix_copy1);
-    pixDestroy(&pix);
     pixDestroy(&pix1);
-    selDestroy(&sel);
 
-    pix1 = pixCreate(size, size, 1);
-    sel = selCreateFromPix(pix1, 6, 6, "plus_sign");
     pix_copy2 = pixCopy(NULL, pixs_payload);
-    pixCloseSafe(pix_copy2, pix1, sel);
+    pixCloseSafe(pix_copy2, pix, sel);
     pixDestroy(&pix_copy2);
-    pixDestroy(&pix1);
-    selDestroy(&sel);
 
-    pix = pixRead("../test8.jpg");
-    sel = selCreateFromPix(pix, 6, 6, "plus_sign");
     pix_copy3 = pixCopy(NULL, pixs_payload);
     pixOpenGeneralized(pix_copy3, pix, sel);
     pixDestroy(&pix_copy3);
     pixDestroy(&pix);
     selDestroy(&sel);
 
-    for(l_int32 i=0; i<5; i++){
+    for (l_int32 i = 0; i < 5; i++) {
         if ((sel = selCreate (i, i, "sel_5dp")) == NULL)
             continue;
         char *selname = selGetName(sel);
         pix_copy4 = pixCopy(NULL, pixs_payload);
-        pixFMorphopGen_1(pix_copy4, pix_copy4,
-                     i, selname);
+        pixMorphDwa_1(pix_copy4, pix_copy4, i, selname);
         pixDestroy(&pix_copy4);
         selDestroy(&sel);
     }
diff --git a/prog/fuzzing/pix3_fuzzer.cc b/prog/fuzzing/pix3_fuzzer.cc
index b130311..fd4c6cb 100644
--- a/prog/fuzzing/pix3_fuzzer.cc
+++ b/prog/fuzzing/pix3_fuzzer.cc
@@ -33,7 +33,7 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 
 	box1 = boxCreate(150, 130, 1500, 355);
 	pix_pointer_payload = pixCopy(NULL, pixs_payload);
-	pixAbsDiffInRect(pix_pointer_payload, box1, 0.5, &l_f);
+	pixAbsDiffInRect(pix_pointer_payload, box1, L_HORIZONTAL_LINE, &l_f);
 	pixDestroy(&pix_pointer_payload);
 	boxDestroy(&box1);
 
@@ -69,8 +69,8 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	boxaDestroy(&boxa1);
 	pixDestroy(&return_pix);
 	
-	for(int i=0;i<size;i++){
-	box1 = boxCreate(150, 130, 1500, 355);
+	for (int i = 0; i < 5; i++) {
+            box1 = boxCreate(150, 130, 1500, 355);
 	    pix_pointer_payload = pixCopy(NULL, pixs_payload);
 	    pixCountArbInRect(pix_pointer_payload, box1, L_SET_WHITE, 2, &l_i);
 	    pixDestroy(&pix_pointer_payload);
@@ -96,7 +96,7 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	pixDestroy(&return_pix);
 
 	pix_pointer_payload = pixCopy(NULL, pixs_payload);
-	for (int i = 0; i < size; i++) {
+	for (int i = 0; i < 5; i++) {
             return_pix = pixMakeMaskFromVal(pix_pointer_payload, i);
 	    pixDestroy(&pix_pointer_payload);
 	    pixDestroy(&return_pix);
