diff --git a/c/dec/decode.c b/c/dec/decode.c
--- a/c/dec/decode.c
+++ b/c/dec/decode.c
@@ -2030,14 +2030,16 @@ static BROTLI_NOINLINE BrotliDecoderErrorCode ProcessCommands(
 static BROTLI_NOINLINE BrotliDecoderErrorCode SafeProcessCommands(
     BrotliDecoderState* s) {
   return ProcessCommandsInternal(1, s);
 }
 
 BrotliDecoderResult BrotliDecoderDecompress(
-    size_t encoded_size, const uint8_t* encoded_buffer, size_t* decoded_size,
-    uint8_t* decoded_buffer) {
+    size_t encoded_size,
+    const uint8_t encoded_buffer[BROTLI_ARRAY_PARAM(encoded_size)],
+    size_t* decoded_size,
+    uint8_t decoded_buffer[BROTLI_ARRAY_PARAM(*decoded_size)]) {
   BrotliDecoderState s;
   BrotliDecoderResult result;
   size_t total_out = 0;
   size_t available_in = encoded_size;
   const uint8_t* next_in = encoded_buffer;
   size_t available_out = *decoded_size;
diff --git a/c/enc/encode.c b/c/enc/encode.c
--- a/c/enc/encode.c
+++ b/c/enc/encode.c
@@ -1467,14 +1467,15 @@ static size_t MakeUncompressedStream(
   output[result++] = 3;
   return result;
 }
 
 BROTLI_BOOL BrotliEncoderCompress(
     int quality, int lgwin, BrotliEncoderMode mode, size_t input_size,
-    const uint8_t* input_buffer, size_t* encoded_size,
-    uint8_t* encoded_buffer) {
+    const uint8_t input_buffer[BROTLI_ARRAY_PARAM(input_size)],
+    size_t* encoded_size,
+    uint8_t encoded_buffer[BROTLI_ARRAY_PARAM(*encoded_size)]) {
   BrotliEncoderState* s;
   size_t out_size = *encoded_size;
   const uint8_t* input_start = input_buffer;
   uint8_t* output_start = encoded_buffer;
   size_t max_out_size = BrotliEncoderMaxCompressedSize(input_size);
   if (out_size == 0) {
