diff --git a/internal/cgen/base/image-impl.c b/internal/cgen/base/image-impl.c
--- a/internal/cgen/base/image-impl.c
+++ b/internal/cgen/base/image-impl.c
@@ -261,24 +261,24 @@ wuffs_base__pixel_swizzler__xx__index__src(wuffs_base__slice_u8 dst,
         d + (2 * 2),
         wuffs_base__load_u16le(dst_palette.ptr + ((size_t)s[2] * 4)));
     wuffs_base__store_u16le(
         d + (3 * 2),
         wuffs_base__load_u16le(dst_palette.ptr + ((size_t)s[3] * 4)));
 
-    s += 1 * loop_unroll_count;
-    d += 2 * loop_unroll_count;
+    s += loop_unroll_count * 1;
+    d += loop_unroll_count * 2;
     n -= loop_unroll_count;
   }
 
   while (n >= 1) {
     wuffs_base__store_u16le(
         d + (0 * 2),
         wuffs_base__load_u16le(dst_palette.ptr + ((size_t)s[0] * 4)));
 
     s += 1 * 1;
-    d += 2 * 1;
+    d += 1 * 2;
     n -= 1;
   }
 
   return len;
 }
 
@@ -314,25 +314,25 @@ wuffs_base__pixel_swizzler__xxx__index__src(wuffs_base__slice_u8 dst,
         d + (2 * 3),
         wuffs_base__load_u32le(dst_palette.ptr + ((size_t)s[2] * 4)));
     wuffs_base__store_u32le(
         d + (3 * 3),
         wuffs_base__load_u32le(dst_palette.ptr + ((size_t)s[3] * 4)));
 
-    s += 1 * loop_unroll_count;
-    d += 3 * loop_unroll_count;
+    s += loop_unroll_count * 1;
+    d += loop_unroll_count * 3;
     n -= loop_unroll_count;
   }
 
   while (n >= 1) {
     uint32_t s0 = wuffs_base__load_u32le(dst_palette.ptr + ((size_t)s[0] * 4));
     d[0] = (uint8_t)(s0 >> 0);
     d[1] = (uint8_t)(s0 >> 8);
     d[2] = (uint8_t)(s0 >> 16);
 
     s += 1 * 1;
-    d += 3 * 1;
+    d += 1 * 3;
     n -= 1;
   }
 
   return len;
 }
 
@@ -367,25 +367,25 @@ wuffs_base__pixel_swizzler__xxx__index_binary_alpha__src_over(
     }
     uint32_t s3 = wuffs_base__load_u32le(dst_palette.ptr + ((size_t)s[3] * 4));
     if (s3) {
       wuffs_base__store_u24le(d + (3 * 4), s3);
     }
 
-    s += 1 * loop_unroll_count;
-    d += 3 * loop_unroll_count;
+    s += loop_unroll_count * 1;
+    d += loop_unroll_count * 3;
     n -= loop_unroll_count;
   }
 
   while (n >= 1) {
     uint32_t s0 = wuffs_base__load_u32le(dst_palette.ptr + ((size_t)s[0] * 4));
     if (s0) {
       wuffs_base__store_u24le(d + (0 * 4), s0);
     }
 
     s += 1 * 1;
-    d += 3 * 1;
+    d += 1 * 3;
     n -= 1;
   }
 
   return len;
 }
 
@@ -415,24 +415,24 @@ wuffs_base__pixel_swizzler__xxxx__index__src(wuffs_base__slice_u8 dst,
         d + (2 * 4),
         wuffs_base__load_u32le(dst_palette.ptr + ((size_t)s[2] * 4)));
     wuffs_base__store_u32le(
         d + (3 * 4),
         wuffs_base__load_u32le(dst_palette.ptr + ((size_t)s[3] * 4)));
 
-    s += 1 * loop_unroll_count;
-    d += 4 * loop_unroll_count;
+    s += loop_unroll_count * 1;
+    d += loop_unroll_count * 4;
     n -= loop_unroll_count;
   }
 
   while (n >= 1) {
     wuffs_base__store_u32le(
         d + (0 * 4),
         wuffs_base__load_u32le(dst_palette.ptr + ((size_t)s[0] * 4)));
 
     s += 1 * 1;
-    d += 4 * 1;
+    d += 1 * 4;
     n -= 1;
   }
 
   return len;
 }
 
@@ -467,25 +467,25 @@ wuffs_base__pixel_swizzler__xxxx__index_binary_alpha__src_over(
     }
     uint32_t s3 = wuffs_base__load_u32le(dst_palette.ptr + ((size_t)s[3] * 4));
     if (s3) {
       wuffs_base__store_u32le(d + (3 * 4), s3);
     }
 
-    s += 1 * loop_unroll_count;
-    d += 4 * loop_unroll_count;
+    s += loop_unroll_count * 1;
+    d += loop_unroll_count * 4;
     n -= loop_unroll_count;
   }
 
   while (n >= 1) {
     uint32_t s0 = wuffs_base__load_u32le(dst_palette.ptr + ((size_t)s[0] * 4));
     if (s0) {
       wuffs_base__store_u32le(d + (0 * 4), s0);
     }
 
     s += 1 * 1;
-    d += 4 * 1;
+    d += 1 * 4;
     n -= 1;
   }
 
   return len;
 }
 
@@ -503,14 +503,14 @@ wuffs_base__pixel_swizzler__xxxx__xxx(wuffs_base__slice_u8 dst,
   // TODO: unroll.
 
   while (n >= 1) {
     wuffs_base__store_u32le(d + (0 * 4),
                             0xFF000000 | wuffs_base__load_u24le(s + (0 * 3)));
 
-    s += 3 * 1;
-    d += 4 * 1;
+    s += 1 * 3;
+    d += 1 * 4;
     n -= 1;
   }
 
   return len;
 }
 
@@ -528,13 +528,13 @@ wuffs_base__pixel_swizzler__xxxx__y(wuffs_base__slice_u8 dst,
 
   while (n >= 1) {
     wuffs_base__store_u32le(d + (0 * 4),
                             0xFF000000 | (0x010101 * (uint32_t)s[0]));
 
     s += 1 * 1;
-    d += 4 * 1;
+    d += 1 * 4;
     n -= 1;
   }
 
   return len;
 }
