diff --git a/src/pix3.c b/src/pix3.c
--- a/src/pix3.c
+++ b/src/pix3.c
@@ -3355,12 +3355,18 @@ l_uint32  *data, *line;
     *pcount = 0;
     if (!pixs)
         return ERROR_INT("pixs not defined", procName, 1);
     d = pixGetDepth(pixs);
     if (d != 1 && d != 2 && d != 4 && d != 8)
         return ERROR_INT("pixs not 1, 2, 4 or 8 bpp", procName, 1);
+    if (val < 0)
+        return ERROR_INT("val < 0", procName, 1);
+    if (d <= 8 && val > (1 << d) - 1) {
+        L_ERROR("invalid val = %d for depth %d\n", procName, val, d);
+        return 1;
+    }
     if (factor < 1)
         return ERROR_INT("sampling factor < 1", procName, 1);
 
     pixGetDimensions(pixs, &w, &h, NULL);
     data = pixGetData(pixs);
     wpl = pixGetWpl(pixs);
diff --git a/src/tiffio.c b/src/tiffio.c
--- a/src/tiffio.c
+++ b/src/tiffio.c
@@ -1913,12 +1913,13 @@ TIFF    *tif;
  * \param[out]   pformat   [optional] tiff format; input NULL to ignore
  * \return  0 if OK, 1 on error
  *
  * <pre>
  * Notes:
  *      (1) Use TIFFClose(); TIFFCleanup() doesn't free internal memstream.
+ *      (2) Returns res = 0 if not set in the file.
  * </pre>
  */
 l_ok
 readHeaderMemTiff(const l_uint8  *cdata,
                   size_t          size,
                   l_int32         n,
@@ -2012,18 +2013,16 @@ l_uint32   w, h;
         return ERROR_INT("spp not in set {1,2,3,4}", procName, 1);
     if (pw) *pw = w;
     if (ph) *ph = h;
     if (pbps) *pbps = bps;
     if (pspp) *pspp = spp;
     if (pres) {
-        *pres = 300;  /* default ppi */
         if (getTiffStreamResolution(tif, &xres, &yres) == 0)
             *pres = (l_int32)xres;
     }
     if (pcmap) {
-        *pcmap = 0;
         if (TIFFGetField(tif, TIFFTAG_COLORMAP, &rmap, &gmap, &bmap))
             *pcmap = 1;
     }
     if (pformat) {
         TIFFGetFieldDefaulted(tif, TIFFTAG_COMPRESSION, &tiffcomp);
         *pformat = getTiffCompressedFormat(tiffcomp);
