diff --git a/programs/fuzz/fuzz_client.c b/programs/fuzz/fuzz_client.c
index 21d03a3c34c3430e51cfcf57bd111fd32551863b..07ca96efa8ef86f4249adfb4ec89e83c8a7f7850 100644
--- a/programs/fuzz/fuzz_client.c
+++ b/programs/fuzz/fuzz_client.c
@@ -1,4 +1,3 @@
-
 #include "mbedtls/ssl.h"
 #include "mbedtls/entropy.h"
 #include "mbedtls/ctr_drbg.h"
diff --git a/programs/fuzz/fuzz_dtlsclient.c b/programs/fuzz/fuzz_dtlsclient.c
index 53f3f786a5239629bb9d19e21806fe1cdf19bb83..6581dcb1e698feaa7c1dca487a46195c0ff1a13a 100644
--- a/programs/fuzz/fuzz_dtlsclient.c
+++ b/programs/fuzz/fuzz_dtlsclient.c
@@ -1,4 +1,3 @@
-
 #include <string.h>
 #include <stdlib.h>
 #include <stdint.h>
diff --git a/programs/fuzz/fuzz_server.c b/programs/fuzz/fuzz_server.c
index 0c88767cfd089d38f3c5e79ce5441dd50eee1c8c..64b35edb9b059700e12015e1a0eb4557ee7318ed 100644
--- a/programs/fuzz/fuzz_server.c
+++ b/programs/fuzz/fuzz_server.c
@@ -1,4 +1,3 @@
-
 #include "mbedtls/ssl.h"
 #include "mbedtls/entropy.h"
 #include "mbedtls/ctr_drbg.h"
diff --git a/programs/fuzz/fuzz_x509crl.c b/programs/fuzz/fuzz_x509crl.c
index 89c3ff69bd5b0bdb195323df8184069091e63ddd..151db92c890c83228bfb964478c367a4f430b772 100644
--- a/programs/fuzz/fuzz_x509crl.c
+++ b/programs/fuzz/fuzz_x509crl.c
@@ -1,4 +1,3 @@
-
 #include <stdint.h>
 #include "mbedtls/x509_crl.h"
 
diff --git a/programs/fuzz/fuzz_x509crt.c b/programs/fuzz/fuzz_x509crt.c
index 9f40cb6ae28043ccd718d29b5076ecf7cffa5528..3eee07258bc97958d45ab5d05820f935d150999c 100644
--- a/programs/fuzz/fuzz_x509crt.c
+++ b/programs/fuzz/fuzz_x509crt.c
@@ -1,4 +1,3 @@
-
 #include <stdint.h>
 #include "mbedtls/x509_crt.h"
 
diff --git a/programs/fuzz/fuzz_x509csr.c b/programs/fuzz/fuzz_x509csr.c
index e19005afe050002d0256b789f909ef6b8a0cf1e5..7946e57eda03d28dc6b5cd2f6de2fcb9748e54ca 100644
--- a/programs/fuzz/fuzz_x509csr.c
+++ b/programs/fuzz/fuzz_x509csr.c
@@ -1,4 +1,3 @@
-
 #include <stdint.h>
 #include "mbedtls/x509_csr.h"
 
