diff --git a/programs/ssl/ssl_context_info.c b/programs/ssl/ssl_context_info.c
index 1e3e26961fa0f4739df4b8b1491d6421a297139c..ee2cdb7b963d0cd5510946f0db03b0b9f00b697b 100644
--- a/programs/ssl/ssl_context_info.c
+++ b/programs/ssl/ssl_context_info.c
@@ -5,8 +5,6 @@
  *  SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
  */
 
-//#define MBEDTLS_ALLOW_PRIVATE_ACCESS
-
 #include "mbedtls/build_info.h"
 #include "mbedtls/debug.h"
 #include "mbedtls/platform.h"
