diff --git a/test/tools/ossfuzz/strictasm_assembly_ossfuzz.cpp b/test/tools/ossfuzz/strictasm_assembly_ossfuzz.cpp
index fe9f9609ef8ab1171411e7f9d5a701b5c7193ef7..e795d047d7092c7fa186ee7d18ee545acdc2be65 100644
--- a/test/tools/ossfuzz/strictasm_assembly_ossfuzz.cpp
+++ b/test/tools/ossfuzz/strictasm_assembly_ossfuzz.cpp
@@ -27,6 +27,8 @@ extern "C" int LLVMFuzzerTestOneInput(uint8_t const* _data, size_t _size)
 	if (_size > 600)
 		return 0;
 
+	YulStringRepository::reset();
+
 	string input(reinterpret_cast<char const*>(_data), _size);
 	AssemblyStack stack(
 		langutil::EVMVersion(),
diff --git a/test/tools/ossfuzz/strictasm_diff_ossfuzz.cpp b/test/tools/ossfuzz/strictasm_diff_ossfuzz.cpp
index f7f077bd59aa80b56ca8c1e996e5424695121a6c..c61bf981726a8ec462a12a2172595a9838670cd9 100644
--- a/test/tools/ossfuzz/strictasm_diff_ossfuzz.cpp
+++ b/test/tools/ossfuzz/strictasm_diff_ossfuzz.cpp
@@ -55,6 +55,8 @@ extern "C" int LLVMFuzzerTestOneInput(uint8_t const* _data, size_t _size)
 	}))
 		return 0;
 
+	YulStringRepository::reset();
+
 	AssemblyStack stack(
 		langutil::EVMVersion(),
 		AssemblyStack::Language::StrictAssembly,
diff --git a/test/tools/ossfuzz/strictasm_opt_ossfuzz.cpp b/test/tools/ossfuzz/strictasm_opt_ossfuzz.cpp
index 28e1984b24a2573ac672eb5d6a32804be150adb5..70c3858478f378d4d24d1323d01733adf7d10916 100644
--- a/test/tools/ossfuzz/strictasm_opt_ossfuzz.cpp
+++ b/test/tools/ossfuzz/strictasm_opt_ossfuzz.cpp
@@ -26,6 +26,8 @@ extern "C" int LLVMFuzzerTestOneInput(uint8_t const* _data, size_t _size)
 	if (_size > 600)
 		return 0;
 
+	YulStringRepository::reset();
+
 	string input(reinterpret_cast<char const*>(_data), _size);
 	AssemblyStack stack(
 		langutil::EVMVersion(),
diff --git a/test/tools/ossfuzz/yulProtoFuzzer.cpp b/test/tools/ossfuzz/yulProtoFuzzer.cpp
index 1cdfc124e089593b7e194b0362a4a19d496ee0c1..604fc88d663dcdf613ccec6988568dd78b04b3d7 100644
--- a/test/tools/ossfuzz/yulProtoFuzzer.cpp
+++ b/test/tools/ossfuzz/yulProtoFuzzer.cpp
@@ -46,6 +46,8 @@ DEFINE_PROTO_FUZZER(Program const& _input)
 	if (yul_source.size() > 1200)
 		return;
 
+	YulStringRepository::reset();
+
 	// AssemblyStack entry point
 	AssemblyStack stack(
 		langutil::EVMVersion(),
diff --git a/test/tools/ossfuzz/yulProto_diff_ossfuzz.cpp b/test/tools/ossfuzz/yulProto_diff_ossfuzz.cpp
index 9cafeaff647e4c69e231d4fc9bb6b6d511e847e2..4102bb4b024f435286ed70dc6649213e49dfc90a 100644
--- a/test/tools/ossfuzz/yulProto_diff_ossfuzz.cpp
+++ b/test/tools/ossfuzz/yulProto_diff_ossfuzz.cpp
@@ -53,6 +53,8 @@ DEFINE_PROTO_FUZZER(Program const& _input)
 	if (yul_source.size() > 1200)
 		return;
 
+	YulStringRepository::reset();
+
 	// AssemblyStack entry point
 	AssemblyStack stack(
 		langutil::EVMVersion(),
