diff --git a/test/tools/fuzzer_common.cpp b/test/tools/fuzzer_common.cpp
index 094a8455485bef34e0082fcab2a6d3368fa61c71..25ad6986cbc8628ec24f640188636958b002a3a4 100644
--- a/test/tools/fuzzer_common.cpp
+++ b/test/tools/fuzzer_common.cpp
@@ -48,7 +48,9 @@ static vector<EVMVersion> s_evmVersions = {
 	EVMVersion::constantinople(),
 	EVMVersion::petersburg(),
 	EVMVersion::istanbul(),
-	EVMVersion::berlin()
+	EVMVersion::berlin(),
+	EVMVersion::london(),
+	EVMVersion::paris()
 };
 
 void FuzzerUtil::testCompilerJsonInterface(string const& _input, bool _optimize, bool _quiet)
diff --git a/test/tools/ossfuzz/protoToYul.cpp b/test/tools/ossfuzz/protoToYul.cpp
index f6759dddb48876e87a05e31bf9160a65653ba185..cd2898e1288ae2ea6e3d06361197d4261912c7c7 100644
--- a/test/tools/ossfuzz/protoToYul.cpp
+++ b/test/tools/ossfuzz/protoToYul.cpp
@@ -112,6 +112,10 @@ EVMVersion ProtoConverter::evmVersionMapping(Program_Version const& _ver)
 		return EVMVersion::istanbul();
 	case Program::BERLIN:
 		return EVMVersion::berlin();
+	case Program::LONDON:
+		return EVMVersion::london();
+	case Program::PARIS:
+		return EVMVersion::paris();
 	}
 }
 
