diff --git a/liblangutil/EVMVersion.h b/liblangutil/EVMVersion.h
index 268df7d9d65aefabb0133f3b8ba0fc6157290f98..d366338451c7a3612cd6fec08851312f7febe48e 100644
--- a/liblangutil/EVMVersion.h
+++ b/liblangutil/EVMVersion.h
@@ -56,10 +56,11 @@ public:
 	static EVMVersion istanbul() { return {Version::Istanbul}; }
 	static EVMVersion berlin() { return {Version::Berlin}; }
 	static EVMVersion london() { return {Version::London}; }
+	static EVMVersion paris() { return {Version::Paris}; }
 
 	static std::optional<EVMVersion> fromString(std::string const& _version)
 	{
-		for (auto const& v: {homestead(), tangerineWhistle(), spuriousDragon(), byzantium(), constantinople(), petersburg(), istanbul(), berlin(), london()})
+		for (auto const& v: {homestead(), tangerineWhistle(), spuriousDragon(), byzantium(), constantinople(), petersburg(), istanbul(), berlin(), london(), paris()})
 			if (_version == v.name())
 				return v;
 		return std::nullopt;
@@ -81,6 +82,7 @@ public:
 		case Version::Istanbul: return "istanbul";
 		case Version::Berlin: return "berlin";
 		case Version::London: return "london";
+		case Version::Paris: return "paris";
 		}
 		return "INVALID";
 	}
@@ -102,7 +104,7 @@ public:
 	bool canOverchargeGasForCall() const { return *this >= tangerineWhistle(); }
 
 private:
-	enum class Version { Homestead, TangerineWhistle, SpuriousDragon, Byzantium, Constantinople, Petersburg, Istanbul, Berlin, London };
+	enum class Version { Homestead, TangerineWhistle, SpuriousDragon, Byzantium, Constantinople, Petersburg, Istanbul, Berlin, London, Paris };
 
 	EVMVersion(Version _version): m_version(_version) {}
 
diff --git a/solc/CommandLineParser.cpp b/solc/CommandLineParser.cpp
index a2e4a0d661b3563e35e91d90e97546a511a7046e..2a1ce022804e811cc369078c52c5454925ba1203 100644
--- a/solc/CommandLineParser.cpp
+++ b/solc/CommandLineParser.cpp
@@ -586,7 +586,7 @@ General Information)").c_str(),
 			g_strEVMVersion.c_str(),
 			po::value<string>()->value_name("version")->default_value(EVMVersion{}.name()),
 			"Select desired EVM version. Either homestead, tangerineWhistle, spuriousDragon, "
-			"byzantium, constantinople, petersburg, istanbul, berlin or london."
+			"byzantium, constantinople, petersburg, istanbul, berlin, london or paris."
 		)
 		(
 			g_strExperimentalViaIR.c_str(),
