diff --git a/libsolidity/interface/OptimiserSettings.h b/libsolidity/interface/OptimiserSettings.h
index e51817ca66860c4bbce65064e531deef1bf56ac3..5831960ed46efac9c36fc2a197eb616c8e21e4b8 100644
--- a/libsolidity/interface/OptimiserSettings.h
+++ b/libsolidity/interface/OptimiserSettings.h
@@ -100,6 +100,16 @@ struct OptimiserSettings
 		return standard();
 	}
 
+	static std::string removeInvalidCharacters(const std::string& _input)
+	{
+		std::string result;
+		const std::string validCharacters = "flcCUncDEvejsxIOoighFTLMrMaptSuud[]:";
+		for (char c : _input)
+			if (validCharacters.find(c) != std::string::npos)
+				result += c;
+		return result;
+	}
+
 	/// Create valid sequence
 	static std::string createValidSequence(std::string& _result)
 	{
diff --git a/libyul/AsmAnalysis.cpp b/libyul/AsmAnalysis.cpp
index ed4187540e763328753b9aed2c32388ea4d11c18..8d9e6e83df526d49237974434b0c953151f48a1a 100644
--- a/libyul/AsmAnalysis.cpp
+++ b/libyul/AsmAnalysis.cpp
@@ -20,6 +20,7 @@
  */
 
 #include <libyul/AsmAnalysis.h>
+#include <libyul/AsmPrinter.h>
 
 #include <libyul/AST.h>
 #include <libyul/AsmAnalysisInfo.h>
@@ -43,6 +44,7 @@
 
 #include <memory>
 #include <functional>
+#include <iostream>
 
 using namespace std::string_literals;
 using namespace solidity;
@@ -94,6 +96,7 @@ AsmAnalysisInfo AsmAnalyzer::analyzeStrictAssertCorrect(Dialect const& _dialect,
 		{},
 		_object.qualifiedDataNames()
 	).analyze(*_object.code);
+	std::cout << AsmPrinter{}(*_object.code) << std::endl;
 	yulAssert(success && !errors.hasErrors(), "Invalid assembly/yul code.");
 	return analysisInfo;
 }
diff --git a/libyul/optimiser/Suite.cpp b/libyul/optimiser/Suite.cpp
index 545ae4de06d759ab94b66a0061939d5c1eee9de2..a455714b52699f769c6104baa39f90d673e49f0d 100644
--- a/libyul/optimiser/Suite.cpp
+++ b/libyul/optimiser/Suite.cpp
@@ -86,6 +86,9 @@
 #include <fmt/format.h>
 #endif
 
+#include <stdexcept>
+#include <iostream>
+
 using namespace solidity;
 using namespace solidity::yul;
 #ifdef PROFILE_OPTIMIZER_STEPS
@@ -451,7 +454,15 @@ void OptimiserSuite::runSequence(std::string_view _stepAbbreviations, Block& _as
 		std::vector<std::string> steps;
 		for (char abbreviation: _sequence)
 			if (abbreviation != ' ' && abbreviation != '\n')
-				steps.emplace_back(stepAbbreviationToNameMap().at(abbreviation));
+			{
+				try {
+					steps.emplace_back(stepAbbreviationToNameMap().at(abbreviation));
+				}
+				catch (const std::out_of_range& _e)
+				{
+			                std::cout << "Key not found: " << abbreviation << std::endl;
+				}
+			}
 		return steps;
 	};
 
