diff --git a/liblangutil/EVMVersion.h b/liblangutil/EVMVersion.h
index 58b54f3dcaa71c7e3239e6a7eb8c4cf484e181c1..774268917483d6559064d7edb9adf3990bbbc542 100644
--- a/liblangutil/EVMVersion.h
+++ b/liblangutil/EVMVersion.h
@@ -60,10 +60,25 @@ public:
 	static EVMVersion paris() { return {Version::Paris}; }
 	static EVMVersion shanghai() { return {Version::Shanghai}; }
 	static EVMVersion cancun() { return {Version::Cancun}; }
+	static EVMVersion prague() { return {Version::Prague}; }
 
 	static std::optional<EVMVersion> fromString(std::string const& _version)
 	{
-		for (auto const& v: {homestead(), tangerineWhistle(), spuriousDragon(), byzantium(), constantinople(), petersburg(), istanbul(), berlin(), london(), paris(), shanghai(), cancun()})
+		for (auto const& v: {
+			homestead(),
+			tangerineWhistle(),
+			spuriousDragon(),
+			byzantium(),
+			constantinople(),
+			petersburg(),
+			istanbul(),
+			berlin(),
+			london(),
+			paris(),
+			shanghai(),
+			cancun(),
+			prague()
+		})
 			if (_version == v.name())
 				return v;
 		return std::nullopt;
@@ -88,6 +103,7 @@ public:
 		case Version::Paris: return "paris";
 		case Version::Shanghai: return "shanghai";
 		case Version::Cancun: return "cancun";
+		case Version::Prague: return "prague";
 		}
 		return "INVALID";
 	}
@@ -115,7 +131,21 @@ public:
 	bool canOverchargeGasForCall() const { return *this >= tangerineWhistle(); }
 
 private:
-	enum class Version { Homestead, TangerineWhistle, SpuriousDragon, Byzantium, Constantinople, Petersburg, Istanbul, Berlin, London, Paris, Shanghai, Cancun };
+	enum class Version {
+		Homestead,
+		TangerineWhistle,
+		SpuriousDragon,
+		Byzantium,
+		Constantinople,
+		Petersburg,
+		Istanbul,
+		Berlin,
+		London,
+		Paris,
+		Shanghai,
+		Cancun,
+		Prague
+	};
 
 	EVMVersion(Version _version): m_version(_version) {}
 
diff --git a/solc/CommandLineParser.cpp b/solc/CommandLineParser.cpp
index 824c899110a24350f419797f2ed964952a705eaa..e61c5ce0a7aeb7341ba1a9fa2b809fed546940ac 100644
--- a/solc/CommandLineParser.cpp
+++ b/solc/CommandLineParser.cpp
@@ -605,7 +605,7 @@ General Information)").c_str(),
 			g_strEVMVersion.c_str(),
 			po::value<std::string>()->value_name("version")->default_value(EVMVersion{}.name()),
 			"Select desired EVM version. Either homestead, tangerineWhistle, spuriousDragon, "
-			"byzantium, constantinople, petersburg, istanbul, berlin, london, paris, shanghai or cancun."
+			"byzantium, constantinople, petersburg, istanbul, berlin, london, paris, shanghai, cancun or prague."
 		)
 	;
 	if (!_forHelp) // Note: We intentionally keep this undocumented for now.
