diff --git a/test/tools/fuzzer_common.cpp b/test/tools/fuzzer_common.cpp
index b054883820fcb61a37adfd1fb8706cc8b534d294..8b7dbffa8cfa7744f1f2802e18f5d8b6e149fbf8 100644
--- a/test/tools/fuzzer_common.cpp
+++ b/test/tools/fuzzer_common.cpp
@@ -49,7 +49,8 @@ static std::vector<EVMVersion> s_evmVersions = {
 	EVMVersion::istanbul(),
 	EVMVersion::berlin(),
 	EVMVersion::london(),
-	EVMVersion::paris()
+	EVMVersion::paris(),
+	EVMVersion::prague()
 };
 
 void FuzzerUtil::testCompilerJsonInterface(std::string const& _input, bool _optimize, bool _quiet)
diff --git a/test/tools/ossfuzz/protoToYul.cpp b/test/tools/ossfuzz/protoToYul.cpp
index 8822666c8b02fbba343fdaee860739d2f6f9a7ae..7913f4d634500fe85665876adb33f1f94ed64c8b 100644
--- a/test/tools/ossfuzz/protoToYul.cpp
+++ b/test/tools/ossfuzz/protoToYul.cpp
@@ -119,6 +119,8 @@ EVMVersion ProtoConverter::evmVersionMapping(Program_Version const& _ver)
 		return EVMVersion::shanghai();
 	case Program::CANCUN:
 		return EVMVersion::cancun();
+	case Program::PRAGUE:
+		return EVMVersion::prague();
 	}
 }
 
