diff --git a/fuzz/c/std/gif_fuzzer.cc b/fuzz/c/std/gif_fuzzer.cc
index 284225b3932f323905c016a7a9298976ab03d946..06c983137777109ab7e143a3f69639f320eb3b0d 100644
--- a/fuzz/c/std/gif_fuzzer.cc
+++ b/fuzz/c/std/gif_fuzzer.cc
@@ -50,7 +50,7 @@ void fuzz(wuffs_base__reader1 src_reader, uint32_t hash) {
 
     wuffs_base__image_config ic = {{0}};
     s = wuffs_gif__decoder__decode_config(&dec, &ic, src_reader);
-    if (s || !wuffs_base__image_config__valid(&ic)) {
+    if (s || !wuffs_base__image_config__is_valid(&ic)) {
       goto exit;
     }
 
