diff --git a/fuzz/c/std/zlib_fuzzer.cc b/fuzz/c/std/zlib_fuzzer.cc
index 8fd556c8daea80ae883980fb807d8560eca3d293..8b5f8878e32049c5406a4215e3a4edd9ee7b7da1 100644
--- a/fuzz/c/std/zlib_fuzzer.cc
+++ b/fuzz/c/std/zlib_fuzzer.cc
@@ -35,6 +35,7 @@ It should print "PASS", amongst other information, and exit(0).
 // If building this program in an environment that doesn't easily accomodate
 // relative includes, you can use the script/inline-c-relative-includes.go
 // program to generate a stand-alone C file.
+#include "../../../gen/c/std/adler32.c"
 #include "../../../gen/c/std/deflate.c"
 #include "../../../gen/c/std/zlib.c"
 #include "../fuzzlib/fuzzlib.cc"
