diff --git a/fuzz/c/fuzzlib/fuzzlib.cc b/fuzz/c/fuzzlib/fuzzlib.cc
index cff1c08fe18fa26e1df64454b2c4b8cdcac16726..96d20eebb05210354b49b464f1354a9f6e2beb74 100644
--- a/fuzz/c/fuzzlib/fuzzlib.cc
+++ b/fuzz/c/fuzzlib/fuzzlib.cc
@@ -46,7 +46,7 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
                                .wi = size,
                                .ri = 0,
                                .closed = true};
-  fuzz((wuffs_base__io_reader){.buf = &src}, hash);
+  fuzz(wuffs_base__io_buffer__reader(&src), hash);
   return 0;
 }
 
diff --git a/fuzz/c/std/gif_fuzzer.cc b/fuzz/c/std/gif_fuzzer.cc
index a2177da116408318e07414c98a113a9eade22f50..380bea359d40d19d1a3e5d1de034e2de47cf58cd 100644
--- a/fuzz/c/std/gif_fuzzer.cc
+++ b/fuzz/c/std/gif_fuzzer.cc
@@ -65,7 +65,7 @@ void fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
     }
 
     wuffs_base__io_buffer dst = {.ptr = (uint8_t*)(pixbuf), .len = pixbuf_size};
-    wuffs_base__io_writer dst_writer = {.buf = &dst};
+    wuffs_base__io_writer dst_writer = wuffs_base__io_buffer__writer(&dst);
 
     while (true) {
       // TODO: handle the frame rect being larger than the image rect. The
diff --git a/fuzz/c/std/zlib_fuzzer.cc b/fuzz/c/std/zlib_fuzzer.cc
index 8b5f8878e32049c5406a4215e3a4edd9ee7b7da1..85538c6a0d36328bec3ac041e8937240a510a498 100644
--- a/fuzz/c/std/zlib_fuzzer.cc
+++ b/fuzz/c/std/zlib_fuzzer.cc
@@ -53,7 +53,7 @@ void fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
   const size_t dstbuf_size = 65536;
   uint8_t dstbuf[dstbuf_size];
   wuffs_base__io_buffer dst = {.ptr = dstbuf, .len = dstbuf_size};
-  wuffs_base__io_writer dst_writer = {.buf = &dst};
+  wuffs_base__io_writer dst_writer = wuffs_base__io_buffer__writer(&dst);
 
   while (true) {
     dst.wi = 0;
