diff --git a/cmd/wuffs-c/internal/cgen/base-header.h b/cmd/wuffs-c/internal/cgen/base-header.h
index 5bb272e5cb0a174d9f693a3d4a64d07a62072579..580c5ee254d90f8c0d3f10e9db444453b0462c87 100644
--- a/cmd/wuffs-c/internal/cgen/base-header.h
+++ b/cmd/wuffs-c/internal/cgen/base-header.h
@@ -586,6 +586,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/c/std/adler32.c b/gen/c/std/adler32.c
index 191573d1347aa36c1cdf8a75d510bdb7871ca2dd..c2ff14f517c38b4468c9cc9aaa953838859e045a 100644
--- a/gen/c/std/adler32.c
+++ b/gen/c/std/adler32.c
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/c/std/crc32.c b/gen/c/std/crc32.c
index f554a0e27761dd90c3e0252ac066f7f1c6b42adf..9acad6143a37a9f8434f5f1f0b8d008e9093c3ae 100644
--- a/gen/c/std/crc32.c
+++ b/gen/c/std/crc32.c
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/c/std/deflate.c b/gen/c/std/deflate.c
index 1fd11ca5be89b6a0f965482a589dd6b0f023d461..83f81f13cf1cba50897213f8eda33bcf65eb74c9 100644
--- a/gen/c/std/deflate.c
+++ b/gen/c/std/deflate.c
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/c/std/gif.c b/gen/c/std/gif.c
index 5a20ebb873cfd6a848513e242ed95c15bd2a7525..6b4936c8da6cba4ae7d2408cc86efbbda4ec4eb7 100644
--- a/gen/c/std/gif.c
+++ b/gen/c/std/gif.c
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/c/std/gzip.c b/gen/c/std/gzip.c
index 2383f8df3d68290a7ccae7a655704202a29a7204..f616b4fef311e87615eab352873b816e552e2cbb 100644
--- a/gen/c/std/gzip.c
+++ b/gen/c/std/gzip.c
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/c/std/zlib.c b/gen/c/std/zlib.c
index 6b842d982cfa37f4eb9867ce0ab6c4619b9f9622..24f8517a2aea102955826c52d3c9fe926899a9b3 100644
--- a/gen/c/std/zlib.c
+++ b/gen/c/std/zlib.c
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/h/std/adler32.h b/gen/h/std/adler32.h
index 06ee53486a255d7b21f6a792bda28b5d33b8c65a..9f53c0ab9684d6e077c3d97f8788418be72ffa2f 100644
--- a/gen/h/std/adler32.h
+++ b/gen/h/std/adler32.h
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/h/std/crc32.h b/gen/h/std/crc32.h
index 39701c5f5b6d6d4d72a9b4e31734cdc9c902e0e5..c1e63546e795cf4c70af8a3dd16521c414c07ee0 100644
--- a/gen/h/std/crc32.h
+++ b/gen/h/std/crc32.h
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/h/std/deflate.h b/gen/h/std/deflate.h
index 820c024575cf3bc6c3130c28568961fb98e0a6d2..ff92234329ae8a7b9e3d6e589ca60ed80f2e7d08 100644
--- a/gen/h/std/deflate.h
+++ b/gen/h/std/deflate.h
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/h/std/gif.h b/gen/h/std/gif.h
index 795a47d343179ec78747dd8a718a7249da30958a..d0cc7940a829e5501b3b3d9af22f3555fb285db4 100644
--- a/gen/h/std/gif.h
+++ b/gen/h/std/gif.h
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/h/std/gzip.h b/gen/h/std/gzip.h
index 0907af26aa90958ab80f4bf7b06e18114aa967cd..bb73418262edc34fbcaa4f4eb79218b30a2cbba0 100644
--- a/gen/h/std/gzip.h
+++ b/gen/h/std/gzip.h
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/gen/h/std/zlib.h b/gen/h/std/zlib.h
index d143453ad664b3648e790a084263216e9bfbd668..54daecbdffe685899362e1c7ab129288a4ddc83e 100644
--- a/gen/h/std/zlib.h
+++ b/gen/h/std/zlib.h
@@ -589,6 +589,24 @@ typedef struct {
   } private_impl;
 } wuffs_base__io_writer;
 
+static inline wuffs_base__io_reader wuffs_base__io_buffer__reader(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_reader ret = ((wuffs_base__io_reader){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
+static inline wuffs_base__io_writer wuffs_base__io_buffer__writer(
+    wuffs_base__io_buffer* buf) {
+  wuffs_base__io_writer ret = ((wuffs_base__io_writer){});
+  if (buf) {
+    ret.buf = buf;
+  }
+  return ret;
+}
+
 // ---------------- Images
 
 // wuffs_base__pixel_format encodes the format of the bytes that constitute an
diff --git a/script/bench-c-deflate-fragmentation.c b/script/bench-c-deflate-fragmentation.c
index adb40ce55823072976ee0aa6fa51dcbd26361e85..b6dde288d6772a8d25bd308c46c5f1e0b9a140db 100644
--- a/script/bench-c-deflate-fragmentation.c
+++ b/script/bench-c-deflate-fragmentation.c
@@ -218,8 +218,8 @@ const char* decode_once(bool frag_dst, bool frag_idat) {
                                 .len = SRC_BUFFER_SIZE,
                                 .wi = idat_splits[num_idat_chunks],
                                 .closed = true};
-  wuffs_base__io_writer dst_writer = {.buf = &dst};
-  wuffs_base__io_reader idat_reader = {.buf = &idat};
+  wuffs_base__io_writer dst_writer = wuffs_base__io_buffer__writer(&dst);
+  wuffs_base__io_reader idat_reader = wuffs_base__io_buffer__reader(&idat);
 
   uint32_t i = 0;  // Number of dst fragments processed, if frag_dst.
   if (frag_dst) {
