diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 5d04847f522e331a8133b108d672acf7bd1b4d7e..8d7bb06f3b46c6a2908272f37bcb4e57889e844c 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -45,12 +45,12 @@ const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
   // Use a {} code block so that "goto exit" doesn't trigger "jump bypasses
   // variable initialization" warnings.
   {
-    wuffs_gif__status s;
-    wuffs_gif__decoder dec;
-    wuffs_gif__decoder__initialize(&dec, WUFFS_VERSION, 0);
+    wuffs_gif__status s = 0;
+    wuffs_gif__decoder dec = ((wuffs_gif__decoder){});
+    wuffs_gif__decoder__check_wuffs_version(&dec, WUFFS_VERSION, sizeof dec);
 
     wuffs_base__image_buffer ib = ((wuffs_base__image_buffer){});
-    wuffs_base__image_config ic = {{0}};
+    wuffs_base__image_config ic = ((wuffs_base__image_config){});
     s = wuffs_gif__decoder__decode_config(&dec, &ic, src_reader);
     if (s) {
       ret = wuffs_gif__status__string(s);
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index 23a467f7213399a6f6f7460f09faecaa883ccfae..ba871516c5cc0004b486dc2e63c062d63789173b 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -42,9 +42,9 @@ It should print "PASS", amongst other information, and exit(0).
 
 const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
   const char* ret = NULL;
-  wuffs_zlib__status s;
-  wuffs_zlib__decoder dec;
-  wuffs_zlib__decoder__initialize(&dec, WUFFS_VERSION, 0);
+  wuffs_zlib__status s = 0;
+  wuffs_zlib__decoder dec = ((wuffs_zlib__decoder){});
+  wuffs_zlib__decoder__check_wuffs_version(&dec, WUFFS_VERSION, sizeof dec);
 
   // Ignore the checksum for 99.99%-ish of all input. When fuzzers generate
   // random input, the checkum is very unlikely to match. Still, it's useful to
