diff --git a/cmd/wuffs-c/internal/cgen/base/base-public.h b/cmd/wuffs-c/internal/cgen/base/base-public.h
index a2c47bb2c7ae45a9009e2a44ca49f0b572972b56..eed95720f4bd23c65ef0533b31764db4bde3a550 100644
--- a/cmd/wuffs-c/internal/cgen/base/base-public.h
+++ b/cmd/wuffs-c/internal/cgen/base/base-public.h
@@ -1075,32 +1075,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1108,6 +1108,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/c/base.c b/gen/c/base.c
index f5146c49d9991cdf4c161e4acddf6513f0de38c3..c4a95a38479b9014d5e54106b9f63511b16f9100 100644
--- a/gen/c/base.c
+++ b/gen/c/base.c
@@ -84,11 +84,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1105,32 +1106,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1138,6 +1139,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
@@ -1270,11 +1272,13 @@ static const char wuffs_base__status__string_data[] = {
     0x6B, 0x5F, 0x77, 0x75, 0x66, 0x66, 0x73, 0x5F, 0x76, 0x65, 0x72, 0x73,
     0x69, 0x6F, 0x6E, 0x20, 0x6E, 0x6F, 0x74, 0x20, 0x63, 0x61, 0x6C, 0x6C,
     0x65, 0x64, 0x00, 0x62, 0x61, 0x64, 0x20, 0x61, 0x72, 0x67, 0x75, 0x6D,
-    0x65, 0x6E, 0x74, 0x00, 0x62, 0x61, 0x64, 0x20, 0x72, 0x65, 0x63, 0x65,
-    0x69, 0x76, 0x65, 0x72, 0x00, 0x62, 0x61, 0x64, 0x20, 0x73, 0x69, 0x7A,
-    0x65, 0x6F, 0x66, 0x20, 0x72, 0x65, 0x63, 0x65, 0x69, 0x76, 0x65, 0x72,
-    0x00, 0x62, 0x61, 0x64, 0x20, 0x77, 0x75, 0x66, 0x66, 0x73, 0x20, 0x76,
-    0x65, 0x72, 0x73, 0x69, 0x6F, 0x6E, 0x00,
+    0x65, 0x6E, 0x74, 0x20, 0x28, 0x6C, 0x65, 0x6E, 0x67, 0x74, 0x68, 0x20,
+    0x74, 0x6F, 0x6F, 0x20, 0x73, 0x68, 0x6F, 0x72, 0x74, 0x29, 0x00, 0x62,
+    0x61, 0x64, 0x20, 0x72, 0x65, 0x63, 0x65, 0x69, 0x76, 0x65, 0x72, 0x00,
+    0x62, 0x61, 0x64, 0x20, 0x73, 0x69, 0x7A, 0x65, 0x6F, 0x66, 0x20, 0x72,
+    0x65, 0x63, 0x65, 0x69, 0x76, 0x65, 0x72, 0x00, 0x62, 0x61, 0x64, 0x20,
+    0x77, 0x75, 0x66, 0x66, 0x73, 0x20, 0x76, 0x65, 0x72, 0x73, 0x69, 0x6F,
+    0x6E, 0x00,
 };
 
 static const uint16_t wuffs_base__status__string_offsets[] = {
@@ -1306,7 +1310,7 @@ static const uint16_t wuffs_base__status__string_offsets[] = {
     0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000,
     0x0000, 0x0000, 0x0000, 0x0000, 0x0079, 0x008F, 0x00B0, 0x0000, 0x0000,
     0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000,
-    0x00CF, 0x00DC, 0x00E9, 0x00FD,
+    0x00CF, 0x00EF, 0x00FC, 0x0110,
 };
 
 const char* wuffs_base__status__string(wuffs_base__status s) {
diff --git a/gen/c/std/adler32.c b/gen/c/std/adler32.c
index 93f0fec8fae77f31b1111e78d4d17a2187e6fd3a..4ecd8d15d148d19e4fc67286336388d2a8247187 100644
--- a/gen/c/std/adler32.c
+++ b/gen/c/std/adler32.c
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/c/std/crc32.c b/gen/c/std/crc32.c
index 90bbdbfc50a9bf5900a11995783bfc34d886c9dd..50dd8f2d7c69d2f3017769b55e11142fcf7234f1 100644
--- a/gen/c/std/crc32.c
+++ b/gen/c/std/crc32.c
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/c/std/deflate.c b/gen/c/std/deflate.c
index 5f19f577bebc09d8f0a28450063bb34aca0264c1..c0db9a350211bdb5ddab91956398341cc6ebbf44 100644
--- a/gen/c/std/deflate.c
+++ b/gen/c/std/deflate.c
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/c/std/gif.c b/gen/c/std/gif.c
index 9e8d9cf937061a79ce8c47735cb71b6470d1e8e1..fc85221c3f4bb7c46ae8aa23bf3c033a542ea563 100644
--- a/gen/c/std/gif.c
+++ b/gen/c/std/gif.c
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/c/std/gzip.c b/gen/c/std/gzip.c
index 94396c1b3ea8fab1af699ee55980e858be4124e0..18c125f0d7477247a7036eddd99ab54f5b6dfbdc 100644
--- a/gen/c/std/gzip.c
+++ b/gen/c/std/gzip.c
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/c/std/lzw.c b/gen/c/std/lzw.c
index ab56e842dc3acdb41106b6c99758db7205599600..4efd214e061a7766f44080dc674d35c1ec25b48b 100644
--- a/gen/c/std/lzw.c
+++ b/gen/c/std/lzw.c
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/c/std/zlib.c b/gen/c/std/zlib.c
index 2f94ab3596c1cad15b315ecd06887b229d1eedbe..7ed19014868f1024bf1cb52e16dd067bc898204f 100644
--- a/gen/c/std/zlib.c
+++ b/gen/c/std/zlib.c
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/h/std/adler32.h b/gen/h/std/adler32.h
index 717dabe7444658e36c33bc3e182f39153176f9d1..fe3f1bcfd00fdb00b4296e5e164380220d206f4a 100644
--- a/gen/h/std/adler32.h
+++ b/gen/h/std/adler32.h
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/h/std/crc32.h b/gen/h/std/crc32.h
index 7c3d7c99fa7dac2d29eb5bb4c006c6465c2779d3..c5928ba6add93828766b24f124658ab0a35d3ea4 100644
--- a/gen/h/std/crc32.h
+++ b/gen/h/std/crc32.h
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/h/std/deflate.h b/gen/h/std/deflate.h
index eab84b54ee53587e9eb70b3adb7cb54de4d40de2..d5cb6ae308e66143d51f64694ed050b7a42d33e5 100644
--- a/gen/h/std/deflate.h
+++ b/gen/h/std/deflate.h
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/h/std/gif.h b/gen/h/std/gif.h
index 37b73f4bb8f8483bf75fcb4891c24f593062e870..06e6f8156555af22e374b3302221a18873187d22 100644
--- a/gen/h/std/gif.h
+++ b/gen/h/std/gif.h
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/h/std/gzip.h b/gen/h/std/gzip.h
index 631fe51845b9d4a03353e538673e5ae7e43d8449..351b9545ceac34862207408b2f21b8e896421c08 100644
--- a/gen/h/std/gzip.h
+++ b/gen/h/std/gzip.h
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/h/std/lzw.h b/gen/h/std/lzw.h
index 537ef34fb796157a1795ea22a88b7f1f967f1c55..34b0ca78532edd5ad833e6ff302b3716396e0fe9 100644
--- a/gen/h/std/lzw.h
+++ b/gen/h/std/lzw.h
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
diff --git a/gen/h/std/zlib.h b/gen/h/std/zlib.h
index dad8152590f793627da4bcd1f2409a432bd09793..6001f4f1a8f19c9658709bc8b37713d8ef7d54e3 100644
--- a/gen/h/std/zlib.h
+++ b/gen/h/std/zlib.h
@@ -73,11 +73,12 @@ typedef struct {
 // details. Use methods such as wuffs_base__status__is_error instead.
 typedef int32_t wuffs_base__status;
 
-#define WUFFS_BASE__STATUS_OK 0                          // 0x00000000
-#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216    // 0xFF000000
-#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432  // 0xFE000000
-#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648         // 0xFD000000
-#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864         // 0xFC000000
+#define WUFFS_BASE__STATUS_OK 0                                    // 0x00000000
+#define WUFFS_BASE__ERROR_BAD_WUFFS_VERSION -16777216              // 0xFF000000
+#define WUFFS_BASE__ERROR_BAD_SIZEOF_RECEIVER -33554432            // 0xFE000000
+#define WUFFS_BASE__ERROR_BAD_RECEIVER -50331648                   // 0xFD000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT -67108864                   // 0xFC000000
+#define WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT -67108864  // 0xFC000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_NOT_CALLED \
   -268435456  // 0xF0000000
 #define WUFFS_BASE__ERROR_CHECK_WUFFS_VERSION_CALLED_TWICE \
@@ -1094,32 +1095,32 @@ typedef struct {
   } private_impl;
 } wuffs_base__image_buffer;
 
-static inline void wuffs_base__image_buffer__set_from_pixbuf(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_pixbuf(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__pixel_buffer pixbuf) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   b->private_impl.config = config;
   b->private_impl.pixbuf = pixbuf;
+  return WUFFS_BASE__STATUS_OK;
 }
 
-// TODO: Should this function return bool? An error type?
-static inline void wuffs_base__image_buffer__set_from_slice(
+static inline wuffs_base__status wuffs_base__image_buffer__set_from_slice(
     wuffs_base__image_buffer* b,
     wuffs_base__image_config config,
     wuffs_base__slice_u8 pixbuf_memory) {
   if (!b) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_RECEIVER;
   }
   *b = ((wuffs_base__image_buffer){});
   // TODO: don't assume 1 byte per pixel. Don't assume packed.
   uint64_t wh = ((uint64_t)config.private_impl.width) *
                 ((uint64_t)config.private_impl.height);
   if (wh > pixbuf_memory.len) {
-    return;
+    return WUFFS_BASE__ERROR_BAD_ARGUMENT_LENGTH_TOO_SHORT;
   }
   b->private_impl.config = config;
   wuffs_base__table_u8* tab = &b->private_impl.pixbuf.planes[0];
@@ -1127,6 +1128,7 @@ static inline void wuffs_base__image_buffer__set_from_slice(
   tab->width = config.private_impl.width;
   tab->height = config.private_impl.height;
   tab->stride = config.private_impl.width;
+  return WUFFS_BASE__STATUS_OK;
 }
 
 // The palette argument is ignored unless its length is exactly 1024.
