diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 96e2418bd2559bc3ab7f2e8ce94f175f65266772..8fef0e228cedb00311f0189ae530a03fbd606079 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -74,9 +74,12 @@ const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
       ret = "out of memory";
       goto exit;
     }
-    // TODO: check wuffs_base__image_buffer__set_from_slice errors?
-    wuffs_base__image_buffer__set_from_slice(
+    s = wuffs_base__image_buffer__set_from_slice(
         &ib, ic, ((wuffs_base__slice_u8){.ptr = pixbuf, .len = pixbuf_size}));
+    if (s) {
+      ret = wuffs_gif__status__string(s);
+      goto exit;
+    }
 
     bool seen_ok = false;
     while (true) {
