diff --git a/script/bench-c-deflate-fragmentation.c b/script/bench-c-deflate-fragmentation.c
index 8c0ba5c24cb92226d949635765f2f822615457b8..e33322a18f1e6096bc8bf8bc75fbcca66535b4a6 100644
--- a/script/bench-c-deflate-fragmentation.c
+++ b/script/bench-c-deflate-fragmentation.c
@@ -66,7 +66,7 @@
 // compiling it.
 #define WUFFS_IMPLEMENTATION
 
-// If building this program in an environment that doesn't easily accomodate
+// If building this program in an environment that doesn't easily accommodate
 // relative includes, you can use the script/inline-c-relative-includes.go
 // program to generate a stand-alone C file.
 #include "../release/c/wuffs-unsupported-snapshot.h"
diff --git a/std/gif/decode_gif.wuffs b/std/gif/decode_gif.wuffs
index 896c4d250b1b28a41f0a7cb51b3004b4d8d263c3..f1662c58dc33541ea54d709819894428b0cf471e 100644
--- a/std/gif/decode_gif.wuffs
+++ b/std/gif/decode_gif.wuffs
@@ -557,7 +557,7 @@ pri func decoder.decode_id_part1?(dst ptr base.pixel_buffer, src base.io_reader)
 	// of bits for the decoder to read has ticked over from 8 to 9 just before
 	// that end code is encountered.
 	//
-	// To accomodate such malformed GIFs, we detect when the previous frame's
+	// To accommodate such malformed GIFs, we detect when the previous frame's
 	// LZW decoding ended abruptly. The previous LZW decode 'works', in that it
 	// decodes as much pixel data as is available, but without seeing that end
 	// code (i.e. returning the "ok" status code), the LZW decoder is stuck in
