diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 7452a3be1d32ce0f7e472e4a7722d0c338dcad52..53e629a2b15dd811ca4995daeabddf2a9d96119b 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -40,7 +40,7 @@ It should print "PASS", amongst other information, and exit(0).
 // compiling it.
 #define WUFFS_IMPLEMENTATION
 
-// If building this program in an environment that doesn't easily accomodate
+// If building this program in an environment that doesn't easily accommodate
 // relative includes, you can use the script/inline-c-relative-includes.go
 // program to generate a stand-alone C file.
 #include "../../../release/c/wuffs-unsupported-snapshot.h"
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index adce8ebd47bceb54359eb6f6c1397aca3e707fc2..412c66fa5914a5a5147854fa2118ae87e93e9625 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -40,7 +40,7 @@ It should print "PASS", amongst other information, and exit(0).
 // compiling it.
 #define WUFFS_IMPLEMENTATION
 
-// If building this program in an environment that doesn't easily accomodate
+// If building this program in an environment that doesn't easily accommodate
 // relative includes, you can use the script/inline-c-relative-includes.go
 // program to generate a stand-alone C file.
 #include "../../../release/c/wuffs-unsupported-snapshot.h"
