diff --git a/cmd/wuffs-c/internal/cgen/base/image-public.h b/cmd/wuffs-c/internal/cgen/base/image-public.h
index 699c3274e254605b767bb6e4bc6f5efe9a887475..8fb41ae4c6b6623ef7af3f3ea4258a58613e38be 100644
--- a/cmd/wuffs-c/internal/cgen/base/image-public.h
+++ b/cmd/wuffs-c/internal/cgen/base/image-public.h
@@ -445,6 +445,7 @@ typedef struct {
   // Do not access the private_impl's fields directly. There is no API/ABI
   // compatibility or safety guarantee if you do so.
   struct {
+    wuffs_base__range_ii_u64 work_buffer_size;
     uint32_t num_loops;
     bool first_frame_is_opaque;
   } private_impl;
@@ -454,10 +455,13 @@ typedef struct {
                          wuffs_base__pixel_subsampling pixsub,
                          uint32_t width,
                          uint32_t height,
+                         uint64_t work_buffer_size0,
+                         uint64_t work_buffer_size1,
                          uint32_t num_loops,
                          bool first_frame_is_opaque);
   inline void invalidate();
   inline bool is_valid();
+  inline wuffs_base__range_ii_u64 work_buffer_size();
   inline uint32_t num_loops();
   inline uint32_t first_frame_is_opaque();
 #endif  // __cplusplus
@@ -471,6 +475,8 @@ wuffs_base__image_config__initialize(wuffs_base__image_config* c,
                                      wuffs_base__pixel_subsampling pixsub,
                                      uint32_t width,
                                      uint32_t height,
+                                     uint64_t work_buffer_size0,
+                                     uint64_t work_buffer_size1,
                                      uint32_t num_loops,
                                      bool first_frame_is_opaque) {
   if (!c) {
@@ -481,6 +487,8 @@ wuffs_base__image_config__initialize(wuffs_base__image_config* c,
     c->pixcfg.private_impl.pixsub = pixsub;
     c->pixcfg.private_impl.width = width;
     c->pixcfg.private_impl.height = height;
+    c->private_impl.work_buffer_size.min_incl = work_buffer_size0;
+    c->private_impl.work_buffer_size.max_incl = work_buffer_size1;
     c->private_impl.num_loops = num_loops;
     c->private_impl.first_frame_is_opaque = first_frame_is_opaque;
     return;
@@ -500,6 +508,11 @@ wuffs_base__image_config__is_valid(wuffs_base__image_config* c) {
   return c && wuffs_base__pixel_config__is_valid(&(c->pixcfg));
 }
 
+static inline wuffs_base__range_ii_u64  //
+wuffs_base__image_config__work_buffer_size(wuffs_base__image_config* c) {
+  return c ? c->private_impl.work_buffer_size : ((wuffs_base__range_ii_u64){});
+}
+
 static inline uint32_t  //
 wuffs_base__image_config__num_loops(wuffs_base__image_config* c) {
   return c ? c->private_impl.num_loops : 0;
@@ -517,9 +530,12 @@ wuffs_base__image_config::initialize(wuffs_base__pixel_format pixfmt,
                                      wuffs_base__pixel_subsampling pixsub,
                                      uint32_t width,
                                      uint32_t height,
+                                     uint64_t work_buffer_size0,
+                                     uint64_t work_buffer_size1,
                                      uint32_t num_loops,
                                      bool first_frame_is_opaque) {
   wuffs_base__image_config__initialize(this, pixfmt, pixsub, width, height,
+                                       work_buffer_size0, work_buffer_size1,
                                        num_loops, first_frame_is_opaque);
 }
 
@@ -533,6 +549,11 @@ wuffs_base__image_config::is_valid() {
   return wuffs_base__image_config__is_valid(this);
 }
 
+inline wuffs_base__range_ii_u64  //
+wuffs_base__image_config::work_buffer_size() {
+  return wuffs_base__image_config__work_buffer_size(this);
+}
+
 inline uint32_t  //
 wuffs_base__image_config::num_loops() {
   return wuffs_base__image_config__num_loops(this);
diff --git a/release/c/wuffs-unsupported-snapshot.h b/release/c/wuffs-unsupported-snapshot.h
index 152731d056ea030d0434a18449e6f705028e5e6b..6ee992306848ad04082f02011eefef7a16140f8f 100644
--- a/release/c/wuffs-unsupported-snapshot.h
+++ b/release/c/wuffs-unsupported-snapshot.h
@@ -141,6 +141,7 @@ extern const char* wuffs_base__error__bad_call_sequence;
 extern const char* wuffs_base__error__bad_receiver;
 extern const char* wuffs_base__error__bad_restart;
 extern const char* wuffs_base__error__bad_sizeof_receiver;
+extern const char* wuffs_base__error__bad_work_buffer_length;
 extern const char* wuffs_base__error__bad_wuffs_version;
 extern const char* wuffs_base__error__cannot_return_a_suspension;
 extern const char* wuffs_base__error__check_wuffs_version_not_applicable;
@@ -1550,6 +1551,7 @@ typedef struct {
   // Do not access the private_impl's fields directly. There is no API/ABI
   // compatibility or safety guarantee if you do so.
   struct {
+    wuffs_base__range_ii_u64 work_buffer_size;
     uint32_t num_loops;
     bool first_frame_is_opaque;
   } private_impl;
@@ -1559,10 +1561,13 @@ typedef struct {
                          wuffs_base__pixel_subsampling pixsub,
                          uint32_t width,
                          uint32_t height,
+                         uint64_t work_buffer_size0,
+                         uint64_t work_buffer_size1,
                          uint32_t num_loops,
                          bool first_frame_is_opaque);
   inline void invalidate();
   inline bool is_valid();
+  inline wuffs_base__range_ii_u64 work_buffer_size();
   inline uint32_t num_loops();
   inline uint32_t first_frame_is_opaque();
 #endif  // __cplusplus
@@ -1576,6 +1581,8 @@ wuffs_base__image_config__initialize(wuffs_base__image_config* c,
                                      wuffs_base__pixel_subsampling pixsub,
                                      uint32_t width,
                                      uint32_t height,
+                                     uint64_t work_buffer_size0,
+                                     uint64_t work_buffer_size1,
                                      uint32_t num_loops,
                                      bool first_frame_is_opaque) {
   if (!c) {
@@ -1586,6 +1593,8 @@ wuffs_base__image_config__initialize(wuffs_base__image_config* c,
     c->pixcfg.private_impl.pixsub = pixsub;
     c->pixcfg.private_impl.width = width;
     c->pixcfg.private_impl.height = height;
+    c->private_impl.work_buffer_size.min_incl = work_buffer_size0;
+    c->private_impl.work_buffer_size.max_incl = work_buffer_size1;
     c->private_impl.num_loops = num_loops;
     c->private_impl.first_frame_is_opaque = first_frame_is_opaque;
     return;
@@ -1605,6 +1614,11 @@ wuffs_base__image_config__is_valid(wuffs_base__image_config* c) {
   return c && wuffs_base__pixel_config__is_valid(&(c->pixcfg));
 }
 
+static inline wuffs_base__range_ii_u64  //
+wuffs_base__image_config__work_buffer_size(wuffs_base__image_config* c) {
+  return c ? c->private_impl.work_buffer_size : ((wuffs_base__range_ii_u64){});
+}
+
 static inline uint32_t  //
 wuffs_base__image_config__num_loops(wuffs_base__image_config* c) {
   return c ? c->private_impl.num_loops : 0;
@@ -1622,9 +1636,12 @@ wuffs_base__image_config::initialize(wuffs_base__pixel_format pixfmt,
                                      wuffs_base__pixel_subsampling pixsub,
                                      uint32_t width,
                                      uint32_t height,
+                                     uint64_t work_buffer_size0,
+                                     uint64_t work_buffer_size1,
                                      uint32_t num_loops,
                                      bool first_frame_is_opaque) {
   wuffs_base__image_config__initialize(this, pixfmt, pixsub, width, height,
+                                       work_buffer_size0, work_buffer_size1,
                                        num_loops, first_frame_is_opaque);
 }
 
@@ -1638,6 +1655,11 @@ wuffs_base__image_config::is_valid() {
   return wuffs_base__image_config__is_valid(this);
 }
 
+inline wuffs_base__range_ii_u64  //
+wuffs_base__image_config::work_buffer_size() {
+  return wuffs_base__image_config__work_buffer_size(this);
+}
+
 inline uint32_t  //
 wuffs_base__image_config::num_loops() {
   return wuffs_base__image_config__num_loops(this);
@@ -3616,6 +3638,8 @@ const char* wuffs_base__error__bad_receiver = "?base: bad receiver";
 const char* wuffs_base__error__bad_restart = "?base: bad restart";
 const char* wuffs_base__error__bad_sizeof_receiver =
     "?base: bad sizeof receiver";
+const char* wuffs_base__error__bad_work_buffer_length =
+    "?base: bad work buffer length";
 const char* wuffs_base__error__bad_wuffs_version = "?base: bad wuffs version";
 const char* wuffs_base__error__cannot_return_a_suspension =
     "?base: cannot return a suspension";
@@ -6401,7 +6425,8 @@ wuffs_gif__decoder__decode_image_config(wuffs_gif__decoder* self,
     if (a_dst != NULL) {
       wuffs_base__image_config__initialize(
           a_dst, 570984584, 0, self->private_impl.f_width,
-          self->private_impl.f_height, v_num_loops, v_ffio);
+          self->private_impl.f_height, ((uint64_t)(self->private_impl.f_width)),
+          ((uint64_t)(self->private_impl.f_width)), v_num_loops, v_ffio);
     }
     self->private_impl.f_call_sequence = 1;
 
@@ -6729,6 +6754,11 @@ wuffs_gif__decoder__decode_frame(wuffs_gif__decoder* self,
   switch (coro_susp_point) {
     WUFFS_BASE__COROUTINE_SUSPENSION_POINT_0;
 
+    if (((uint64_t)(a_work_buffer.len)) !=
+        ((uint64_t)(self->private_impl.f_width))) {
+      status = wuffs_base__error__bad_work_buffer_length;
+      goto exit;
+    }
     if (self->private_impl.f_call_sequence != 2) {
       WUFFS_BASE__COROUTINE_SUSPENSION_POINT(1);
       status = wuffs_gif__decoder__decode_frame_config(self, NULL, a_src);
diff --git a/std/gif/decode_gif.wuffs b/std/gif/decode_gif.wuffs
index 0892b952a6ced9c3b30d329cb4d2b547c8bcdee7..0b9ae34f4828733c285df8418a3839640557948a 100644
--- a/std/gif/decode_gif.wuffs
+++ b/std/gif/decode_gif.wuffs
@@ -148,6 +148,8 @@ pub func decoder.decode_image_config!??(dst nptr base.image_config, src base.io_
 			pixsub:0,
 			width:this.width,
 			height:this.height,
+			work_buffer_size0:this.width as base.u64,
+			work_buffer_size1:this.width as base.u64,
 			num_loops:num_loops,
 			first_frame_is_opaque:ffio)
 	}
@@ -234,6 +236,9 @@ pri func decoder.skip_frame!??(src base.io_reader) {
 
 // TODO: honor pixbuf_origin_x and pixbuf_origin_y args.
 pub func decoder.decode_frame!??(dst ptr base.pixel_buffer, pixbuf_origin_x base.u32, pixbuf_origin_y base.u32, src base.io_reader, work_buffer slice base.u8) {
+	if args.work_buffer.length() != (this.width as base.u64) {
+		return status "?bad work buffer length"
+	}
 	if this.call_sequence != 2 {
 		this.decode_frame_config!??(dst:nullptr, src:args.src)
 	}
