diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 857fb1a358efb127795ee5f1910ab4013b945b05..cd9ccd694e58173501801047537c100053958cc1 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -109,11 +109,12 @@ const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
 
     bool seen_ok = false;
     while (true) {
-      z = wuffs_gif__decoder__decode_frame(&dec, &pb, 0, 0, src_reader,
+      z = wuffs_gif__decoder__decode_frame(&dec, &pb, src_reader,
                                            ((wuffs_base__slice_u8){
                                                .ptr = workbuf,
                                                .len = workbuf_size,
-                                           }));
+                                           }),
+                                           NULL);
       if (z) {
         if ((z != wuffs_base__warning__end_of_data) || !seen_ok) {
           ret = z;
