diff --git a/release/c/wuffs-unsupported-snapshot.h b/release/c/wuffs-unsupported-snapshot.h
index 6d1a2b04bfa2a552cc30d32775168aae10cf7a6c..ce59de62631484282e5eacb79c3f20f841b742f4 100644
--- a/release/c/wuffs-unsupported-snapshot.h
+++ b/release/c/wuffs-unsupported-snapshot.h
@@ -7511,8 +7511,9 @@ wuffs_gif__decoder__workbuf_len(const wuffs_gif__decoder* self) {
     return ((wuffs_base__range_ii_u64){});
   }
 
-  return wuffs_base__utility__make_range_ii_u64(&self->private_impl.f_util, 0,
-                                                0);
+  return wuffs_base__utility__make_range_ii_u64(
+      &self->private_impl.f_util, ((uint64_t)(self->private_impl.f_width)),
+      ((uint64_t)(self->private_impl.f_width)));
 }
 
 // -------- func gif.decoder.restart_frame
@@ -7776,7 +7777,7 @@ wuffs_gif__decoder__decode_frame(wuffs_gif__decoder* self,
   switch (coro_susp_point) {
     WUFFS_BASE__COROUTINE_SUSPENSION_POINT_0;
 
-    if (((uint64_t)(a_workbuf.len)) !=
+    if (((uint64_t)(a_workbuf.len)) <
         ((uint64_t)(self->private_impl.f_width))) {
       status = wuffs_base__error__bad_workbuf_length;
       goto exit;
diff --git a/std/gif/decode_gif.wuffs b/std/gif/decode_gif.wuffs
index 7cac24abe53c8c61a2734de309a6fccd8e16cc73..8e6678305277a41e690d8469478d291e558bca49 100644
--- a/std/gif/decode_gif.wuffs
+++ b/std/gif/decode_gif.wuffs
@@ -146,6 +146,7 @@ pub func decoder.decode_image_config!??(dst nptr base.image_config, src base.io_
 			pixsub:0,
 			width:this.width,
 			height:this.height,
+			// TODO: workbuf_len should be an aspect of the decoder, not the image_config.
 			workbuf_len0:this.width as base.u64,
 			workbuf_len1:this.width as base.u64,
 			first_frame_io_position:this.frame_config_io_position,
@@ -179,7 +180,9 @@ pub func decoder.frame_dirty_rect() base.rect_ie_u32 {
 }
 
 pub func decoder.workbuf_len() base.range_ii_u64 {
-	return this.util.make_range_ii_u64(min_incl:0, max_incl:0)
+	return this.util.make_range_ii_u64(
+		min_incl:this.width as base.u64,
+		max_incl:this.width as base.u64)
 }
 
 pub func decoder.restart_frame!??(index base.u64, io_position base.u64) {
@@ -251,7 +254,7 @@ pri func decoder.skip_frame!??(src base.io_reader) {
 
 // TODO: honor args.opts.
 pub func decoder.decode_frame!??(dst ptr base.pixel_buffer, src base.io_reader, workbuf slice base.u8, opts nptr base.decode_frame_options) {
-	if args.workbuf.length() != (this.width as base.u64) {
+	if args.workbuf.length() < (this.width as base.u64) {
 		return status "?bad workbuf length"
 	}
 	if this.call_sequence != 2 {
