diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index 0f25df03960ad334df26b567f11d7c5ea23c1c60..f5a4ba2392c106b6f30242b0963ff3b773a92e2e 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -48,7 +48,7 @@ It should print "PASS", amongst other information, and exit(0).
 #include "../../../release/c/wuffs-unsupported-snapshot.c"
 #include "../fuzzlib/fuzzlib.c"
 
-#define WORK_BUFFER_SIZE (32768 + 512)
+#define WORK_BUFFER_SIZE WUFFS_ZLIB__MAX_INCL_WORKBUF_LEN
 uint8_t work_buffer[WORK_BUFFER_SIZE];
 
 const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
