diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index ea792585f16f2a79583f7b3479e752c38d05f2e9..c6c0f66c0e54b425efafe807e9185d464fbf39bc 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -185,6 +185,7 @@ extern const char* wuffs_base__error__disabled_by_previous_error;
 extern const char* wuffs_base__error__initialize_falsely_claimed_already_zeroed;
 extern const char* wuffs_base__error__initialize_not_called;
 extern const char* wuffs_base__error__interleaved_coroutine_calls;
+extern const char* wuffs_base__error__not_enough_data;
 
 static inline bool  //
 wuffs_base__status__is_complete(wuffs_base__status z) {
@@ -4806,6 +4807,7 @@ const char* wuffs_base__error__initialize_not_called =
     "?base: initialize not called";
 const char* wuffs_base__error__interleaved_coroutine_calls =
     "?base: interleaved coroutine calls";
+const char* wuffs_base__error__not_enough_data = "?base: not enough data";
 
 // ---------------- Images
 
@@ -10686,6 +10688,14 @@ wuffs_gif__decoder__decode_id_part2(wuffs_gif__decoder* self,
   label_0_break:;
     self->private_impl.f_compressed_ri = 0;
     self->private_impl.f_compressed_wi = 0;
+    if ((self->private_impl.f_dst_y < self->private_impl.f_frame_rect_y1) &&
+        (self->private_impl.f_frame_rect_x0 !=
+         self->private_impl.f_frame_rect_x1) &&
+        (self->private_impl.f_frame_rect_y0 !=
+         self->private_impl.f_frame_rect_y1)) {
+      status = wuffs_base__error__not_enough_data;
+      goto exit;
+    }
 
     goto ok;
   ok:
diff --git a/std/gif/decode_gif.wuffs b/std/gif/decode_gif.wuffs
index b9dc0475c8bb9dd6ec3f982f68dbe093fcabd2ec..dbeb1e6d23dad3ef44de63fd081d38c0cb50f1fa 100644
--- a/std/gif/decode_gif.wuffs
+++ b/std/gif/decode_gif.wuffs
@@ -763,7 +763,11 @@ pri func decoder.decode_id_part2?(dst ptr base.pixel_buffer, src base.io_reader,
 	this.compressed_ri = 0
 	this.compressed_wi = 0
 
-	// TODO: check for "not enough pixel data".
+	if (this.dst_y < this.frame_rect_y1) and
+		(this.frame_rect_x0 != this.frame_rect_x1) and
+		(this.frame_rect_y0 != this.frame_rect_y1) {
+		return base."?not enough data"
+	}
 }
 
 pri func decoder.copy_to_image_buffer!(pb ptr base.pixel_buffer, src slice base.u8) base.status {
