diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 2858c1c6fca4e867277e09dc4f23b10be5abcd6c..5e083fa9887b3dc02b5fc5b1ebfd11e03a8d290c 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -57,19 +57,19 @@ const char* fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
   // variable initialization" warnings.
   {
     wuffs_gif__decoder dec;
-    const char* status = wuffs_gif__decoder__initialize(
+    wuffs_base__status status = wuffs_gif__decoder__initialize(
         &dec, sizeof dec, WUFFS_VERSION,
         (hash & 1) ? WUFFS_INITIALIZE__LEAVE_INTERNAL_BUFFERS_UNINITIALIZED
                    : 0);
-    if (status) {
-      ret = status;
+    if (!wuffs_base__status__is_ok(&status)) {
+      ret = wuffs_base__status__message(&status);
       goto exit;
     }
 
     wuffs_base__image_config ic = ((wuffs_base__image_config){});
     status = wuffs_gif__decoder__decode_image_config(&dec, &ic, src);
-    if (status) {
-      ret = status;
+    if (!wuffs_base__status__is_ok(&status)) {
+      ret = wuffs_base__status__message(&status);
       goto exit;
     }
     if (!wuffs_base__image_config__is_valid(&ic)) {
@@ -107,8 +107,8 @@ const char* fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
 
     wuffs_base__pixel_buffer pb = ((wuffs_base__pixel_buffer){});
     status = wuffs_base__pixel_buffer__set_from_slice(&pb, &ic.pixcfg, pixbuf);
-    if (status) {
-      ret = status;
+    if (!wuffs_base__status__is_ok(&status)) {
+      ret = wuffs_base__status__message(&status);
       goto exit;
     }
 
@@ -116,9 +116,9 @@ const char* fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
     while (true) {
       wuffs_base__frame_config fc = ((wuffs_base__frame_config){});
       status = wuffs_gif__decoder__decode_frame_config(&dec, &fc, src);
-      if (status) {
-        if ((status != wuffs_base__warning__end_of_data) || !seen_ok) {
-          ret = status;
+      if (!wuffs_base__status__is_ok(&status)) {
+        if ((status.repr != wuffs_base__warning__end_of_data) || !seen_ok) {
+          ret = wuffs_base__status__message(&status);
         }
         goto exit;
       }
@@ -134,9 +134,9 @@ const char* fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
         goto exit;
       }
 
-      if (status) {
-        if ((status != wuffs_base__warning__end_of_data) || !seen_ok) {
-          ret = status;
+      if (!wuffs_base__status__is_ok(&status)) {
+        if ((status.repr != wuffs_base__warning__end_of_data) || !seen_ok) {
+          ret = wuffs_base__status__message(&status);
         }
         goto exit;
       }
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index b4525826e4b72f1e86392e8fa0bf395e90946ddb..80a75a2b42f36d78f51b2e832afbe5be73840bc4 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -62,11 +62,11 @@ uint8_t work_buffer[1];
 
 const char* fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
   wuffs_zlib__decoder dec;
-  const char* status = wuffs_zlib__decoder__initialize(
+  wuffs_base__status status = wuffs_zlib__decoder__initialize(
       &dec, sizeof dec, WUFFS_VERSION,
       (hash & 1) ? WUFFS_INITIALIZE__LEAVE_INTERNAL_BUFFERS_UNINITIALIZED : 0);
-  if (status) {
-    return status;
+  if (!wuffs_base__status__is_ok(&status)) {
+    return wuffs_base__status__message(&status);
   }
 
   // Ignore the checksum for 99.99%-ish of all input. When fuzzers generate
@@ -89,7 +89,7 @@ const char* fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
                                                        .ptr = work_buffer,
                                                        .len = WORK_BUFFER_SIZE,
                                                    }));
-    if (status != wuffs_base__suspension__short_write) {
+    if (status.repr != wuffs_base__suspension__short_write) {
       break;
     }
     if (dst.meta.wi == 0) {
@@ -98,5 +98,5 @@ const char* fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
       intentional_segfault();
     }
   }
-  return status;
+  return wuffs_base__status__message(&status);
 }
