diff --git a/fuzz/c/fuzzlib/fuzzlib.c b/fuzz/c/fuzzlib/fuzzlib.c
index 74e6baa78d18c67a2ebc9411a89c8320f8e906e6..0ce866c3398ccf81c47ac7df36d2621848062756 100644
--- a/fuzz/c/fuzzlib/fuzzlib.c
+++ b/fuzz/c/fuzzlib/fuzzlib.c
@@ -23,13 +23,16 @@
 
 volatile int* intentional_segfault_ptr = NULL;
 
-void intentional_segfault() {
+void  //
+intentional_segfault() {
   *intentional_segfault_ptr = 0;
 }
 
-const char* fuzz(wuffs_base__io_buffer* src, uint32_t hash);
+const char*  //
+fuzz(wuffs_base__io_buffer* src, uint32_t hash);
 
-static const char* llvmFuzzerTestOneInput(const uint8_t* data, size_t size) {
+static const char*  //
+llvmFuzzerTestOneInput(const uint8_t* data, size_t size) {
   // Hash input as per https://en.wikipedia.org/wiki/Jenkins_hash_function
   size_t i = 0;
   uint32_t hash = 0;
@@ -72,7 +75,8 @@ static const char* llvmFuzzerTestOneInput(const uint8_t* data, size_t size) {
 extern "C" {
 #endif
 
-int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+int  //
+LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
   llvmFuzzerTestOneInput(data, size);
   return 0;
 }
@@ -98,9 +102,11 @@ static struct {
   size_t len;
 } relative_cwd;
 
-static int visit(char* filename);
+static int  //
+visit(char* filename);
 
-static int visit_dir(int fd) {
+static int  //
+visit_dir(int fd) {
   int cwd_fd = open(".", O_RDONLY, 0);
   if (fchdir(fd)) {
     printf("failed\n");
@@ -145,7 +151,8 @@ static int visit_dir(int fd) {
   return 0;
 }
 
-static int visit_reg(int fd, off_t size) {
+static int  //
+visit_reg(int fd, off_t size) {
   if ((size < 0) || (0x7FFFFFFF < size)) {
     printf("failed\n");
     fprintf(stderr, "FAIL: file size out of bounds");
@@ -179,7 +186,8 @@ static int visit_reg(int fd, off_t size) {
   return 0;
 }
 
-static int visit(char* filename) {
+static int  //
+visit(char* filename) {
   num_files_processed++;
   if (!filename || (filename[0] == '\x00')) {
     fprintf(stderr, "FAIL: invalid filename\n");
@@ -235,7 +243,8 @@ static int visit(char* filename) {
   return v;
 }
 
-int main(int argc, char** argv) {
+int  //
+main(int argc, char** argv) {
   num_files_processed = 0;
   relative_cwd.len = 0;
 
diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index d9460c69748821022e83715342a07305df838800..af67e920e549e36d27bff786c571f556ea645aee 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -48,7 +48,8 @@ It should print "PASS", amongst other information, and exit(0).
 #include "../../../release/c/wuffs-unsupported-snapshot.c"
 #include "../fuzzlib/fuzzlib.c"
 
-const char* fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
+const char* //
+fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
   const char* ret = NULL;
   wuffs_base__slice_u8 pixbuf = ((wuffs_base__slice_u8){});
   wuffs_base__slice_u8 workbuf = ((wuffs_base__slice_u8){});
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index 785aa14f146106df9d4c3b69567263f63577915c..ff6735013c691d4d091be6b0a654bda2652ea9c4 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -60,7 +60,8 @@ uint8_t work_buffer[WORK_BUFFER_SIZE];
 uint8_t work_buffer[1];
 #endif
 
-const char* fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
+const char*  //
+fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
   wuffs_zlib__decoder dec;
   wuffs_base__status status = wuffs_zlib__decoder__initialize(
       &dec, sizeof dec, WUFFS_VERSION,
