diff --git a/fuzz/c/std/json_fuzzer.c b/fuzz/c/std/json_fuzzer.c
index 29942115d0cce3602a211cf66640e6b26967f6be..e8b99b0a2c3061f6234c35dafed6070908d606aa 100644
--- a/fuzz/c/std/json_fuzzer.c
+++ b/fuzz/c/std/json_fuzzer.c
@@ -59,7 +59,7 @@ It should print "PASS", amongst other information, and exit(0).
 #include "../../../release/c/wuffs-unsupported-snapshot.c"
 #include "../fuzzlib/fuzzlib.c"
 
-#define TOK_BUFFER_SIZE 4096
+#define TOK_BUFFER_ARRAY_SIZE 4096
 #define STACK_SIZE (WUFFS_JSON__DECODER_DEPTH_MAX_INCL + 1)
 
 // Each stack element is 1 byte. The low 7 bits denote the container:
@@ -225,11 +225,12 @@ fuzz_complex(wuffs_base__io_buffer* full_src, uint32_t hash_24_bits) {
     return wuffs_base__status__message(&status);
   }
 
-  wuffs_base__token tok_array[TOK_BUFFER_SIZE];
+  wuffs_base__token tok_array[TOK_BUFFER_ARRAY_SIZE];
   wuffs_base__token_buffer tok = ((wuffs_base__token_buffer){
       .data = ((wuffs_base__slice_token){
           .ptr = tok_array,
-          .len = (tok_limit < TOK_BUFFER_SIZE) ? tok_limit : TOK_BUFFER_SIZE,
+          .len = (tok_limit < TOK_BUFFER_ARRAY_SIZE) ? tok_limit
+                                                     : TOK_BUFFER_ARRAY_SIZE,
       }),
   });
 
@@ -331,11 +332,11 @@ fuzz_simple(wuffs_base__io_buffer* full_src) {
     return wuffs_base__status__message(&status);
   }
 
-  wuffs_base__token tok_array[TOK_BUFFER_SIZE];
+  wuffs_base__token tok_array[TOK_BUFFER_ARRAY_SIZE];
   wuffs_base__token_buffer tok = ((wuffs_base__token_buffer){
       .data = ((wuffs_base__slice_token){
           .ptr = tok_array,
-          .len = TOK_BUFFER_SIZE,
+          .len = TOK_BUFFER_ARRAY_SIZE,
       }),
   });
 
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index f665a66daac2da7c19cba5ce608de71973b78f43..4835c6e4575705101d791c79a435ff4c9372f236 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -61,16 +61,17 @@ It should print "PASS", amongst other information, and exit(0).
 #include "../../../release/c/wuffs-unsupported-snapshot.c"
 #include "../fuzzlib/fuzzlib.c"
 
-#define DST_BUFFER_SIZE 65536
+#define DST_BUFFER_ARRAY_SIZE 65536
 
 // Wuffs allows either statically or dynamically allocated work buffers. This
 // program exercises static allocation.
-#define WORK_BUFFER_SIZE WUFFS_ZLIB__DECODER_WORKBUF_LEN_MAX_INCL_WORST_CASE
-#if WORK_BUFFER_SIZE > 0
-uint8_t work_buffer[WORK_BUFFER_SIZE];
+#define WORK_BUFFER_ARRAY_SIZE \
+  WUFFS_ZLIB__DECODER_WORKBUF_LEN_MAX_INCL_WORST_CASE
+#if WORK_BUFFER_ARRAY_SIZE > 0
+uint8_t work_buffer_array[WORK_BUFFER_ARRAY_SIZE];
 #else
 // Not all C/C++ compilers support 0-length arrays.
-uint8_t work_buffer[1];
+uint8_t work_buffer_array[1];
 #endif
 
 const char*  //
@@ -88,21 +89,22 @@ fuzz(wuffs_base__io_buffer* src, uint32_t hash) {
   // verify that checksumming does not lead to e.g. buffer overflows.
   wuffs_zlib__decoder__set_ignore_checksum(&dec, hash & 0xFFFE);
 
-  uint8_t dst_buffer[DST_BUFFER_SIZE];
+  uint8_t dst_buffer[DST_BUFFER_ARRAY_SIZE];
   wuffs_base__io_buffer dst = ((wuffs_base__io_buffer){
       .data = ((wuffs_base__slice_u8){
           .ptr = dst_buffer,
-          .len = DST_BUFFER_SIZE,
+          .len = DST_BUFFER_ARRAY_SIZE,
       }),
   });
 
   while (true) {
     dst.meta.wi = 0;
-    status = wuffs_zlib__decoder__transform_io(&dec, &dst, src,
-                                               ((wuffs_base__slice_u8){
-                                                   .ptr = work_buffer,
-                                                   .len = WORK_BUFFER_SIZE,
-                                               }));
+    status =
+        wuffs_zlib__decoder__transform_io(&dec, &dst, src,
+                                          ((wuffs_base__slice_u8){
+                                              .ptr = work_buffer_array,
+                                              .len = WORK_BUFFER_ARRAY_SIZE,
+                                          }));
     if (status.repr != wuffs_base__suspension__short_write) {
       break;
     }
