diff --git a/fuzz/c/std/json_fuzzer.c b/fuzz/c/std/json_fuzzer.c
index c1ae095faa3d1c4e7666813f478a22878462ac26..ae377be8c63077374fda97ab55bd1381e0ac41b7 100644
--- a/fuzz/c/std/json_fuzzer.c
+++ b/fuzz/c/std/json_fuzzer.c
@@ -234,7 +234,7 @@ void set_quirks(wuffs_json__decoder* dec, uint32_t hash_12_bits) {
       WUFFS_JSON__QUIRK_ALLOW_LEADING_ASCII_RECORD_SEPARATOR,
       WUFFS_JSON__QUIRK_ALLOW_LEADING_UNICODE_BYTE_ORDER_MARK,
       WUFFS_JSON__QUIRK_ALLOW_TRAILING_NEW_LINE,
-      WUFFS_JSON__QUIRK_REPLACE_INVALID_UTF_8,
+      WUFFS_JSON__QUIRK_REPLACE_INVALID_UNICODE,
       0,
   };
 
