diff --git a/fuzz/c/std/json_fuzzer.c b/fuzz/c/std/json_fuzzer.c
index 3e76676ef97c48b74d9f45c70962ea3e00ba89ee..9ccbbe19296dc149c4a185157eff0f165f7a408c 100644
--- a/fuzz/c/std/json_fuzzer.c
+++ b/fuzz/c/std/json_fuzzer.c
@@ -107,7 +107,7 @@ fuzz_one_token(wuffs_base__token t,
     return "fuzz: internal error: inconsistent link bits";
   }
 
-  uint64_t vbc = wuffs_base__token__value_base_category(&t);
+  int64_t vbc = wuffs_base__token__value_base_category(&t);
   uint64_t vbd = wuffs_base__token__value_base_detail(&t);
 
   switch (vbc) {
