diff --git a/fuzz/c/std/json_fuzzer.c b/fuzz/c/std/json_fuzzer.c
index c82946b58a00501ceebcaa9ab1a1212a6d540bbc..0a83aa5870bd161e8c512afa4e2efde3f528ea8d 100644
--- a/fuzz/c/std/json_fuzzer.c
+++ b/fuzz/c/std/json_fuzzer.c
@@ -240,6 +240,7 @@ void set_quirks(wuffs_json__decoder* dec, uint64_t hash_44_bits) {
       WUFFS_JSON__QUIRK_ALLOW_INF_NAN_NUMBERS,
       WUFFS_JSON__QUIRK_ALLOW_LEADING_ASCII_RECORD_SEPARATOR,
       WUFFS_JSON__QUIRK_ALLOW_LEADING_UNICODE_BYTE_ORDER_MARK,
+      WUFFS_JSON__QUIRK_ALLOW_TRAILING_COMMENT,
       WUFFS_JSON__QUIRK_ALLOW_TRAILING_NEW_LINE,
       WUFFS_JSON__QUIRK_REPLACE_INVALID_UNICODE,
       0,
