diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index 3f4ad79e35a61834b12c1d1dd5db8a80948bd8c0..855abdac11a645109e823fbf81bdc59d0b5ebd00 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -7661,8 +7661,12 @@ extern const char wuffs_json__error__unsupported_recursion_depth[];
 
 #define WUFFS_JSON__QUIRK_ALLOW_LEADING_UNICODE_BYTE_ORDER_MARK 1225364496
 
+#define WUFFS_JSON__QUIRK_ALLOW_TRAILING_FILLER 1225364497
+
 #define WUFFS_JSON__QUIRK_ALLOW_TRAILING_COMMENT 1225364497
 
+#define WUFFS_JSON__QUIRK_EXPECT_TRAILING_NEW_LINE_OR_EOF 1225364498
+
 #define WUFFS_JSON__QUIRK_ALLOW_TRAILING_NEW_LINE 1225364498
 
 #define WUFFS_JSON__QUIRK_JSON_POINTER_ALLOW_TILDE_R_TILDE_N 1225364499
@@ -7760,15 +7764,15 @@ struct wuffs_json__decoder__struct {
     bool f_quirks[21];
     bool f_allow_leading_ars;
     bool f_allow_leading_ubom;
-    bool f_allow_trailing_comment;
     bool f_end_of_data;
+    uint8_t f_trailer_stop;
     uint8_t f_comment_type;
 
     uint32_t p_decode_tokens[1];
     uint32_t p_decode_leading[1];
     uint32_t p_decode_comment[1];
     uint32_t p_decode_inf_nan[1];
-    uint32_t p_decode_trailing_new_line[1];
+    uint32_t p_decode_trailer[1];
   } private_impl;
 
   struct {
@@ -25737,7 +25741,7 @@ wuffs_json__decoder__decode_inf_nan(
     wuffs_base__io_buffer* a_src);
 
 static wuffs_base__status
-wuffs_json__decoder__decode_trailing_new_line(
+wuffs_json__decoder__decode_trailer(
     wuffs_json__decoder* self,
     wuffs_base__token_buffer* a_dst,
     wuffs_base__io_buffer* a_src);
@@ -25960,6 +25964,12 @@ wuffs_json__decoder__decode_tokens(
       status = wuffs_base__make_status(wuffs_base__note__end_of_data);
       goto ok;
     }
+    if (self->private_impl.f_quirks[18]) {
+      if (self->private_impl.f_quirks[11] || self->private_impl.f_quirks[12] || self->private_impl.f_quirks[17]) {
+        status = wuffs_base__make_status(wuffs_json__error__bad_quirk_combination);
+        goto exit;
+      }
+    }
     if (self->private_impl.f_quirks[15] || self->private_impl.f_quirks[16]) {
       if (a_dst) {
         a_dst->meta.wi = ((size_t)(iop_a_dst - a_dst->data.ptr));
@@ -26832,7 +26842,9 @@ wuffs_json__decoder__decode_tokens(
             if (status.repr) {
               goto suspend;
             }
-            goto label__outer__continue;
+            if (self->private_impl.f_comment_type > 0) {
+              goto label__outer__continue;
+            }
           }
         }
         status = wuffs_base__make_status(wuffs_json__error__bad_input);
@@ -26845,7 +26857,7 @@ wuffs_json__decoder__decode_tokens(
       v_expect = v_expect_after_value;
     }
     label__outer__break:;
-    if (self->private_impl.f_quirks[18]) {
+    if (self->private_impl.f_quirks[17] || self->private_impl.f_quirks[18]) {
       if (a_dst) {
         a_dst->meta.wi = ((size_t)(iop_a_dst - a_dst->data.ptr));
       }
@@ -26853,7 +26865,7 @@ wuffs_json__decoder__decode_tokens(
         a_src->meta.ri = ((size_t)(iop_a_src - a_src->data.ptr));
       }
       WUFFS_BASE__COROUTINE_SUSPENSION_POINT(24);
-      status = wuffs_json__decoder__decode_trailing_new_line(self, a_dst, a_src);
+      status = wuffs_json__decoder__decode_trailer(self, a_dst, a_src);
       if (a_dst) {
         iop_a_dst = a_dst->data.ptr + a_dst->meta.wi;
       }
@@ -27240,8 +27252,8 @@ wuffs_json__decoder__decode_comment(
         goto label__0__continue;
       }
       if (a_src && a_src->meta.closed) {
-        status = wuffs_base__make_status(wuffs_json__error__bad_input);
-        goto exit;
+        status = wuffs_base__make_status(NULL);
+        goto ok;
       }
       status = wuffs_base__make_status(wuffs_base__suspension__short_read);
       WUFFS_BASE__COROUTINE_SUSPENSION_POINT_MAYBE_SUSPEND(2);
@@ -27348,8 +27360,6 @@ wuffs_json__decoder__decode_comment(
         }
       }
     }
-    status = wuffs_base__make_status(wuffs_json__error__bad_input);
-    goto exit;
 
     goto ok;
     ok:
@@ -27535,10 +27545,10 @@ wuffs_json__decoder__decode_inf_nan(
   return status;
 }
 
-// -------- func json.decoder.decode_trailing_new_line
+// -------- func json.decoder.decode_trailer
 
 static wuffs_base__status
-wuffs_json__decoder__decode_trailing_new_line(
+wuffs_json__decoder__decode_trailer(
     wuffs_json__decoder* self,
     wuffs_base__token_buffer* a_dst,
     wuffs_base__io_buffer* a_src) {
@@ -27571,11 +27581,15 @@ wuffs_json__decoder__decode_trailing_new_line(
     io2_a_src = io0_a_src + a_src->meta.wi;
   }
 
-  uint32_t coro_susp_point = self->private_impl.p_decode_trailing_new_line[0];
+  uint32_t coro_susp_point = self->private_impl.p_decode_trailer[0];
   switch (coro_susp_point) {
     WUFFS_BASE__COROUTINE_SUSPENSION_POINT_0;
 
-    self->private_impl.f_allow_trailing_comment = self->private_impl.f_quirks[17];
+    if (self->private_impl.f_quirks[18]) {
+      self->private_impl.f_trailer_stop = 10;
+    } else {
+      self->private_impl.f_trailer_stop = 0;
+    }
     label__outer__continue:;
     while (true) {
       if (((uint64_t)(io2_a_dst - iop_a_dst)) <= 0) {
@@ -27608,44 +27622,44 @@ wuffs_json__decoder__decode_trailing_new_line(
                 (((uint64_t)(v_whitespace_length)) << WUFFS_BASE__TOKEN__LENGTH__SHIFT));
             v_whitespace_length = 0;
           }
-          if (self->private_impl.f_allow_trailing_comment) {
-            if (a_dst) {
-              a_dst->meta.wi = ((size_t)(iop_a_dst - a_dst->data.ptr));
-            }
-            if (a_src) {
-              a_src->meta.ri = ((size_t)(iop_a_src - a_src->data.ptr));
-            }
-            WUFFS_BASE__COROUTINE_SUSPENSION_POINT(3);
-            status = wuffs_json__decoder__decode_comment(self, a_dst, a_src);
-            if (a_dst) {
-              iop_a_dst = a_dst->data.ptr + a_dst->meta.wi;
-            }
-            if (a_src) {
-              iop_a_src = a_src->data.ptr + a_src->meta.ri;
-            }
-            if (status.repr) {
-              goto suspend;
-            }
-            v_c = 0;
-            v_whitespace_length = 0;
-            if (self->private_impl.f_comment_type == 1) {
-              self->private_impl.f_allow_trailing_comment = false;
-              goto label__outer__continue;
-            } else if (self->private_impl.f_comment_type == 2) {
-              goto label__outer__break;
-            }
+          if (self->private_impl.f_trailer_stop > 0) {
+            status = wuffs_base__make_status(wuffs_json__error__bad_input);
+            goto exit;
           }
-          status = wuffs_base__make_status(wuffs_json__error__bad_input);
-          goto exit;
+          if (a_dst) {
+            a_dst->meta.wi = ((size_t)(iop_a_dst - a_dst->data.ptr));
+          }
+          if (a_src) {
+            a_src->meta.ri = ((size_t)(iop_a_src - a_src->data.ptr));
+          }
+          WUFFS_BASE__COROUTINE_SUSPENSION_POINT(3);
+          status = wuffs_json__decoder__decode_comment(self, a_dst, a_src);
+          if (a_dst) {
+            iop_a_dst = a_dst->data.ptr + a_dst->meta.wi;
+          }
+          if (a_src) {
+            iop_a_src = a_src->data.ptr + a_src->meta.ri;
+          }
+          if (status.repr) {
+            goto suspend;
+          }
+          v_c = 0;
+          v_whitespace_length = 0;
+          if (self->private_impl.f_comment_type > 0) {
+            goto label__outer__continue;
+          }
+          status = wuffs_base__make_status(NULL);
+          goto ok;
         }
         (iop_a_src += 1, wuffs_base__make_empty_struct());
-        if ((v_whitespace_length >= 65534) || (v_c == 10)) {
+        if ((v_whitespace_length >= 65534) || (v_c == self->private_impl.f_trailer_stop)) {
           *iop_a_dst++ = wuffs_base__make_token(
               (((uint64_t)(0)) << WUFFS_BASE__TOKEN__VALUE_MINOR__SHIFT) |
               (((uint64_t)((v_whitespace_length + 1))) << WUFFS_BASE__TOKEN__LENGTH__SHIFT));
           v_whitespace_length = 0;
-          if (v_c == 10) {
-            goto label__outer__break;
+          if (v_c == self->private_impl.f_trailer_stop) {
+            status = wuffs_base__make_status(NULL);
+            goto ok;
           }
           goto label__outer__continue;
         }
@@ -27656,13 +27670,13 @@ wuffs_json__decoder__decode_trailing_new_line(
 
     goto ok;
     ok:
-    self->private_impl.p_decode_trailing_new_line[0] = 0;
+    self->private_impl.p_decode_trailer[0] = 0;
     goto exit;
   }
 
   goto suspend;
   suspend:
-  self->private_impl.p_decode_trailing_new_line[0] = wuffs_base__status__is_suspension(&status) ? coro_susp_point : 0;
+  self->private_impl.p_decode_trailer[0] = wuffs_base__status__is_suspension(&status) ? coro_susp_point : 0;
 
   goto exit;
   exit:
diff --git a/script/print-json-token-debug-format.c b/script/print-json-token-debug-format.c
index 29342c520f24f98b5679c3689ec3c60985ef6be4..b92a3a997433741ced89986409eb35646786470f 100644
--- a/script/print-json-token-debug-format.c
+++ b/script/print-json-token-debug-format.c
@@ -298,7 +298,7 @@ main1(int argc, char** argv) {
         WUFFS_JSON__QUIRK_ALLOW_INF_NAN_NUMBERS,
         WUFFS_JSON__QUIRK_ALLOW_LEADING_ASCII_RECORD_SEPARATOR,
         WUFFS_JSON__QUIRK_ALLOW_LEADING_UNICODE_BYTE_ORDER_MARK,
-        WUFFS_JSON__QUIRK_ALLOW_TRAILING_NEW_LINE,
+        WUFFS_JSON__QUIRK_ALLOW_TRAILING_FILLER,
         WUFFS_JSON__QUIRK_REPLACE_INVALID_UNICODE,
         0,
     };
diff --git a/std/json/decode_json.wuffs b/std/json/decode_json.wuffs
index df48da1c669bcb85c986710e763f450e26b0ae55..bd83ae883c5652546b92c5a74ef05576f9463201 100644
--- a/std/json/decode_json.wuffs
+++ b/std/json/decode_json.wuffs
@@ -15,14 +15,15 @@
 pub struct decoder? implements base.token_decoder(
 	quirks : array[QUIRKS_COUNT] base.bool,
 
-	allow_leading_ars      : base.bool,
-	allow_leading_ubom     : base.bool,
-	allow_trailing_comment : base.bool,
+	allow_leading_ars  : base.bool,
+	allow_leading_ubom : base.bool,
 
 	end_of_data : base.bool,
 
+	trailer_stop : base.u8,
+
 	// comment_type is set as a side-effect of decode_comment?.
-	//  - 0 means failure.
+	//  - 0 means no comment.
 	//  - 1 means a block comment.
 	//  - 2 means a line  comment.
 	comment_type : base.u8,
@@ -123,6 +124,14 @@ pub func decoder.decode_tokens?(dst: base.token_writer, src: base.io_reader, wor
 		return base."@end of data"
 	}
 
+	if this.quirks[QUIRK_EXPECT_TRAILING_NEW_LINE_OR_EOF - QUIRKS_BASE] {
+		if this.quirks[QUIRK_ALLOW_COMMENT_BLOCK - QUIRKS_BASE] or
+			this.quirks[QUIRK_ALLOW_COMMENT_LINE - QUIRKS_BASE] or
+			this.quirks[QUIRK_ALLOW_TRAILING_FILLER - QUIRKS_BASE] {
+			return "#bad quirk combination"
+		}
+	}
+
 	if this.quirks[QUIRK_ALLOW_LEADING_ASCII_RECORD_SEPARATOR - QUIRKS_BASE] or
 		this.quirks[QUIRK_ALLOW_LEADING_UNICODE_BYTE_ORDER_MARK - QUIRKS_BASE] {
 		this.decode_leading?(dst: args.dst, src: args.src)
@@ -1176,7 +1185,9 @@ pub func decoder.decode_tokens?(dst: base.token_writer, src: base.io_reader, wor
 			if this.quirks[QUIRK_ALLOW_COMMENT_BLOCK - QUIRKS_BASE] or
 				this.quirks[QUIRK_ALLOW_COMMENT_LINE - QUIRKS_BASE] {
 				this.decode_comment?(dst: args.dst, src: args.src)
-				continue.outer
+				if this.comment_type > 0 {
+					continue.outer
+				}
 			}
 		}
 
@@ -1191,8 +1202,9 @@ pub func decoder.decode_tokens?(dst: base.token_writer, src: base.io_reader, wor
 		expect = expect_after_value
 	} endwhile.outer
 
-	if this.quirks[QUIRK_ALLOW_TRAILING_NEW_LINE - QUIRKS_BASE] {
-		this.decode_trailing_new_line?(dst: args.dst, src: args.src)
+	if this.quirks[QUIRK_ALLOW_TRAILING_FILLER - QUIRKS_BASE] or
+		this.quirks[QUIRK_EXPECT_TRAILING_NEW_LINE_OR_EOF - QUIRKS_BASE] {
+		this.decode_trailer?(dst: args.dst, src: args.src)
 	}
 
 	this.end_of_data = true
@@ -1438,7 +1450,7 @@ pri func decoder.decode_comment?(dst: base.token_writer, src: base.io_reader) {
 			continue
 		}
 		if args.src.is_closed() {
-			return "#bad input"
+			return ok
 		}
 		yield? base."$short read"
 	} endwhile
@@ -1562,8 +1574,6 @@ pri func decoder.decode_comment?(dst: base.token_writer, src: base.io_reader) {
 			} endwhile
 		} endwhile.comment_line
 	}
-
-	return "#bad input"
 }
 
 pri func decoder.decode_inf_nan?(dst: base.token_writer, src: base.io_reader) {
@@ -1677,11 +1687,15 @@ pri func decoder.decode_inf_nan?(dst: base.token_writer, src: base.io_reader) {
 	} endwhile
 }
 
-pri func decoder.decode_trailing_new_line?(dst: base.token_writer, src: base.io_reader) {
+pri func decoder.decode_trailer?(dst: base.token_writer, src: base.io_reader) {
 	var c                 : base.u8
 	var whitespace_length : base.u32[..= 0xFFFE]
 
-	this.allow_trailing_comment = this.quirks[QUIRK_ALLOW_TRAILING_COMMENT - QUIRKS_BASE]
+	if this.quirks[QUIRK_EXPECT_TRAILING_NEW_LINE_OR_EOF - QUIRKS_BASE] {
+		this.trailer_stop = '\n'
+	} else {
+		this.trailer_stop = 0
+	}
 
 	while.outer true {
 		if args.dst.length() <= 0 {
@@ -1714,27 +1728,25 @@ pri func decoder.decode_trailing_new_line?(dst: base.token_writer, src: base.io_
 						value_major: 0, value_minor: 0, continued: 0, length: whitespace_length)
 					whitespace_length = 0
 				}
-				if this.allow_trailing_comment {
-					this.decode_comment?(dst: args.dst, src: args.src)
-					c = 0
-					whitespace_length = 0
-					if this.comment_type == 1 {  // Block comment.
-						this.allow_trailing_comment = false
-						continue.outer
-					} else if this.comment_type == 2 {  // Line comment.
-						break.outer
-					}
+				if this.trailer_stop > 0 {
+					return "#bad input"
 				}
-				return "#bad input"
+				this.decode_comment?(dst: args.dst, src: args.src)
+				c = 0
+				whitespace_length = 0
+				if this.comment_type > 0 {
+					continue.outer
+				}
+				return ok
 			}
 
 			args.src.skip_u32_fast!(actual: 1, worst_case: 1)
-			if (whitespace_length >= 0xFFFE) or (c == '\n') {
+			if (whitespace_length >= 0xFFFE) or (c == this.trailer_stop) {
 				args.dst.write_simple_token_fast!(
 					value_major: 0, value_minor: 0, continued: 0, length: whitespace_length + 1)
 				whitespace_length = 0
-				if c == '\n' {
-					break.outer
+				if c == this.trailer_stop {
+					return ok
 				}
 				continue.outer
 			}
diff --git a/std/json/decode_quirks.wuffs b/std/json/decode_quirks.wuffs
index 5f31b5ba4cad8107ebda50f067c18bb14353e17d..9970c356cfddc53e3ca7742f3ef020e5acc80046 100644
--- a/std/json/decode_quirks.wuffs
+++ b/std/json/decode_quirks.wuffs
@@ -97,16 +97,17 @@ pub const QUIRK_ALLOW_BACKSLASH_X_AS_CODE_POINTS : base.u32 = 0x4909_9400 | 0x09
 pub const QUIRK_ALLOW_BACKSLASH_ZERO : base.u32 = 0x4909_9400 | 0x0A
 
 // When this quirk is enabled, "/* C/C++ style block comments */" are accepted
-// anywhere whitespace would be, although see the QUIRK_ALLOW_TRAILING_COMMENT
-// comment for additional interaction when combining multiple quirks.
+// anywhere whitespace would be. See also QUIRK_ALLOW_TRAILING_FILLER.
 //
 // They produce WUFFS_BASE__TOKEN__VBD__FILLER__COMMENT_BLOCK tokens. The token
 // chain's source bytes includes the starting "/*" and the ending "*/".
+//
+// To avoid ambiguity (as comments can contain new lines), this quirk cannot be
+// combined with QUIRK_EXPECT_TRAILING_NEW_LINE_OR_EOF.
 pub const QUIRK_ALLOW_COMMENT_BLOCK : base.u32 = 0x4909_9400 | 0x0B
 
 // When this quirk is enabled, "// C/C++ style line comments\n" are accepted
-// anywhere whitespace would be, although see the QUIRK_ALLOW_TRAILING_COMMENT
-// comment for additional interaction when combining multiple quirks.
+// anywhere whitespace would be. See also QUIRK_ALLOW_TRAILING_FILLER.
 //
 // A line comment may not omit the ending "\n", even if there is no input
 // afterwards (i.e. the prospective line comment ends with the end-of-file).
@@ -117,6 +118,9 @@ pub const QUIRK_ALLOW_COMMENT_BLOCK : base.u32 = 0x4909_9400 | 0x0B
 // Even if the line comments are on consecutive lines, each line comment is a
 // separate token chain. There may be whitespace tokens between one line
 // comment's ending "\n" and the next one's starting "//".
+//
+// To avoid ambiguity (as comments can contain new lines), this quirk cannot be
+// combined with QUIRK_EXPECT_TRAILING_NEW_LINE_OR_EOF.
 pub const QUIRK_ALLOW_COMMENT_LINE : base.u32 = 0x4909_9400 | 0x0C
 
 // When this quirk is enabled, there may be a comma after the final array
@@ -140,8 +144,9 @@ pub const QUIRK_ALLOW_INF_NAN_NUMBERS : base.u32 = 0x4909_9400 | 0x0E
 // When combined with QUIRK_ALLOW_LEADING_UNICODE_BYTE_ORDER_MARK, either mark
 // may come first in the byte stream.
 //
-// When combined with QUIRK_ALLOW_TRAILING_NEW_LINE, this format is also known
-// as RFC 7464, Json Text Sequences and MIME type "application/json-seq".
+// When combined with QUIRK_EXPECT_TRAILING_NEW_LINE_OR_EOF, this format is
+// also known as RFC 7464, Json Text Sequences and MIME type
+// "application/json-seq".
 pub const QUIRK_ALLOW_LEADING_ASCII_RECORD_SEPARATOR : base.u32 = 0x4909_9400 | 0x0F
 
 // When this quirk is enabled, the input byte stream may optionally start with
@@ -152,40 +157,43 @@ pub const QUIRK_ALLOW_LEADING_ASCII_RECORD_SEPARATOR : base.u32 = 0x4909_9400 |
 // may come first in the byte stream.
 pub const QUIRK_ALLOW_LEADING_UNICODE_BYTE_ORDER_MARK : base.u32 = 0x4909_9400 | 0x10
 
-// When this quirk is enabled and both:
-//  - QUIRK_ALLOW_TRAILING_NEW_LINE is enabled,
-//  - at least one of QUIRK_ALLOW_COMMENT_ETC is enabled,
-// the trailing whitespace may optionally contain a single comment. As per
-// QUIRK_ALLOW_TRAILING_NEW_LINE, processing will still stop at the first
-// trailing '\n' (outside of a block comment), even if more comments followed.
-//
-// For a trailing block comment, new lines within the comment are not counted
-// and after the comment concludes, the decoder will continue consuming
-// whitespace up to and including the next '\n' (or end-of-file).
-//
-// For a trailing line comment, the decoder stops immediately after the
-// comment. If not stopped by end-of-file, this stops after the '\n' that
-// concludes (and is part of) the comment. One implication is that if multiple
-// line comments trail a JSON value, only the first one will be processed.
+// When this quirk is enabled, following a successful decoding of a top-level
+// JSON value, any trailing whitespace (ASCII characters 0x09, 0x0A, 0x0D or
+// 0x20) and/or comments (if QUIRK_ALLOW_COMMENT_ETC is enabled) are also
+// consumed (and WUFFS_BASE__TOKEN__VBC__FILLER tokens emitted) up to but
+// excluding the end-of-file or the next non-filler byte.
+//
+// Trailing non-filler is not an error. Decoding simply stops before it.
+//
+// To avoid ambiguity, this quirk cannot be combined with
+// QUIRK_EXPECT_TRAILING_NEW_LINE_OR_EOF. Unlike that quirk, enabling this
+// quirk will consume multiple trailing '\n' bytes.
+pub const QUIRK_ALLOW_TRAILING_FILLER : base.u32 = 0x4909_9400 | 0x11
+
+// Deprecated: equivalent to QUIRK_ALLOW_TRAILING_FILLER.
 pub const QUIRK_ALLOW_TRAILING_COMMENT : base.u32 = 0x4909_9400 | 0x11
 
 // When this quirk is enabled, following a successful decoding of a top-level
 // JSON value, any trailing whitespace (ASCII characters 0x09, 0x0A, 0x0D or
 // 0x20) is also consumed (and WUFFS_BASE__TOKEN__VBC__FILLER tokens emitted)
-// up to the end-of-file or up to and including a single new line (ASCII 0x0A,
-// also known as '\n'), whichever comes first. This trailing whitespace is not
-// mandatory, but it is consumed if present.
+// up to but excluding the end-of-file or up to and including a single new line
+// (ASCII 0x0A, also known as '\n'), whichever comes first. A trailing '\n' is
+// not mandatory (if at end-of-file), but it is consumed if present and will
+// stop decoding.
+//
+// Trailing non-whitespace after a trailing '\n' is ignored.
 //
-// When enabled, trailing non-whitespace (before a '\n') is an error (unless it
-// is a comment and QUIRK_ALLOW_TRAILING_COMMENT is enabled; see above). For
-// example, with "007" input, decoding with this quirk disabled (the default
-// case) will consume just 1 byte ("0") and leave the rest ("07") unread
-// without error, as "0" is a perfectly valid JSON value (but "00" is not).
-// Decoding "007" (or "007\n") with this quirk enabled will return an error.
+// Trailing non-whitespace, before EOF or '\n', is an error. For example, with
+// "007" input, decoding with this quirk disabled (the default case) will
+// consume just 1 byte ("0") and leave the rest ("07") unread without error, as
+// "0" is a perfectly valid JSON value (but "00" is not). Decoding "007" (or
+// "007\n") with this quirk enabled will return an error.
 //
-// When enabled, the decoder will not consume more than one trailing '\n'
-// (outside of a block comment), nor will it consume any other whitespace
-// immediately after a trailing '\n'.
+// To avoid ambiguity (as comments can contain new lines), this quirk cannot be
+// combined with any of:
+//  - QUIRK_ALLOW_COMMENT_BLOCK
+//  - QUIRK_ALLOW_COMMENT_LINE
+//  - QUIRK_ALLOW_TRAILING_FILLER
 //
 // If a JSON encoder avoids emitting (optional) '\n' bytes, other than a single
 // '\n' after each top-level value in a multi-JSON-value stream, this format is
@@ -199,13 +207,9 @@ pub const QUIRK_ALLOW_TRAILING_COMMENT : base.u32 = 0x4909_9400 | 0x11
 // When combined with QUIRK_ALLOW_LEADING_ASCII_RECORD_SEPARATOR, this format
 // is also known as RFC 7464, Json Text Sequences and MIME type
 // "application/json-seq".
-//
-// When combined with QUIRK_ALLOW_COMMENT_BLOCK or QUIRK_ALLOW_COMMENT_LINE, it
-// is an error for a comment to occur in this trailing whitespace, unless
-// QUIRK_ALLOW_TRAILING_COMMENT is also enabled. Be aware that block comments
-// can contain multiple new lines, so combining such quirks can break the
-// "exactly one JSON value per line" assumption for newline-delimited but
-// otherwise compact (minified) JSON.
+pub const QUIRK_EXPECT_TRAILING_NEW_LINE_OR_EOF : base.u32 = 0x4909_9400 | 0x12
+
+// Deprecated: equivalent to QUIRK_EXPECT_TRAILING_NEW_LINE_OR_EOF.
 pub const QUIRK_ALLOW_TRAILING_NEW_LINE : base.u32 = 0x4909_9400 | 0x12
 
 // When this quirk is enabled, JSON Pointer strings containing "~r" or "~n",
