diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index 4f104d483ad80a74c791730d47d8dd6f93c04f40..c6167b91b936c31a8fcf9cb21b7a92a8be9cd348 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -87,7 +87,8 @@ fuzz(wuffs_base__io_buffer* src, uint64_t hash) {
   // Ignore the checksum for 99.99%-ish of all input. When fuzzers generate
   // random input, the checkum is very unlikely to match. Still, it's useful to
   // verify that checksumming does not lead to e.g. buffer overflows.
-  wuffs_zlib__decoder__set_ignore_checksum(&dec, hash & 0xFFFE);
+  wuffs_zlib__decoder__set_quirk_enabled(
+      &dec, WUFFS_BASE__QUIRK_IGNORE_CHECKSUM, hash & 0xFFFE);
 
   uint8_t dst_buffer[DST_BUFFER_ARRAY_SIZE];
   wuffs_base__io_buffer dst = ((wuffs_base__io_buffer){
