diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index 0f4edbae171bfce82f0dcb618a7e18a7e7788cd0..813f7e712ae18aef6e370ddb659a966b57d3507d 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -26348,6 +26348,16 @@ wuffs_gzip__decoder__set_quirk_enabled(
     wuffs_gzip__decoder* self,
     uint32_t a_quirk,
     bool a_enabled) {
+  if (!self) {
+    return wuffs_base__make_empty_struct();
+  }
+  if (self->private_impl.magic != WUFFS_BASE__MAGIC) {
+    return wuffs_base__make_empty_struct();
+  }
+
+  if (a_quirk == 1) {
+    self->private_impl.f_ignore_checksum = a_enabled;
+  }
   return wuffs_base__make_empty_struct();
 }
 
@@ -29967,6 +29977,16 @@ wuffs_zlib__decoder__set_quirk_enabled(
     wuffs_zlib__decoder* self,
     uint32_t a_quirk,
     bool a_enabled) {
+  if (!self) {
+    return wuffs_base__make_empty_struct();
+  }
+  if (self->private_impl.magic != WUFFS_BASE__MAGIC) {
+    return wuffs_base__make_empty_struct();
+  }
+
+  if (a_quirk == 1) {
+    self->private_impl.f_ignore_checksum = a_enabled;
+  }
   return wuffs_base__make_empty_struct();
 }
 
@@ -31398,6 +31418,17 @@ wuffs_png__decoder__set_quirk_enabled(
     wuffs_png__decoder* self,
     uint32_t a_quirk,
     bool a_enabled) {
+  if (!self) {
+    return wuffs_base__make_empty_struct();
+  }
+  if (self->private_impl.magic != WUFFS_BASE__MAGIC) {
+    return wuffs_base__make_empty_struct();
+  }
+
+  if (a_quirk == 1) {
+    self->private_impl.f_ignore_checksum = a_enabled;
+    wuffs_zlib__decoder__set_quirk_enabled(&self->private_data.f_zlib, a_quirk, a_enabled);
+  }
   return wuffs_base__make_empty_struct();
 }
 
diff --git a/std/gzip/decode_gzip.wuffs b/std/gzip/decode_gzip.wuffs
index e8e5ce1d74462728c6fafc00828fb4604344d28a..67fb478b9e6d7ff6488e7fd159fd8a8705358943 100644
--- a/std/gzip/decode_gzip.wuffs
+++ b/std/gzip/decode_gzip.wuffs
@@ -37,6 +37,9 @@ pub func decoder.set_ignore_checksum!(ic: base.bool) {
 }
 
 pub func decoder.set_quirk_enabled!(quirk: base.u32, enabled: base.bool) {
+	if args.quirk == base.QUIRK_IGNORE_CHECKSUM {
+		this.ignore_checksum = args.enabled
+	}
 }
 
 pub func decoder.workbuf_len() base.range_ii_u64 {
diff --git a/std/png/decode_png.wuffs b/std/png/decode_png.wuffs
index 9363d4c2a0633f0557d96a870a974d9ff6a9526b..29d98c403832fa3f90f67a5d79be2af89ec00e64 100644
--- a/std/png/decode_png.wuffs
+++ b/std/png/decode_png.wuffs
@@ -99,6 +99,10 @@ pub func decoder.set_ignore_checksum!(ic: base.bool) {
 }
 
 pub func decoder.set_quirk_enabled!(quirk: base.u32, enabled: base.bool) {
+	if args.quirk == base.QUIRK_IGNORE_CHECKSUM {
+		this.ignore_checksum = args.enabled
+		this.zlib.set_quirk_enabled!(quirk: args.quirk, enabled: args.enabled)
+	}
 }
 
 pub func decoder.decode_image_config?(dst: nptr base.image_config, src: base.io_reader) {
diff --git a/std/zlib/decode_zlib.wuffs b/std/zlib/decode_zlib.wuffs
index 8fef738d308c33c1151797e107bad450c2f7a722..e77e52b8457bd47fd76f11a55e320d90fd6036eb 100644
--- a/std/zlib/decode_zlib.wuffs
+++ b/std/zlib/decode_zlib.wuffs
@@ -64,6 +64,9 @@ pub func decoder.set_ignore_checksum!(ic: base.bool) {
 }
 
 pub func decoder.set_quirk_enabled!(quirk: base.u32, enabled: base.bool) {
+	if args.quirk == base.QUIRK_IGNORE_CHECKSUM {
+		this.ignore_checksum = args.enabled
+	}
 }
 
 pub func decoder.workbuf_len() base.range_ii_u64 {
