diff --git a/fuzz/c/std/bmp_fuzzer.c b/fuzz/c/std/bmp_fuzzer.c
index 7be150a31b6e9da24df7c0af910bdaca7fdb645f..139e10933b8ed76d11a9dd17e1dcc7901b52e673 100644
--- a/fuzz/c/std/bmp_fuzzer.c
+++ b/fuzz/c/std/bmp_fuzzer.c
@@ -22,7 +22,7 @@
 This fuzzer (the fuzz function) is typically run indirectly, by a framework
 such as https://github.com/google/oss-fuzz calling LLVMFuzzerTestOneInput.
 
-When working on the fuzz implementation, or as a sanity check, defining
+When working on the fuzz implementation, or as a coherence check, defining
 WUFFS_CONFIG__FUZZLIB_MAIN will let you manually run fuzz over a set of files:
 
 gcc -DWUFFS_CONFIG__FUZZLIB_MAIN bmp_fuzzer.c
@@ -43,8 +43,8 @@ It should print "PASS", amongst other information, and exit(0).
 #define WUFFS_IMPLEMENTATION
 
 // Defining the WUFFS_CONFIG__MODULE* macros are optional, but it lets users of
-// release/c/etc.c whitelist which parts of Wuffs to build. That file contains
-// the entire Wuffs standard library, implementing a variety of codecs and file
+// release/c/etc.c choose which parts of Wuffs to build. That file contains the
+// entire Wuffs standard library, implementing a variety of codecs and file
 // formats. Without this macro definition, an optimizing compiler or linker may
 // very well discard Wuffs code for unused codecs, but listing the Wuffs
 // modules we use makes that process explicit. Preprocessing means that such
diff --git a/fuzz/c/std/cbor_fuzzer.c b/fuzz/c/std/cbor_fuzzer.c
index 4dd650687c067610c5b9ba1a3456c2d51414cc20..1df5b555fafb56ff1899e3220198b7beca5adc2e 100644
--- a/fuzz/c/std/cbor_fuzzer.c
+++ b/fuzz/c/std/cbor_fuzzer.c
@@ -22,7 +22,7 @@
 This fuzzer (the fuzz function) is typically run indirectly, by a framework
 such as https://github.com/google/oss-fuzz calling LLVMFuzzerTestOneInput.
 
-When working on the fuzz implementation, or as a sanity check, defining
+When working on the fuzz implementation, or as a coherence check, defining
 WUFFS_CONFIG__FUZZLIB_MAIN will let you manually run fuzz over a set of files:
 
 gcc -DWUFFS_CONFIG__FUZZLIB_MAIN cbor_fuzzer.c
@@ -43,8 +43,8 @@ It should print "PASS", amongst other information, and exit(0).
 #define WUFFS_IMPLEMENTATION
 
 // Defining the WUFFS_CONFIG__MODULE* macros are optional, but it lets users of
-// release/c/etc.c whitelist which parts of Wuffs to build. That file contains
-// the entire Wuffs standard library, implementing a variety of codecs and file
+// release/c/etc.c choose which parts of Wuffs to build. That file contains the
+// entire Wuffs standard library, implementing a variety of codecs and file
 // formats. Without this macro definition, an optimizing compiler or linker may
 // very well discard Wuffs code for unused codecs, but listing the Wuffs
 // modules we use makes that process explicit. Preprocessing means that such
@@ -401,7 +401,7 @@ fuzz(wuffs_base__io_buffer* full_src, uint64_t hash) {
   //
   // The fuzz_simple implementation shows how easy decoding with Wuffs is when
   // all you want is to run LLVMFuzzerTestOneInput's built-in (Wuffs API
-  // agnostic) checks (e.g. the ASan address sanitizer) and you don't really
+  // independent) checks (e.g. the ASan address sanitizer) and you don't really
   // care what the output is, just that it doesn't crash.
   //
   // The fuzz_complex implementation adds many more Wuffs API specific checks
diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 0f5de60a1ed205bc2ba4bf22b29121ba068669d2..cf58b33c2c71e7e13048e36dca093907ffcf31b3 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -22,7 +22,7 @@
 This fuzzer (the fuzz function) is typically run indirectly, by a framework
 such as https://github.com/google/oss-fuzz calling LLVMFuzzerTestOneInput.
 
-When working on the fuzz implementation, or as a sanity check, defining
+When working on the fuzz implementation, or as a coherence check, defining
 WUFFS_CONFIG__FUZZLIB_MAIN will let you manually run fuzz over a set of files:
 
 gcc -DWUFFS_CONFIG__FUZZLIB_MAIN gif_fuzzer.c
@@ -43,8 +43,8 @@ It should print "PASS", amongst other information, and exit(0).
 #define WUFFS_IMPLEMENTATION
 
 // Defining the WUFFS_CONFIG__MODULE* macros are optional, but it lets users of
-// release/c/etc.c whitelist which parts of Wuffs to build. That file contains
-// the entire Wuffs standard library, implementing a variety of codecs and file
+// release/c/etc.c choose which parts of Wuffs to build. That file contains the
+// entire Wuffs standard library, implementing a variety of codecs and file
 // formats. Without this macro definition, an optimizing compiler or linker may
 // very well discard Wuffs code for unused codecs, but listing the Wuffs
 // modules we use makes that process explicit. Preprocessing means that such
diff --git a/fuzz/c/std/json_fuzzer.c b/fuzz/c/std/json_fuzzer.c
index 2e1fa2c3307bd639136529144c9dde9513fe6484..0c22678492f47a5f2746005522b7d075322c09f7 100644
--- a/fuzz/c/std/json_fuzzer.c
+++ b/fuzz/c/std/json_fuzzer.c
@@ -22,7 +22,7 @@
 This fuzzer (the fuzz function) is typically run indirectly, by a framework
 such as https://github.com/google/oss-fuzz calling LLVMFuzzerTestOneInput.
 
-When working on the fuzz implementation, or as a sanity check, defining
+When working on the fuzz implementation, or as a coherence check, defining
 WUFFS_CONFIG__FUZZLIB_MAIN will let you manually run fuzz over a set of files:
 
 gcc -DWUFFS_CONFIG__FUZZLIB_MAIN json_fuzzer.c
@@ -43,8 +43,8 @@ It should print "PASS", amongst other information, and exit(0).
 #define WUFFS_IMPLEMENTATION
 
 // Defining the WUFFS_CONFIG__MODULE* macros are optional, but it lets users of
-// release/c/etc.c whitelist which parts of Wuffs to build. That file contains
-// the entire Wuffs standard library, implementing a variety of codecs and file
+// release/c/etc.c choose which parts of Wuffs to build. That file contains the
+// entire Wuffs standard library, implementing a variety of codecs and file
 // formats. Without this macro definition, an optimizing compiler or linker may
 // very well discard Wuffs code for unused codecs, but listing the Wuffs
 // modules we use makes that process explicit. Preprocessing means that such
@@ -421,7 +421,7 @@ fuzz(wuffs_base__io_buffer* full_src, uint64_t hash) {
   //
   // The fuzz_simple implementation shows how easy decoding with Wuffs is when
   // all you want is to run LLVMFuzzerTestOneInput's built-in (Wuffs API
-  // agnostic) checks (e.g. the ASan address sanitizer) and you don't really
+  // independent) checks (e.g. the ASan address sanitizer) and you don't really
   // care what the output is, just that it doesn't crash.
   //
   // The fuzz_complex implementation adds many more Wuffs API specific checks
diff --git a/fuzz/c/std/png_fuzzer.c b/fuzz/c/std/png_fuzzer.c
index 33b4d2e15fc984e2e9eb48fe140fd52fff498d73..1191fda0221a66afa153d593ad622a40bf7f30f7 100644
--- a/fuzz/c/std/png_fuzzer.c
+++ b/fuzz/c/std/png_fuzzer.c
@@ -22,7 +22,7 @@
 This fuzzer (the fuzz function) is typically run indirectly, by a framework
 such as https://github.com/google/oss-fuzz calling LLVMFuzzerTestOneInput.
 
-When working on the fuzz implementation, or as a sanity check, defining
+When working on the fuzz implementation, or as a coherence check, defining
 WUFFS_CONFIG__FUZZLIB_MAIN will let you manually run fuzz over a set of files:
 
 gcc -DWUFFS_CONFIG__FUZZLIB_MAIN png_fuzzer.c
@@ -43,8 +43,8 @@ It should print "PASS", amongst other information, and exit(0).
 #define WUFFS_IMPLEMENTATION
 
 // Defining the WUFFS_CONFIG__MODULE* macros are optional, but it lets users of
-// release/c/etc.c whitelist which parts of Wuffs to build. That file contains
-// the entire Wuffs standard library, implementing a variety of codecs and file
+// release/c/etc.c choose which parts of Wuffs to build. That file contains the
+// entire Wuffs standard library, implementing a variety of codecs and file
 // formats. Without this macro definition, an optimizing compiler or linker may
 // very well discard Wuffs code for unused codecs, but listing the Wuffs
 // modules we use makes that process explicit. Preprocessing means that such
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index c6167b91b936c31a8fcf9cb21b7a92a8be9cd348..428c5e7eae0fa1fd6a6e9fd1bd35a35028b37568 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -22,7 +22,7 @@
 This fuzzer (the fuzz function) is typically run indirectly, by a framework
 such as https://github.com/google/oss-fuzz calling LLVMFuzzerTestOneInput.
 
-When working on the fuzz implementation, or as a sanity check, defining
+When working on the fuzz implementation, or as a coherence check, defining
 WUFFS_CONFIG__FUZZLIB_MAIN will let you manually run fuzz over a set of files:
 
 gcc -DWUFFS_CONFIG__FUZZLIB_MAIN zlib_fuzzer.c
@@ -43,8 +43,8 @@ It should print "PASS", amongst other information, and exit(0).
 #define WUFFS_IMPLEMENTATION
 
 // Defining the WUFFS_CONFIG__MODULE* macros are optional, but it lets users of
-// release/c/etc.c whitelist which parts of Wuffs to build. That file contains
-// the entire Wuffs standard library, implementing a variety of codecs and file
+// release/c/etc.c choose which parts of Wuffs to build. That file contains the
+// entire Wuffs standard library, implementing a variety of codecs and file
 // formats. Without this macro definition, an optimizing compiler or linker may
 // very well discard Wuffs code for unused codecs, but listing the Wuffs
 // modules we use makes that process explicit. Preprocessing means that such
