diff --git a/script/manual-test-parse-number-f64.cc b/script/manual-test-parse-number-f64.cc
index efe7c6dfef73c735e8ddbbb37ec8da82442fa599..c11f27dc73fbc11a00bf7a41123ab137fc8b568f 100644
--- a/script/manual-test-parse-number-f64.cc
+++ b/script/manual-test-parse-number-f64.cc
@@ -38,8 +38,8 @@
 #define WUFFS_IMPLEMENTATION
 
 // Defining the WUFFS_CONFIG__MODULE* macros are optional, but it lets users of
-// release/c/etc.c whitelist which parts of Wuffs to build. That file contains
-// the entire Wuffs standard library, implementing a variety of codecs and file
+// release/c/etc.c choose which parts of Wuffs to build. That file contains the
+// entire Wuffs standard library, implementing a variety of codecs and file
 // formats. Without this macro definition, an optimizing compiler or linker may
 // very well discard Wuffs code for unused codecs, but listing the Wuffs
 // modules we use makes that process explicit. Preprocessing means that such
diff --git a/script/print-json-token-debug-format.c b/script/print-json-token-debug-format.c
index 79a7bb72d51f9eec2c15f9e3bc927c5c0e57f7f6..14af81784f48effb2519d29c7439cdb0b36ace6b 100644
--- a/script/print-json-token-debug-format.c
+++ b/script/print-json-token-debug-format.c
@@ -74,8 +74,8 @@
 #define WUFFS_IMPLEMENTATION
 
 // Defining the WUFFS_CONFIG__MODULE* macros are optional, but it lets users of
-// release/c/etc.c whitelist which parts of Wuffs to build. That file contains
-// the entire Wuffs standard library, implementing a variety of codecs and file
+// release/c/etc.c choose which parts of Wuffs to build. That file contains the
+// entire Wuffs standard library, implementing a variety of codecs and file
 // formats. Without this macro definition, an optimizing compiler or linker may
 // very well discard Wuffs code for unused codecs, but listing the Wuffs
 // modules we use makes that process explicit. Preprocessing means that such
diff --git a/script/process-json-numbers.c b/script/process-json-numbers.c
index f4a781f204795af92817d288e9cd4edfb0d4e7f6..e44a6dcc240c30330978cb6ffd620570f9390096 100644
--- a/script/process-json-numbers.c
+++ b/script/process-json-numbers.c
@@ -53,8 +53,8 @@
 #define WUFFS_IMPLEMENTATION
 
 // Defining the WUFFS_CONFIG__MODULE* macros are optional, but it lets users of
-// release/c/etc.c whitelist which parts of Wuffs to build. That file contains
-// the entire Wuffs standard library, implementing a variety of codecs and file
+// release/c/etc.c choose which parts of Wuffs to build. That file contains the
+// entire Wuffs standard library, implementing a variety of codecs and file
 // formats. Without this macro definition, an optimizing compiler or linker may
 // very well discard Wuffs code for unused codecs, but listing the Wuffs
 // modules we use makes that process explicit. Preprocessing means that such
diff --git a/std/bmp/decode_bmp.wuffs b/std/bmp/decode_bmp.wuffs
index 9115b084b54096bd23e52eb758ecbc1a30d2a85f..18cfa08a39a425407775178fc730a8515655f1f3 100644
--- a/std/bmp/decode_bmp.wuffs
+++ b/std/bmp/decode_bmp.wuffs
@@ -1157,8 +1157,8 @@ pub func decoder.tell_me_more?(dst: base.io_writer, minfo: nptr base.more_inform
 			y: this.io_redirect_pos,
 			z: 0xFFFF_FFFF_FFFF_FFFF)
 	}
-	// Setting io_redirect_fourcc to a dummy value of 1 will cause future calls
-	// to return an error.
+	// Setting io_redirect_fourcc to a placeholder value of 1 will cause future
+	// calls to return an error.
 	this.io_redirect_fourcc = 1
 }
 
