diff --git a/fuzz/c/std/bmp_fuzzer.c b/fuzz/c/std/bmp_fuzzer.c
index 139e10933b8ed76d11a9dd17e1dcc7901b52e673..2f0f45d1d056c5acb21b2451655abb9d6a9d052e 100644
--- a/fuzz/c/std/bmp_fuzzer.c
+++ b/fuzz/c/std/bmp_fuzzer.c
@@ -66,7 +66,7 @@ fuzz(wuffs_base__io_buffer* src, uint64_t hash) {
   wuffs_base__status status = wuffs_bmp__decoder__initialize(
       &dec, sizeof dec, WUFFS_VERSION,
       (hash & 1) ? WUFFS_INITIALIZE__LEAVE_INTERNAL_BUFFERS_UNINITIALIZED : 0);
-  hash >>= 1;
+  hash = wuffs_base__u64__rotate_right(hash, 1);
   if (!wuffs_base__status__is_ok(&status)) {
     return wuffs_base__status__message(&status);
   }
diff --git a/fuzz/c/std/cbor_fuzzer.c b/fuzz/c/std/cbor_fuzzer.c
index 1df5b555fafb56ff1899e3220198b7beca5adc2e..6b49617fe56becea28f8b0de4b7c4d19cccaed59 100644
--- a/fuzz/c/std/cbor_fuzzer.c
+++ b/fuzz/c/std/cbor_fuzzer.c
@@ -221,12 +221,13 @@ fuzz_one_token(wuffs_base__token t,
 }
 
 uint64_t  //
-buffer_limit(uint64_t hash_6_bits, uint64_t min, uint64_t max) {
+buffer_limit(uint64_t hash, uint64_t min, uint64_t max) {
+  hash &= 0x3F;
   uint64_t n;
-  if (hash_6_bits < 0x20) {
-    n = min + hash_6_bits;
+  if (hash < 0x20) {
+    n = min + hash;
   } else {
-    n = max - (0x3F - hash_6_bits);
+    n = max - (0x3F - hash);
   }
   if (n < min) {
     return min;
@@ -237,15 +238,15 @@ buffer_limit(uint64_t hash_6_bits, uint64_t min, uint64_t max) {
 }
 
 const char*  //
-fuzz_complex(wuffs_base__io_buffer* full_src, uint64_t hash_56_bits) {
+fuzz_complex(wuffs_base__io_buffer* full_src, uint64_t hash) {
   uint64_t tok_limit = buffer_limit(
-      hash_56_bits & 0x3F, WUFFS_CBOR__DECODER_DST_TOKEN_BUFFER_LENGTH_MIN_INCL,
+      hash & 0x3F, WUFFS_CBOR__DECODER_DST_TOKEN_BUFFER_LENGTH_MIN_INCL,
       TOK_BUFFER_ARRAY_SIZE);
-  uint64_t hash_50_bits = hash_56_bits >> 6;
+  hash = wuffs_base__u64__rotate_right(hash, 6);
 
-  uint64_t src_limit =
-      buffer_limit(hash_50_bits & 0x3F,
-                   WUFFS_CBOR__DECODER_SRC_IO_BUFFER_LENGTH_MIN_INCL, 4096);
+  uint64_t src_limit = buffer_limit(
+      hash & 0x3F, WUFFS_CBOR__DECODER_SRC_IO_BUFFER_LENGTH_MIN_INCL, 4096);
+  hash = wuffs_base__u64__rotate_right(hash, 6);
 
   // ----
 
@@ -407,7 +408,7 @@ fuzz(wuffs_base__io_buffer* full_src, uint64_t hash) {
   // The fuzz_complex implementation adds many more Wuffs API specific checks
   // (e.g. that the sum of the tokens' lengths do not exceed the input length).
   if ((hash & 0xFF) != 0xA5) {
-    return fuzz_complex(full_src, hash >> 8);
+    return fuzz_complex(full_src, wuffs_base__u64__rotate_right(hash, 8));
   }
   return fuzz_simple(full_src);
 }
diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 2da0d1baa8e1560caff9542fe068a8445b7ec6ba..1dcfb4458ff1cfe99103047f8de1cc28a5bc0568 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -61,7 +61,7 @@ It should print "PASS", amongst other information, and exit(0).
 #include "../fuzzlib/fuzzlib.c"
 #include "../fuzzlib/fuzzlib_image_decoder.c"
 
-void set_quirks(wuffs_gif__decoder* dec, uint64_t hash_55_bits) {
+void set_quirks(wuffs_gif__decoder* dec, uint64_t hash) {
   uint32_t quirks[] = {
       WUFFS_GIF__QUIRK_DELAY_NUM_DECODED_FRAMES,
       WUFFS_GIF__QUIRK_FIRST_FRAME_LOCAL_PALETTE_MEANS_BLACK_BACKGROUND,
@@ -75,8 +75,8 @@ void set_quirks(wuffs_gif__decoder* dec, uint64_t hash_55_bits) {
 
   uint32_t i;
   for (i = 0; quirks[i]; i++) {
-    uint64_t bit = 1 << (i % 55);
-    if (hash_55_bits & bit) {
+    uint64_t bit = 1 << (i & 63);
+    if (hash & bit) {
       wuffs_gif__decoder__set_quirk_enabled(dec, quirks[i], true);
     }
   }
@@ -88,13 +88,13 @@ fuzz(wuffs_base__io_buffer* src, uint64_t hash) {
   wuffs_base__status status = wuffs_gif__decoder__initialize(
       &dec, sizeof dec, WUFFS_VERSION,
       (hash & 1) ? WUFFS_INITIALIZE__LEAVE_INTERNAL_BUFFERS_UNINITIALIZED : 0);
-  hash >>= 1;
+  hash = wuffs_base__u64__rotate_right(hash, 1);
   uint64_t hash_8_bits = hash & 0xFF;
-  uint64_t hash_55_bits = hash >> 8;
+  hash = wuffs_base__u64__rotate_right(hash, 8);
   if (!wuffs_base__status__is_ok(&status)) {
     return wuffs_base__status__message(&status);
   }
-  set_quirks(&dec, hash_55_bits);
+  set_quirks(&dec, hash);
   return fuzz_image_decoder(
       src, hash_8_bits,
       wuffs_gif__decoder__upcast_as__wuffs_base__image_decoder(&dec));
diff --git a/fuzz/c/std/json_fuzzer.c b/fuzz/c/std/json_fuzzer.c
index 0c22678492f47a5f2746005522b7d075322c09f7..1a6a858a172ba64c4822fc03214e35144b017367 100644
--- a/fuzz/c/std/json_fuzzer.c
+++ b/fuzz/c/std/json_fuzzer.c
@@ -209,12 +209,13 @@ fuzz_one_token(wuffs_base__token t,
 }
 
 uint64_t  //
-buffer_limit(uint64_t hash_6_bits, uint64_t min, uint64_t max) {
+buffer_limit(uint64_t hash, uint64_t min, uint64_t max) {
+  hash &= 0x3F;
   uint64_t n;
-  if (hash_6_bits < 0x20) {
-    n = min + hash_6_bits;
+  if (hash < 0x20) {
+    n = min + hash;
   } else {
-    n = max - (0x3F - hash_6_bits);
+    n = max - (0x3F - hash);
   }
   if (n < min) {
     return min;
@@ -224,7 +225,7 @@ buffer_limit(uint64_t hash_6_bits, uint64_t min, uint64_t max) {
   return n;
 }
 
-void set_quirks(wuffs_json__decoder* dec, uint64_t hash_44_bits) {
+void set_quirks(wuffs_json__decoder* dec, uint64_t hash) {
   uint32_t quirks[] = {
       WUFFS_JSON__QUIRK_ALLOW_BACKSLASH_A,
       WUFFS_JSON__QUIRK_ALLOW_BACKSLASH_CAPITAL_U,
@@ -247,24 +248,23 @@ void set_quirks(wuffs_json__decoder* dec, uint64_t hash_44_bits) {
 
   uint32_t i;
   for (i = 0; quirks[i]; i++) {
-    uint64_t bit = 1 << (i % 44);
-    if (hash_44_bits & bit) {
+    uint64_t bit = 1 << (i & 63);
+    if (hash & bit) {
       wuffs_json__decoder__set_quirk_enabled(dec, quirks[i], true);
     }
   }
 }
 
 const char*  //
-fuzz_complex(wuffs_base__io_buffer* full_src, uint64_t hash_56_bits) {
+fuzz_complex(wuffs_base__io_buffer* full_src, uint64_t hash) {
   uint64_t tok_limit = buffer_limit(
-      hash_56_bits & 0x3F, WUFFS_JSON__DECODER_DST_TOKEN_BUFFER_LENGTH_MIN_INCL,
+      hash & 0x3F, WUFFS_JSON__DECODER_DST_TOKEN_BUFFER_LENGTH_MIN_INCL,
       TOK_BUFFER_ARRAY_SIZE);
-  uint64_t hash_50_bits = hash_56_bits >> 6;
+  hash = wuffs_base__u64__rotate_right(hash, 6);
 
-  uint64_t src_limit =
-      buffer_limit(hash_50_bits & 0x3F,
-                   WUFFS_JSON__DECODER_SRC_IO_BUFFER_LENGTH_MIN_INCL, 4096);
-  uint64_t hash_44_bits = hash_50_bits >> 6;
+  uint64_t src_limit = buffer_limit(
+      hash & 0x3F, WUFFS_JSON__DECODER_SRC_IO_BUFFER_LENGTH_MIN_INCL, 4096);
+  hash = wuffs_base__u64__rotate_right(hash, 6);
 
   // ----
 
@@ -275,7 +275,7 @@ fuzz_complex(wuffs_base__io_buffer* full_src, uint64_t hash_56_bits) {
   if (!wuffs_base__status__is_ok(&status)) {
     return wuffs_base__status__message(&status);
   }
-  set_quirks(&dec, hash_44_bits);
+  set_quirks(&dec, hash);
 
   wuffs_base__token tok_array[TOK_BUFFER_ARRAY_SIZE];
   wuffs_base__token_buffer tok = ((wuffs_base__token_buffer){
@@ -427,7 +427,7 @@ fuzz(wuffs_base__io_buffer* full_src, uint64_t hash) {
   // The fuzz_complex implementation adds many more Wuffs API specific checks
   // (e.g. that the sum of the tokens' lengths do not exceed the input length).
   if ((hash & 0xFF) != 0xA5) {
-    return fuzz_complex(full_src, hash >> 8);
+    return fuzz_complex(full_src, wuffs_base__u64__rotate_right(hash, 8));
   }
   return fuzz_simple(full_src);
 }
diff --git a/fuzz/c/std/png_fuzzer.c b/fuzz/c/std/png_fuzzer.c
index 1191fda0221a66afa153d593ad622a40bf7f30f7..d6cc049f20b1aedd6b46ba7dd04c52462cbfab96 100644
--- a/fuzz/c/std/png_fuzzer.c
+++ b/fuzz/c/std/png_fuzzer.c
@@ -70,7 +70,7 @@ fuzz(wuffs_base__io_buffer* src, uint64_t hash) {
   wuffs_base__status status = wuffs_png__decoder__initialize(
       &dec, sizeof dec, WUFFS_VERSION,
       (hash & 1) ? WUFFS_INITIALIZE__LEAVE_INTERNAL_BUFFERS_UNINITIALIZED : 0);
-  hash >>= 1;
+  hash = wuffs_base__u64__rotate_right(hash, 1);
   if (!wuffs_base__status__is_ok(&status)) {
     return wuffs_base__status__message(&status);
   }
