diff --git a/script/adler32-standalone.c b/script/adler32-standalone.c
index 805b2121096de92c0adc2414a5bd91353770d9cc..a2e22b963c52eb47c9ec52b5183b6f449ade63eb 100644
--- a/script/adler32-standalone.c
+++ b/script/adler32-standalone.c
@@ -175,9 +175,8 @@ main(int argc, char** argv) {
   // https://play.golang.org/p/IU2T58P00C
   const uint32_t expected_hash_of_1mib_of_zeroes = 0x00f00001UL;
 
-  int i;
   int num_bad = 0;
-  for (i = 0; i < num_reps; i++) {
+  for (int i = 0; i < num_reps; i++) {
     uint32_t actual_hash = calculate_hash(g_buffer, BUFFER_SIZE);
     if (!g_flags.no_check && (actual_hash != expected_hash_of_1mib_of_zeroes)) {
       num_bad++;
diff --git a/script/bench-c-deflate-fragmentation.c b/script/bench-c-deflate-fragmentation.c
index ba7e2733f3b9eb056997fc50096318d9b8fd57b1..73734d318415ee62ec470929ef3e36c6757c1959 100644
--- a/script/bench-c-deflate-fragmentation.c
+++ b/script/bench-c-deflate-fragmentation.c
@@ -330,8 +330,7 @@ decode(bool frag_dst, bool frag_idat) {
   struct timeval bench_start_tv;
   gettimeofday(&bench_start_tv, NULL);
 
-  int i;
-  for (i = 0; i < reps; i++) {
+  for (int i = 0; i < reps; i++) {
     const char* msg = decode_once(frag_dst, frag_idat);
     if (msg) {
       return msg;
@@ -399,8 +398,7 @@ main(int argc, char** argv) {
       "it, first\n"
       "# install Go, then run \"go get golang.org/x/perf/cmd/benchstat\".\n");
 
-  int i;
-  for (i = 0; i < 5; i++) {
+  for (int i = 0; i < 5; i++) {
     msg = decode(true, true);
     if (msg) {
       return fail(msg);
diff --git a/script/print-json-token-debug-format.c b/script/print-json-token-debug-format.c
index 866723b63ba8b953ebb6ff479bec3f45d5f73e37..e8dabd427ac51f11e4a70b2eea0824706ff16743 100644
--- a/script/print-json-token-debug-format.c
+++ b/script/print-json-token-debug-format.c
@@ -312,8 +312,7 @@ main1(int argc, char** argv) {
         WUFFS_JSON__QUIRK_REPLACE_INVALID_UNICODE,
         0,
     };
-    uint32_t i;
-    for (i = 0; quirks[i]; i++) {
+    for (uint32_t i = 0; quirks[i]; i++) {
       wuffs_base__token_decoder__set_quirk_enabled(g_dec, quirks[i], true);
     }
   }
