diff --git a/fuzz/c/fuzzlib/fuzzlib.c b/fuzz/c/fuzzlib/fuzzlib.c
index d4c72efbc352f06287134aa63e5d5c2d7fffc142..1fb18618a292f5b33f9b6581b574fc028fb3943c 100644
--- a/fuzz/c/fuzzlib/fuzzlib.c
+++ b/fuzz/c/fuzzlib/fuzzlib.c
@@ -337,8 +337,7 @@ main(int argc, char** argv) {
     fprintf(stderr, "FAIL: %s\n", z);
     return 1;
   }
-  int i;
-  for (i = 0; i < g_flags.remaining_argc; i++) {
+  for (int i = 0; i < g_flags.remaining_argc; i++) {
     int v = visit(g_flags.remaining_argv[i]);
     if (v) {
       return v;
diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index b3d014649cdce8354ba76618624a845419356f81..ba3118250d698d34ff53b74fb94c878b671f59cd 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -85,8 +85,7 @@ void set_quirks(wuffs_gif__decoder* dec, uint64_t hash) {
       0,
   };
 
-  uint32_t i;
-  for (i = 0; quirks[i]; i++) {
+  for (uint32_t i = 0; quirks[i]; i++) {
     uint64_t bit = 1 << (i & 63);
     if (hash & bit) {
       wuffs_gif__decoder__set_quirk_enabled(dec, quirks[i], true);
diff --git a/fuzz/c/std/json_fuzzer.cc b/fuzz/c/std/json_fuzzer.cc
index 4086cd5619ad61f9a047d3963d12d5036f36b227..5cb0e226c583c5dd6b600e6f0b062ad403b0e733 100644
--- a/fuzz/c/std/json_fuzzer.cc
+++ b/fuzz/c/std/json_fuzzer.cc
@@ -261,8 +261,7 @@ uint32_t g_quirks[] = {
 };
 
 void set_quirks(wuffs_json__decoder* dec, uint64_t hash) {
-  uint32_t i;
-  for (i = 0; g_quirks[i]; i++) {
+  for (uint32_t i = 0; g_quirks[i]; i++) {
     uint64_t bit = 1 << (i & 63);
     if (hash & bit) {
       wuffs_json__decoder__set_quirk_enabled(dec, g_quirks[i], true);
