diff --git a/fuzz/c/fuzzlib/fuzzlib.c b/fuzz/c/fuzzlib/fuzzlib.c
index 0af1623ebf7ee54c409e05abaee3ec9a74bb16ef..de34988f2281ae69c5d1611a22e45d44c634a1d7 100644
--- a/fuzz/c/fuzzlib/fuzzlib.c
+++ b/fuzz/c/fuzzlib/fuzzlib.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #include <inttypes.h>
 #include <stdio.h>
diff --git a/fuzz/c/fuzzlib/fuzzlib_image_decoder.c b/fuzz/c/fuzzlib/fuzzlib_image_decoder.c
index 980b9549eb4daae3be6da1cc753b62a722cdc3e1..38f14cf3ca682ed6e5ef2788edd66d49ae3556c1 100644
--- a/fuzz/c/fuzzlib/fuzzlib_image_decoder.c
+++ b/fuzz/c/fuzzlib/fuzzlib_image_decoder.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #ifndef WUFFS_INCLUDE_GUARD
 #error "Wuffs' .h files need to be included before this file"
diff --git a/fuzz/c/std/bmp_fuzzer.c b/fuzz/c/std/bmp_fuzzer.c
index a0971b80cff1b722e94f44455c48f1aae9fa4dc8..19691d426ece64517735fe42651d18ceee35b98b 100644
--- a/fuzz/c/std/bmp_fuzzer.c
+++ b/fuzz/c/std/bmp_fuzzer.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/fuzz/c/std/bzip2_fuzzer.c b/fuzz/c/std/bzip2_fuzzer.c
index 17a785a21bada181d36c241d700f53d37c67d313..875327f5dfbf758772a8bb6a8ccc1121a0a6be63 100644
--- a/fuzz/c/std/bzip2_fuzzer.c
+++ b/fuzz/c/std/bzip2_fuzzer.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/fuzz/c/std/cbor_fuzzer.c b/fuzz/c/std/cbor_fuzzer.c
index 80dc8815461b5bacc0226fde764ab7a7167b4e93..be7fc5b40422ea7355c36e0e01733267ec2518e4 100644
--- a/fuzz/c/std/cbor_fuzzer.c
+++ b/fuzz/c/std/cbor_fuzzer.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 928033112b2eacefb48548bb3436eae724d3cca9..0ee4daa0ed5232a8972d8985d683e90efa3078f0 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/fuzz/c/std/jpeg_fuzzer.c b/fuzz/c/std/jpeg_fuzzer.c
index 67763847414079986ae8412e8984045b25646564..62d7c07632def0291a2c9c8b425181b005d190b7 100644
--- a/fuzz/c/std/jpeg_fuzzer.c
+++ b/fuzz/c/std/jpeg_fuzzer.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/fuzz/c/std/json_fuzzer.cc b/fuzz/c/std/json_fuzzer.cc
index 42dbdd706b6cbbeb2723001cd629f20943cdc670..102aa8af7043d0f64d4012b6facdc229403a8598 100644
--- a/fuzz/c/std/json_fuzzer.cc
+++ b/fuzz/c/std/json_fuzzer.cc
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/fuzz/c/std/pixel_swizzler_fuzzer.c b/fuzz/c/std/pixel_swizzler_fuzzer.c
index 6d1d15457b52a565f4ae293ad14288d869494160..a57513a7e48c33bb85f6de3316423e0a28069b34 100644
--- a/fuzz/c/std/pixel_swizzler_fuzzer.c
+++ b/fuzz/c/std/pixel_swizzler_fuzzer.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/fuzz/c/std/png_fuzzer.c b/fuzz/c/std/png_fuzzer.c
index defa14e0094c761d91184bf32206c57e401d7d90..2a87b7620417f31ac30d7840c6bc21ba070589aa 100644
--- a/fuzz/c/std/png_fuzzer.c
+++ b/fuzz/c/std/png_fuzzer.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/fuzz/c/std/tga_fuzzer.c b/fuzz/c/std/tga_fuzzer.c
index 841d359e6c684eb5d7b71f2c7e44c2bd7ae773cf..6fbb0b5551cb5d80139a9a6dbbe6b1abd85fac0e 100644
--- a/fuzz/c/std/tga_fuzzer.c
+++ b/fuzz/c/std/tga_fuzzer.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index 1b69d6ed23b0a3df061d11e8413bb7fde3fb5029..3209f746c982777e29ee0f898ca7455e95432263 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
