diff --git a/hello-wuffs-c/main.c b/hello-wuffs-c/main.c
index 1cc41821db9d773c1eef3f5c489ca5b98da45978..7f357384e8c095ccfe4baffbc4a8aa4d974b8f73 100644
--- a/hello-wuffs-c/main.c
+++ b/hello-wuffs-c/main.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #include <inttypes.h>
 #include <stdio.h>
diff --git a/hello-wuffs-c/naive-parse.c b/hello-wuffs-c/naive-parse.c
index c6de641060abe22bcdd19c0082c5632a94f7c238..2ddb575d4bf81d9d750b8090b182a61fe2620122 100644
--- a/hello-wuffs-c/naive-parse.c
+++ b/hello-wuffs-c/naive-parse.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #include <inttypes.h>
 #include <stdlib.h>
diff --git a/hello-wuffs-c/parse.wuffs b/hello-wuffs-c/parse.wuffs
index 0e15fa32ad0906ac324a1d1dceb59d560015f45e..2e6ba6f5b9af42ef979a72d4ed27a8a8c28a1c0d 100644
--- a/hello-wuffs-c/parse.wuffs
+++ b/hello-wuffs-c/parse.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri status "#not a digit"
 pri status "#too large"
diff --git a/hello-wuffs-c/wuffs-base.c b/hello-wuffs-c/wuffs-base.c
index 828c873a2883ac25df56071e9cc43eb4d59c37f1..65fe9bfb8c7fa9f05ce64800457f8baf61e15e1d 100644
--- a/hello-wuffs-c/wuffs-base.c
+++ b/hello-wuffs-c/wuffs-base.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/hello-wuffs-c/wuffs-parse.c b/hello-wuffs-c/wuffs-parse.c
index f9afd6d45a64b4f1c2705275eec700acf8705918..adbec6268e59dcd073217ee0bee71a1aa35e3985 100644
--- a/hello-wuffs-c/wuffs-parse.c
+++ b/hello-wuffs-c/wuffs-parse.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #include <stdio.h>
 
diff --git a/internal/cgen/auxiliary/base.cc b/internal/cgen/auxiliary/base.cc
index 8c048e664c414b4de1fc628a82e8ff86b53bbf26..3ec7572995ecd5d34233f707bbd726ae1a622bf6 100644
--- a/internal/cgen/auxiliary/base.cc
+++ b/internal/cgen/auxiliary/base.cc
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Auxiliary - Base
 
diff --git a/internal/cgen/auxiliary/base.hh b/internal/cgen/auxiliary/base.hh
index f1b96b7dc2decd76b701b7632c134ed791490261..f0b2447c7e7f2b8c374aced680b5c3fcd423796f 100644
--- a/internal/cgen/auxiliary/base.hh
+++ b/internal/cgen/auxiliary/base.hh
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Auxiliary - Base
 
diff --git a/internal/cgen/auxiliary/cbor.cc b/internal/cgen/auxiliary/cbor.cc
index 944753c236dbd190b285e804fcb831e9da07cafb..fb55a5d71bb89f97338f978a505f8be623d20e32 100644
--- a/internal/cgen/auxiliary/cbor.cc
+++ b/internal/cgen/auxiliary/cbor.cc
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Auxiliary - CBOR
 
diff --git a/internal/cgen/auxiliary/cbor.hh b/internal/cgen/auxiliary/cbor.hh
index 1a1096511616771b4ba2e3e9e81ecca086b80b2c..c45e11490ce09d18769882f0681f1896ee510bb6 100644
--- a/internal/cgen/auxiliary/cbor.hh
+++ b/internal/cgen/auxiliary/cbor.hh
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Auxiliary - CBOR
 
diff --git a/internal/cgen/auxiliary/image.cc b/internal/cgen/auxiliary/image.cc
index 1199f6dee9bf9b6d9eed22430bf12f003c08b065..a723ecaa692d0a75176f08b0f84fb04ef8df891f 100644
--- a/internal/cgen/auxiliary/image.cc
+++ b/internal/cgen/auxiliary/image.cc
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Auxiliary - Image
 
diff --git a/internal/cgen/auxiliary/image.hh b/internal/cgen/auxiliary/image.hh
index 90327516ef6bc609087cf69008224f4bc34aabb6..efe3d6c2e175e937c8be265a0565f48697247948 100644
--- a/internal/cgen/auxiliary/image.hh
+++ b/internal/cgen/auxiliary/image.hh
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Auxiliary - Image
 
diff --git a/internal/cgen/auxiliary/json.cc b/internal/cgen/auxiliary/json.cc
index 1c2d5ec866c5e4f5796f2f20bed1ac47a7e77581..ffef6124c2afed840521f1479463ad4296a100b1 100644
--- a/internal/cgen/auxiliary/json.cc
+++ b/internal/cgen/auxiliary/json.cc
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Auxiliary - JSON
 
diff --git a/internal/cgen/auxiliary/json.hh b/internal/cgen/auxiliary/json.hh
index bc5fd9baffd8f4dbbcfbc495105fbca6b927e5cd..c0ce43f8d4b1e0d217405e36665fcedf265b3359 100644
--- a/internal/cgen/auxiliary/json.hh
+++ b/internal/cgen/auxiliary/json.hh
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Auxiliary - JSON
 
diff --git a/internal/cgen/base/all-impl.c b/internal/cgen/base/all-impl.c
index 2f63c61c4c490cc279a4bf2f3fbcf36c6b9adf07..4a9ed5326c535a5775d872c9594acb34cd0ff3ac 100644
--- a/internal/cgen/base/all-impl.c
+++ b/internal/cgen/base/all-impl.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #ifndef WUFFS_INCLUDE_GUARD__BASE
 #define WUFFS_INCLUDE_GUARD__BASE
diff --git a/internal/cgen/base/floatconv-submodule-code.c b/internal/cgen/base/floatconv-submodule-code.c
index 460dbd7601a1e1c90e9cfb2c3b92ec92af174110..672f65e1ad2f4fa07d08b6ef68aa2b9a64edb161 100644
--- a/internal/cgen/base/floatconv-submodule-code.c
+++ b/internal/cgen/base/floatconv-submodule-code.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- IEEE 754 Floating Point
 
diff --git a/internal/cgen/base/floatconv-submodule-data.c b/internal/cgen/base/floatconv-submodule-data.c
index 465f3d72cbfa464f3832870bc3e18b0abfde1383..4f6d7bd9a604b299059a61a1d52e42b0273e9bff 100644
--- a/internal/cgen/base/floatconv-submodule-data.c
+++ b/internal/cgen/base/floatconv-submodule-data.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- IEEE 754 Floating Point
 
diff --git a/internal/cgen/base/fundamental-private.h b/internal/cgen/base/fundamental-private.h
index 7c9cd4e945689762165dd6287241424597745e11..f1aad9ed94b5f6961bf80580f7be700874556087 100644
--- a/internal/cgen/base/fundamental-private.h
+++ b/internal/cgen/base/fundamental-private.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Fundamentals
 
diff --git a/internal/cgen/base/fundamental-public.h b/internal/cgen/base/fundamental-public.h
index 84090df5f4a960f750069560bc0d45931cd35f34..4fdc5bb40fa8f53500d761ced077088190195788 100644
--- a/internal/cgen/base/fundamental-public.h
+++ b/internal/cgen/base/fundamental-public.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Version
 
diff --git a/internal/cgen/base/image-private.h b/internal/cgen/base/image-private.h
index f96d0951078a18e42b8895f27f2bd07abe60ca5d..8779729328d2d03b3c8857e79148fa74f7c2f01b 100644
--- a/internal/cgen/base/image-private.h
+++ b/internal/cgen/base/image-private.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Images
 
diff --git a/internal/cgen/base/image-public.h b/internal/cgen/base/image-public.h
index 22c0c5f1572c8f2e2e3c2c9d2e7553f6362ad030..ce16e1f0e0cc260deca6f6f150c057fe6085614e 100644
--- a/internal/cgen/base/image-public.h
+++ b/internal/cgen/base/image-public.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Images
 
diff --git a/internal/cgen/base/intconv-submodule.c b/internal/cgen/base/intconv-submodule.c
index 86191eb739eb2d0fdb80ee37201c18eeb34ef618..d9c4b9f262c8012bf8713ff4c8b4e02071f20ae4 100644
--- a/internal/cgen/base/intconv-submodule.c
+++ b/internal/cgen/base/intconv-submodule.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Integer
 
diff --git a/internal/cgen/base/io-private.h b/internal/cgen/base/io-private.h
index 20ee0b3a2c1b2c262070dda8d5666a9800ff59df..4b879a3f8716d21a4b30dd34ca1749523ecf126d 100644
--- a/internal/cgen/base/io-private.h
+++ b/internal/cgen/base/io-private.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- I/O
 
diff --git a/internal/cgen/base/io-public.h b/internal/cgen/base/io-public.h
index 07715ff964d083a42dcd0f9467a9e9bdaf3fcc29..5513795904338d2e6557ba3f5c02273daf5f2145 100644
--- a/internal/cgen/base/io-public.h
+++ b/internal/cgen/base/io-public.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- I/O
 //
diff --git a/internal/cgen/base/magic-submodule.c b/internal/cgen/base/magic-submodule.c
index cd1d96dc33a83bcaec0097c340e2c018d893d466..76184604145c1ff04334804e014411c9b762588e 100644
--- a/internal/cgen/base/magic-submodule.c
+++ b/internal/cgen/base/magic-submodule.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Magic Numbers
 
diff --git a/internal/cgen/base/memory-private.h b/internal/cgen/base/memory-private.h
index 4835c0041f89a46346c2450cee63e67b5f02c714..29d64bbd98d80a80d32bffcc697087009d6dbca7 100644
--- a/internal/cgen/base/memory-private.h
+++ b/internal/cgen/base/memory-private.h
@@ -5,5 +5,7 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Memory Allocation
diff --git a/internal/cgen/base/memory-public.h b/internal/cgen/base/memory-public.h
index 9eff2d9c940b3f151c79534bd150d6971c65a113..bef9cf8d652139281e176b90640d798cd250a714 100644
--- a/internal/cgen/base/memory-public.h
+++ b/internal/cgen/base/memory-public.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Memory Allocation
 
diff --git a/internal/cgen/base/pixconv-submodule-regular.c b/internal/cgen/base/pixconv-submodule-regular.c
index e511c576064b9cc9a90300dbb88bffedc2b9b889..662fc11575c892cd60c3eebcac1a3413191b2859 100644
--- a/internal/cgen/base/pixconv-submodule-regular.c
+++ b/internal/cgen/base/pixconv-submodule-regular.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Pixel Swizzler
 
diff --git a/internal/cgen/base/pixconv-submodule-x86-avx2.c b/internal/cgen/base/pixconv-submodule-x86-avx2.c
index ba5947c0bf9ec71708b41e3b0f25509688a5ae84..c0b31d3e68974a1eba3fbfd71697f9fefbdc2369 100644
--- a/internal/cgen/base/pixconv-submodule-x86-avx2.c
+++ b/internal/cgen/base/pixconv-submodule-x86-avx2.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/internal/cgen/base/pixconv-submodule-ycck.c b/internal/cgen/base/pixconv-submodule-ycck.c
index 66e2e1bd2b076be85be80fb047b4f956efcacefc..2a63680336d9565b729e60ebd45986eef985c728 100644
--- a/internal/cgen/base/pixconv-submodule-ycck.c
+++ b/internal/cgen/base/pixconv-submodule-ycck.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/internal/cgen/base/range-private.h b/internal/cgen/base/range-private.h
index 775374ac582b8fb7a4d988ee4d87ed902ebbe85d..12da4e974795ebcb6dfe86b6b897e406f7528a76 100644
--- a/internal/cgen/base/range-private.h
+++ b/internal/cgen/base/range-private.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Ranges and Rects
 
diff --git a/internal/cgen/base/range-public.h b/internal/cgen/base/range-public.h
index 84ed5eceb3ef311ef7a4328e025ad715714c8fe4..06facfac3c424e551c3daf1826b400ee7a6dd50d 100644
--- a/internal/cgen/base/range-public.h
+++ b/internal/cgen/base/range-public.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Ranges and Rects
 
diff --git a/internal/cgen/base/strconv-private.h b/internal/cgen/base/strconv-private.h
index a91892545f850b1eb200b02c4ecd230708261f7e..05d22aafd79d487d6797f6defec9c1ae0d8b9f11 100644
--- a/internal/cgen/base/strconv-private.h
+++ b/internal/cgen/base/strconv-private.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- String Conversions
 
diff --git a/internal/cgen/base/strconv-public.h b/internal/cgen/base/strconv-public.h
index 0b2afe5b46a5eb7d91de4982275fe1fdbb1e3d4c..1f5efb1851c9decdcdb50538f4f8370bac6564dc 100644
--- a/internal/cgen/base/strconv-public.h
+++ b/internal/cgen/base/strconv-public.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- String Conversions
 
diff --git a/internal/cgen/base/token-private.h b/internal/cgen/base/token-private.h
index 7740402fd3780ef6779a49272c04c4e447b9f98c..c8a8b2d9463ff7dfcdca4c5eb43d8be62a1b0a76 100644
--- a/internal/cgen/base/token-private.h
+++ b/internal/cgen/base/token-private.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Tokens
 
diff --git a/internal/cgen/base/token-public.h b/internal/cgen/base/token-public.h
index e1846f79769d336f2c78c944689d763a23dbc090..acb3eb7adef190c9174bb0a913261ac3cd892394 100644
--- a/internal/cgen/base/token-public.h
+++ b/internal/cgen/base/token-public.h
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Tokens
 
diff --git a/internal/cgen/base/utf8-submodule.c b/internal/cgen/base/utf8-submodule.c
index 1d01d8dc609c73d88aa1c162daf14f124d64399b..e49854bde5f9af91824b9f03707ab612404d1900 100644
--- a/internal/cgen/base/utf8-submodule.c
+++ b/internal/cgen/base/utf8-submodule.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ---------------- Unicode and UTF-8
 
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index 57c64814abb145961037c22b32ab627d46ab4b1f..dcbe66f40bed34c278eb497042714dcbd3e4aa80 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -42,6 +42,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #include <stdbool.h>
 #include <stdint.h>
diff --git a/release/c/wuffs-v0.4.c b/release/c/wuffs-v0.4.c
index 5e8ec2fb4447cb7d3b492ab78991e09bc177dc91..751b66703a2ed0d323ac7bfbb5260b9c2c90741b 100644
--- a/release/c/wuffs-v0.4.c
+++ b/release/c/wuffs-v0.4.c
@@ -42,6 +42,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #include <stdbool.h>
 #include <stdint.h>
diff --git a/script/adler32-standalone.c b/script/adler32-standalone.c
index 55433d734fa7819a2846a0a813d25bb4e520bf77..a3d6c02da157a5902fed47a08b9229aae758dc86 100644
--- a/script/adler32-standalone.c
+++ b/script/adler32-standalone.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/script/bench-c-deflate-fragmentation.c b/script/bench-c-deflate-fragmentation.c
index 1eee0e5f80214c941751333db981bac901ecc19f..652ac7292ae072efae7f837da64d3c795f1914cc 100644
--- a/script/bench-c-deflate-fragmentation.c
+++ b/script/bench-c-deflate-fragmentation.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/script/manual-test-parse-number-f64.cc b/script/manual-test-parse-number-f64.cc
index eb3cad46ab13a84642be2387deff3e59750d1069..a3e38b5e8da1392d262430212ad962514b9c60b9 100644
--- a/script/manual-test-parse-number-f64.cc
+++ b/script/manual-test-parse-number-f64.cc
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/script/manual-test-truncated-input.cc b/script/manual-test-truncated-input.cc
index 0362aa8b3bd035144d08dec0bc5f3668a447ca74..6022490f9be689e1ed58d91524ee1871e0ce6e81 100644
--- a/script/manual-test-truncated-input.cc
+++ b/script/manual-test-truncated-input.cc
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/script/mmap-ring-buffer.c b/script/mmap-ring-buffer.c
index 3ac9a3bdcce47ad5c1890ae26e303aedc4b8bcae..52a2cb775bcf7b100972a73d3103b7fdfefc4f70 100644
--- a/script/mmap-ring-buffer.c
+++ b/script/mmap-ring-buffer.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/script/print-average-pixel.cc b/script/print-average-pixel.cc
index ae26d9f270f284695f53e799ddee9399e8ae5a14..69cfa6a27812b90214d0385c308ae82dfa0e991e 100644
--- a/script/print-average-pixel.cc
+++ b/script/print-average-pixel.cc
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/script/print-image-metadata.cc b/script/print-image-metadata.cc
index 0518dfadd093171dee0ff2e39b38388aadeb5839..03a9b6ef8f55a3b5c5598aaa3b2bd5daa0117b0f 100644
--- a/script/print-image-metadata.cc
+++ b/script/print-image-metadata.cc
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/script/print-json-token-debug-format.c b/script/print-json-token-debug-format.c
index fe1a244083225ec7d0591fe4cddfee763c176de8..522e9847bdc6b9c2b76f363b7403869cbc1ed8bb 100644
--- a/script/print-json-token-debug-format.c
+++ b/script/print-json-token-debug-format.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/script/process-json-numbers.c b/script/process-json-numbers.c
index 55c9d4457b2e1a520da723a4f5428e0502603156..782d34922cd38bc4e48891d6a5a912504a60bae6 100644
--- a/script/process-json-numbers.c
+++ b/script/process-json-numbers.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/script/wuffs-deflate-decoder-decode-huffman.c b/script/wuffs-deflate-decoder-decode-huffman.c
index 60d33caac393ebd94c51e71a8ecbc6719db15a27..99ecc19f09b3ddb4da66d8a02309fde07a55c45d 100644
--- a/script/wuffs-deflate-decoder-decode-huffman.c
+++ b/script/wuffs-deflate-decoder-decode-huffman.c
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----------------
 
diff --git a/std/adler32/common_adler32.wuffs b/std/adler32/common_adler32.wuffs
index 37bb7151945059fe98ff4e739bb0fb3f18140aa7..f3b8a7fc4bea9fbf3a52c2c80adae69b19c03d15 100644
--- a/std/adler32/common_adler32.wuffs
+++ b/std/adler32/common_adler32.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // TODO: drop the '?' but still generate wuffs_adler32__hasher__initialize?
 pub struct hasher? implements base.hasher_u32(
diff --git a/std/adler32/common_up_arm_neon.wuffs b/std/adler32/common_up_arm_neon.wuffs
index cf95c2319aef5783659d66ff33c7c3337ec92e88..7e0b9fcca6ddf22e2c3bc30abda09de228b55bc3 100644
--- a/std/adler32/common_up_arm_neon.wuffs
+++ b/std/adler32/common_up_arm_neon.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func hasher.up_arm_neon!(x: roslice base.u8),
         choose cpu_arch >= arm_neon,
diff --git a/std/adler32/common_up_x86_sse42.wuffs b/std/adler32/common_up_x86_sse42.wuffs
index 57686df3b17799550a9536f1567059c8254e70a6..e80670256b3e67b2b8cb948ea0aaa30017c9159b 100644
--- a/std/adler32/common_up_x86_sse42.wuffs
+++ b/std/adler32/common_up_x86_sse42.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func hasher.up_x86_sse42!(x: roslice base.u8),
         choose cpu_arch >= x86_sse42,
diff --git a/std/bmp/decode_bmp.wuffs b/std/bmp/decode_bmp.wuffs
index 799a9424a7c5f37cb2b6294ce34d20b5f6e1ee52..3995a2f7eefb7aea2b8f473491d61843b7371769 100644
--- a/std/bmp/decode_bmp.wuffs
+++ b/std/bmp/decode_bmp.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad header"
 pub status "#bad RLE compression"
diff --git a/std/bzip2/decode_bzip2.wuffs b/std/bzip2/decode_bzip2.wuffs
index c93fc7c107e8b9dc9ee2e5e2b76d2c8bc7056dc5..30cc7e678e49e53ae4f24c2b23cff3f3be9dfa5e 100644
--- a/std/bzip2/decode_bzip2.wuffs
+++ b/std/bzip2/decode_bzip2.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad Huffman code (over-subscribed)"
 pub status "#bad Huffman code (under-subscribed)"
diff --git a/std/bzip2/decode_flush_fast.wuffs b/std/bzip2/decode_flush_fast.wuffs
index f8204c177a9515a648f4ea30f28c7e446df9e9d4..777a462fce0c1b3f1d710f262520b832260a774c 100644
--- a/std/bzip2/decode_flush_fast.wuffs
+++ b/std/bzip2/decode_flush_fast.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.flush_fast!(dst: base.io_writer) {
     var flush_pointer       : base.u32[..= 1_048575]
diff --git a/std/bzip2/decode_flush_slow.wuffs b/std/bzip2/decode_flush_slow.wuffs
index 3621e5800783e386865aae22a78cb69981bdcb1a..be6b229867eedf5a81659cbbebd12ac436c47fa9 100644
--- a/std/bzip2/decode_flush_slow.wuffs
+++ b/std/bzip2/decode_flush_slow.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.flush_slow?(dst: base.io_writer) {
     var flush_pointer       : base.u32[..= 1_048575]
diff --git a/std/bzip2/decode_huffman_fast.wuffs b/std/bzip2/decode_huffman_fast.wuffs
index b307bfd3a08442a96684505aa7ffc26315971e3a..3c7264cabfd87f392b808ee43d8799819363fc02 100644
--- a/std/bzip2/decode_huffman_fast.wuffs
+++ b/std/bzip2/decode_huffman_fast.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.decode_huffman_fast!(src: base.io_reader) base.status {
     var bits       : base.u32
diff --git a/std/bzip2/decode_huffman_slow.wuffs b/std/bzip2/decode_huffman_slow.wuffs
index f8da8f3e06e3cebb976fb805f96a08dc29ca18cb..16d384b85aa90754655c546030bd53f2597a1dd2 100644
--- a/std/bzip2/decode_huffman_slow.wuffs
+++ b/std/bzip2/decode_huffman_slow.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.decode_huffman_slow?(src: base.io_reader) {
     var c          : base.u8
diff --git a/std/cbor/decode_cbor.wuffs b/std/cbor/decode_cbor.wuffs
index 8d81b3b1fda2fb673afe9806895acde97fcd353b..660f2a183f5ff860cc7d6473c4cbdc152832823c 100644
--- a/std/cbor/decode_cbor.wuffs
+++ b/std/cbor/decode_cbor.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad input"
 pub status "#unsupported recursion depth"
diff --git a/std/crc32/common_crc32.wuffs b/std/crc32/common_crc32.wuffs
index dc19d3a4012c209b609befd4eeb957b670a56d80..13c1d8c4d756661de6155dfe1d9a5f90ded41ac5 100644
--- a/std/crc32/common_crc32.wuffs
+++ b/std/crc32/common_crc32.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // TODO: drop the '?' but still generate wuffs_crc32__ieee_hasher__initialize?
 pub struct ieee_hasher? implements base.hasher_u32(
diff --git a/std/crc32/common_up_arm_crc32.wuffs b/std/crc32/common_up_arm_crc32.wuffs
index c5f2fa92807b0a3dffd63faccbb0847e4c9e62ee..9d72ae5c037b2ff479c86e6a5154834370344192 100644
--- a/std/crc32/common_up_arm_crc32.wuffs
+++ b/std/crc32/common_up_arm_crc32.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func ieee_hasher.up_arm_crc32!(x: roslice base.u8),
         choose cpu_arch >= arm_crc32,
diff --git a/std/crc32/common_up_x86_avx2.wuffs b/std/crc32/common_up_x86_avx2.wuffs
index d35d03899f95f4cb55f3c7b0554efa6a8fb6a3d0..94ead63305bb087821e5ab1ca0e85a1ea3596f31 100644
--- a/std/crc32/common_up_x86_avx2.wuffs
+++ b/std/crc32/common_up_x86_avx2.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/crc32/common_up_x86_sse42.wuffs b/std/crc32/common_up_x86_sse42.wuffs
index a51379b1adeca1d4b5e77fc678a988890a23ed4b..9e26b906649350b2b7d764e6c1e84958b24e40e4 100644
--- a/std/crc32/common_up_x86_sse42.wuffs
+++ b/std/crc32/common_up_x86_sse42.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/deflate/common_consts.wuffs b/std/deflate/common_consts.wuffs
index ae9270c4ff17570ce4af596928fa16323bb3e68d..65b44dad670400d58d30050ce169b6644a2020c4 100644
--- a/std/deflate/common_consts.wuffs
+++ b/std/deflate/common_consts.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // CODE_ORDER is defined in the RFC section 3.2.7.
 pri const CODE_ORDER : roarray[19] base.u8[..= 18] = [
diff --git a/std/deflate/decode_deflate.wuffs b/std/deflate/decode_deflate.wuffs
index eb1fe5930bc0b293f9f6653b8cdc7f96d2097215..6ef92d93d6b1add1540612b499ed36db2f03bd48 100644
--- a/std/deflate/decode_deflate.wuffs
+++ b/std/deflate/decode_deflate.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad Huffman code (over-subscribed)"
 pub status "#bad Huffman code (under-subscribed)"
diff --git a/std/deflate/decode_huffman_bmi2.wuffs b/std/deflate/decode_huffman_bmi2.wuffs
index d9a71f212395f445f5095ebd04243d81f08c21c3..8bdb9ca6936ea30de362823ec9c6b0102d169989 100644
--- a/std/deflate/decode_huffman_bmi2.wuffs
+++ b/std/deflate/decode_huffman_bmi2.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // decode_huffman_bmi2 is exactly the same as decode_huffman_fast64 except for
 // the "choose cpu_arch >= x86_bmi2". Unsurprisingly, having Bit Manipulation
diff --git a/std/deflate/decode_huffman_fast32.wuffs b/std/deflate/decode_huffman_fast32.wuffs
index bd611a0a6cc1f2db4bf479185a4910752c3aaae6..04fab25ad490f6b2f2e6a68348eca8276b0b4660 100644
--- a/std/deflate/decode_huffman_fast32.wuffs
+++ b/std/deflate/decode_huffman_fast32.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // TODO: describe how the xxx_fastxx version differs from the xxx_slow one, the
 // assumptions that xxx_fastxx makes, and how that makes it fast.
diff --git a/std/deflate/decode_huffman_fast64.wuffs b/std/deflate/decode_huffman_fast64.wuffs
index 5d33871b6f85eef2a7765621db5f7260fe9eeb21..66222252ebca33d5dc5bb1f060d7ae64d1789523 100644
--- a/std/deflate/decode_huffman_fast64.wuffs
+++ b/std/deflate/decode_huffman_fast64.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // TODO: describe how the xxx_fastxx version differs from the xxx_slow one, the
 // assumptions that xxx_fastxx makes, and how that makes it fast.
diff --git a/std/deflate/decode_huffman_slow.wuffs b/std/deflate/decode_huffman_slow.wuffs
index e27ca188de838639c2760f892a3532eb0564de8b..176a65b062d948d147c32c4baa7d62ba65bdc6bf 100644
--- a/std/deflate/decode_huffman_slow.wuffs
+++ b/std/deflate/decode_huffman_slow.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.decode_huffman_slow?(dst: base.io_writer, src: base.io_reader) {
     var bits               : base.u32
diff --git a/std/gif/common_consts.wuffs b/std/gif/common_consts.wuffs
index 812e2c3df185c54217c08dc334328a87f2a21449..a80015a6ce9d06ead2f1d687e00172415cfb185c 100644
--- a/std/gif/common_consts.wuffs
+++ b/std/gif/common_consts.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad LZW code"
 pub status "#bad extension label"
diff --git a/std/gif/decode_gif.wuffs b/std/gif/decode_gif.wuffs
index 63fc84229458ddd8e216fbea4099179a5d2cbd39..83782f8de8bbd375b854940326300afcaace4db5 100644
--- a/std/gif/decode_gif.wuffs
+++ b/std/gif/decode_gif.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub struct decoder? implements base.image_decoder(
         // These upper bounds should be 0xFFFF, on a strict reading of the GIF
diff --git a/std/gif/decode_lzw.wuffs b/std/gif/decode_lzw.wuffs
index e0c641fac6b683873b1d07b67db011941b250fc2..05b81da314927723483839c4c7165021a1c5878d 100644
--- a/std/gif/decode_lzw.wuffs
+++ b/std/gif/decode_lzw.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/gif/decode_quirks.wuffs b/std/gif/decode_quirks.wuffs
index d58b0fff5a90db0d43e33c00cb97b0674c70cbdb..c9720b83bab6288ff0c5f3e358a47d812502ee11 100644
--- a/std/gif/decode_quirks.wuffs
+++ b/std/gif/decode_quirks.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/gzip/decode_gzip.wuffs b/std/gzip/decode_gzip.wuffs
index b2c958411703976d4acb115dcb4c4bede8d6c33f..48ae928817ce6e1cc61e77bd3b5dee2e2213afff 100644
--- a/std/gzip/decode_gzip.wuffs
+++ b/std/gzip/decode_gzip.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 use "std/crc32"
 use "std/deflate"
diff --git a/std/jpeg/common_consts.wuffs b/std/jpeg/common_consts.wuffs
index dad83091bf4fcc5641f4fd9114d422d6c00c80be..b6db49c147eceb25c2a2aa774ee98883f825002d 100644
--- a/std/jpeg/common_consts.wuffs
+++ b/std/jpeg/common_consts.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/jpeg/decode_idct_default.wuffs b/std/jpeg/decode_idct_default.wuffs
index b866d317c9b6cc4c4610be054035c961e822be2c..59283e2c6c65185213b3b470f1c2bdd3af91b0af 100644
--- a/std/jpeg/decode_idct_default.wuffs
+++ b/std/jpeg/decode_idct_default.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.decode_idct!(dst_buffer: slice base.u8, dst_stride: base.u64, q: base.u32[..= 3]),
         choosy,
diff --git a/std/jpeg/decode_idct_x86_avx2.wuffs b/std/jpeg/decode_idct_x86_avx2.wuffs
index dc5789d3182878468a1b2cc40313005cc849e1f8..9c14eb490457db2124de673c5cb444310b48c82b 100644
--- a/std/jpeg/decode_idct_x86_avx2.wuffs
+++ b/std/jpeg/decode_idct_x86_avx2.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.decode_idct_x86_avx2!(dst_buffer: slice base.u8, dst_stride: base.u64, q: base.u32[..= 3]),
         choose cpu_arch >= x86_avx2,
diff --git a/std/jpeg/decode_jpeg.wuffs b/std/jpeg/decode_jpeg.wuffs
index 40175a6f639934bb0e32376da3e9b2232faa97a6..8c2eb584afafda287579501f8cee07e97cf9baaf 100644
--- a/std/jpeg/decode_jpeg.wuffs
+++ b/std/jpeg/decode_jpeg.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad DHT marker"
 pub status "#bad DQT marker"
diff --git a/std/jpeg/decode_load_smooth.wuffs b/std/jpeg/decode_load_smooth.wuffs
index 75712658c039aa7d3d47c77973b30fa92ca044b9..c42d642c6434fe2f0443051da5eafd8b87547fa0 100644
--- a/std/jpeg/decode_load_smooth.wuffs
+++ b/std/jpeg/decode_load_smooth.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // ----
 
diff --git a/std/jpeg/decode_mcu_default.wuffs b/std/jpeg/decode_mcu_default.wuffs
index 75e1931c531485de375ec400e79e9ec5c7c22e00..397cc741e6bbd4e419d916eb43f2d8c2d6a368c1 100644
--- a/std/jpeg/decode_mcu_default.wuffs
+++ b/std/jpeg/decode_mcu_default.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.decode_mcu!(workbuf: slice base.u8, mx: base.u32[..= 0x1FFF], my: base.u32[..= 0x1FFF]) base.u32,
         choosy,
diff --git a/std/jpeg/decode_mcu_progressive_ac_high_bits.wuffs b/std/jpeg/decode_mcu_progressive_ac_high_bits.wuffs
index aaef165d7acd174f1e115e2ec90463d19dc81d5e..1c8795b62a405900300c568ff67893268ac4563f 100644
--- a/std/jpeg/decode_mcu_progressive_ac_high_bits.wuffs
+++ b/std/jpeg/decode_mcu_progressive_ac_high_bits.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.decode_mcu_progressive_ac_high_bits!(workbuf: slice base.u8, mx: base.u32[..= 0x1FFF], my: base.u32[..= 0x1FFF]) base.u32 {
     var ret : base.u32
diff --git a/std/jpeg/decode_mcu_progressive_ac_low_bit.wuffs b/std/jpeg/decode_mcu_progressive_ac_low_bit.wuffs
index 0ea58b8a041eda786f04e26f0fbf6e834eb91121..f08c010843a992c2aa4ea1709791d543b1cd698c 100644
--- a/std/jpeg/decode_mcu_progressive_ac_low_bit.wuffs
+++ b/std/jpeg/decode_mcu_progressive_ac_low_bit.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.decode_mcu_progressive_ac_low_bit!(workbuf: slice base.u8, mx: base.u32[..= 0x1FFF], my: base.u32[..= 0x1FFF]) base.u32 {
     var ret : base.u32
diff --git a/std/jpeg/decode_mcu_progressive_dc_high_bits.wuffs b/std/jpeg/decode_mcu_progressive_dc_high_bits.wuffs
index 222be1ec059cba2251a8805c9dde7253b5ff8a6d..54242a9abab911c3480879dccaef2f14934e4650 100644
--- a/std/jpeg/decode_mcu_progressive_dc_high_bits.wuffs
+++ b/std/jpeg/decode_mcu_progressive_dc_high_bits.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.decode_mcu_progressive_dc_high_bits!(workbuf: slice base.u8, mx: base.u32[..= 0x1FFF], my: base.u32[..= 0x1FFF]) base.u32 {
     var ret : base.u32
diff --git a/std/jpeg/decode_mcu_progressive_dc_low_bit.wuffs b/std/jpeg/decode_mcu_progressive_dc_low_bit.wuffs
index 3768db363cd3f479d7d10f21991309fbc7a72763..2afa0dbfae8e35665002186b7508e76e9bc12205 100644
--- a/std/jpeg/decode_mcu_progressive_dc_low_bit.wuffs
+++ b/std/jpeg/decode_mcu_progressive_dc_low_bit.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.decode_mcu_progressive_dc_low_bit!(workbuf: slice base.u8, mx: base.u32[..= 0x1FFF], my: base.u32[..= 0x1FFF]) base.u32 {
     var ret : base.u32
diff --git a/std/json/common_consts.wuffs b/std/json/common_consts.wuffs
index 71d7adb291512e755306e29b3e040f56b996e6f7..36947c00c46f92f5151813cf7e9a58cc03f49b1d 100644
--- a/std/json/common_consts.wuffs
+++ b/std/json/common_consts.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad C0 control code"
 pub status "#bad UTF-8"
diff --git a/std/json/decode_json.wuffs b/std/json/decode_json.wuffs
index e8824079ec54cdb236585af5415c1606c779b809..3c83a4059abb6201047c3e7f562988b95e30d12f 100644
--- a/std/json/decode_json.wuffs
+++ b/std/json/decode_json.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub struct decoder? implements base.token_decoder(
         quirks : array[QUIRKS_COUNT] base.bool,
diff --git a/std/json/decode_quirks.wuffs b/std/json/decode_quirks.wuffs
index 9d3ebb53441413fae6799423cfad3060bd1719cc..72b8d93f161553966f537c8c82915ed424a34dfd 100644
--- a/std/json/decode_quirks.wuffs
+++ b/std/json/decode_quirks.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/lzw/decode_lzw.wuffs b/std/lzw/decode_lzw.wuffs
index 857826db9d8af848d6ecfe114d33f0285b8eca98..3b474aad99389433b266777846d1cc681682a0e1 100644
--- a/std/lzw/decode_lzw.wuffs
+++ b/std/lzw/decode_lzw.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad code"
 pub status "#truncated input"
diff --git a/std/lzw/decode_quirks.wuffs b/std/lzw/decode_quirks.wuffs
index fc58e2f80983b66e45dd4664b42d0d2178731342..0bdd6ef210765ddb9dd12abac4938bfdeea0a8da 100644
--- a/std/lzw/decode_quirks.wuffs
+++ b/std/lzw/decode_quirks.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/netpbm/decode_netpbm.wuffs b/std/netpbm/decode_netpbm.wuffs
index ba76841a73be35f28fe4bbbc81bc97b16147b789..ca38fec3187fce228f1a765de55c95dcfad19e27 100644
--- a/std/netpbm/decode_netpbm.wuffs
+++ b/std/netpbm/decode_netpbm.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad header"
 pub status "#truncated input"
diff --git a/std/nie/decode_nie.wuffs b/std/nie/decode_nie.wuffs
index 77d3bae5b64c8a16cce3a278b05d817bf59ef5f8..e5ee5baa29f13db48b3ae2e663e15ea580406eae 100644
--- a/std/nie/decode_nie.wuffs
+++ b/std/nie/decode_nie.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad header"
 pub status "#truncated input"
diff --git a/std/png/common_consts.wuffs b/std/png/common_consts.wuffs
index 5a3353337fce4d548c3e13666508d1a7ec9fb3a7..35cdb2d52494529f6c0ef2a46ee7e700d78bdcc7 100644
--- a/std/png/common_consts.wuffs
+++ b/std/png/common_consts.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad animation sequence number"
 pub status "#bad checksum"
diff --git a/std/png/decode_filter_arm_neon.wuffs b/std/png/decode_filter_arm_neon.wuffs
index 5b1fd7c977fc9fc466da9dd46e7581eeb89af8c3..86b4abaa41f5ec415b64e1572a141d93aca9dc81 100644
--- a/std/png/decode_filter_arm_neon.wuffs
+++ b/std/png/decode_filter_arm_neon.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/png/decode_filter_fallback.wuffs b/std/png/decode_filter_fallback.wuffs
index 976271cb582dfe5e32a23c8e3565d58221d4f413..5b0cb68397ae82f19675291d8f84dc9f8cdc9ff2 100644
--- a/std/png/decode_filter_fallback.wuffs
+++ b/std/png/decode_filter_fallback.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/png/decode_filter_x86_sse42.wuffs b/std/png/decode_filter_x86_sse42.wuffs
index a5eecc454b31d6a8abc29b4debc55707cf01cacd..7d6e6b050d7de63951317ee0909530ced5d06cd9 100644
--- a/std/png/decode_filter_x86_sse42.wuffs
+++ b/std/png/decode_filter_x86_sse42.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/png/decode_png.wuffs b/std/png/decode_png.wuffs
index 9ebe16084ca8aea5d20ab0836dfdf00b4deed03c..8b3d51f0f9a0c529a86a373c698373caa581d7c6 100644
--- a/std/png/decode_png.wuffs
+++ b/std/png/decode_png.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 use "std/crc32"
 use "std/zlib"
diff --git a/std/png/decode_swizzle_default.wuffs b/std/png/decode_swizzle_default.wuffs
index 72e968ea59757a7806a8ac250ec059a6a0a99ccd..329f2c33d17662104d6939f916bcb22a20c68f6b 100644
--- a/std/png/decode_swizzle_default.wuffs
+++ b/std/png/decode_swizzle_default.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.filter_and_swizzle!(dst: ptr base.pixel_buffer, workbuf: slice base.u8) base.status,
         choosy,
diff --git a/std/png/decode_swizzle_tricky.wuffs b/std/png/decode_swizzle_tricky.wuffs
index 3a8c77f3b27d8bfd3d747fc73a75796756101feb..d99c6c96632d41d8e24c611f1f763fda5c17fc51 100644
--- a/std/png/decode_swizzle_tricky.wuffs
+++ b/std/png/decode_swizzle_tricky.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri func decoder.filter_and_swizzle_tricky!(dst: ptr base.pixel_buffer, workbuf: slice base.u8) base.status {
     var dst_pixfmt          : base.pixel_format
diff --git a/std/tga/decode_tga.wuffs b/std/tga/decode_tga.wuffs
index 3aa961ad86de0b3ed5fd1ddfb5e8557e5e4b5e2d..4bc8e387eaf9790bc6fb0bfa0f4abe479c8f5f1a 100644
--- a/std/tga/decode_tga.wuffs
+++ b/std/tga/decode_tga.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad header"
 pub status "#bad run length encoding"
diff --git a/std/wbmp/decode_wbmp.wuffs b/std/wbmp/decode_wbmp.wuffs
index 680a5f3d9e06f4528cbcfd42530aa6eae2fc796d..1aacc2eb05261e04048df5ede879881a3bd74737 100644
--- a/std/wbmp/decode_wbmp.wuffs
+++ b/std/wbmp/decode_wbmp.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pub status "#bad header"
 pub status "#truncated input"
diff --git a/std/xxhash32/common_xxhash32.wuffs b/std/xxhash32/common_xxhash32.wuffs
index 1c515c958460a4cb01387e40230ca46c0f23d400..d1515c4cb17233f633222c2d213840853b1d19d5 100644
--- a/std/xxhash32/common_xxhash32.wuffs
+++ b/std/xxhash32/common_xxhash32.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri const XXH_PRIME32_1 : base.u32 = 0x9E37_79B1
 pri const XXH_PRIME32_2 : base.u32 = 0x85EB_CA77
diff --git a/std/xxhash64/common_xxhash64.wuffs b/std/xxhash64/common_xxhash64.wuffs
index 0d2c0d0dd81017d7753c8c6e2eb9c2f9d0f05d34..e34d73cdf1f8be11a46de335b0ff9c018af4efd3 100644
--- a/std/xxhash64/common_xxhash64.wuffs
+++ b/std/xxhash64/common_xxhash64.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 pri const XXH_PRIME64_1 : base.u64 = 0x9E37_79B1_85EB_CA87
 pri const XXH_PRIME64_2 : base.u64 = 0xC2B2_AE3D_27D4_EB4F
diff --git a/std/zlib/decode_quirks.wuffs b/std/zlib/decode_quirks.wuffs
index df2cde5bc091ac3902b39e4d97df6e196f3902fa..f0e7bbb91c1c7b8e5f959700fa1a7bbe61ee8091 100644
--- a/std/zlib/decode_quirks.wuffs
+++ b/std/zlib/decode_quirks.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 // --------
 
diff --git a/std/zlib/decode_zlib.wuffs b/std/zlib/decode_zlib.wuffs
index c73fbb600f2fa7e3103f52746cbae16322e7dde4..050d3755403b7e3b6dbd9389a3dd829738116c2a 100644
--- a/std/zlib/decode_zlib.wuffs
+++ b/std/zlib/decode_zlib.wuffs
@@ -5,6 +5,8 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
 
 use "std/adler32"
 use "std/deflate"
