diff --git a/fuzz/c/std/pixel_swizzler_fuzzer.c b/fuzz/c/std/pixel_swizzler_fuzzer.c
index 6fbdf24d39e6a69c3e2e293b9cd67612c344af5a..c9c384c4ef4b5852f6e4cd0c5b1eb565063ff633 100644
--- a/fuzz/c/std/pixel_swizzler_fuzzer.c
+++ b/fuzz/c/std/pixel_swizzler_fuzzer.c
@@ -72,6 +72,7 @@ It should print "PASS", amongst other information, and exit(0).
 const uint32_t pixfmts[] = {
     WUFFS_BASE__PIXEL_FORMAT__Y,
     WUFFS_BASE__PIXEL_FORMAT__Y_16BE,
+    WUFFS_BASE__PIXEL_FORMAT__Y_16LE,
     WUFFS_BASE__PIXEL_FORMAT__YA_NONPREMUL,
     WUFFS_BASE__PIXEL_FORMAT__INDEXED__BGRA_NONPREMUL,
     WUFFS_BASE__PIXEL_FORMAT__INDEXED__BGRA_BINARY,
