diff --git a/fuzz/c/std/pixel_swizzler_fuzzer.c b/fuzz/c/std/pixel_swizzler_fuzzer.c
index c9c384c4ef4b5852f6e4cd0c5b1eb565063ff633..21b449d0280dabad040f2d32085ff771800aee60 100644
--- a/fuzz/c/std/pixel_swizzler_fuzzer.c
+++ b/fuzz/c/std/pixel_swizzler_fuzzer.c
@@ -81,6 +81,7 @@ const uint32_t pixfmts[] = {
     WUFFS_BASE__PIXEL_FORMAT__BGRA_NONPREMUL,
     WUFFS_BASE__PIXEL_FORMAT__BGRA_NONPREMUL_4X16LE,
     WUFFS_BASE__PIXEL_FORMAT__BGRA_PREMUL,
+    WUFFS_BASE__PIXEL_FORMAT__BGRA_PREMUL_4X16LE,
     WUFFS_BASE__PIXEL_FORMAT__BGRX,
     WUFFS_BASE__PIXEL_FORMAT__RGB,
     WUFFS_BASE__PIXEL_FORMAT__RGBA_NONPREMUL,
