diff --git a/fuzz/c/std/pixel_swizzler_fuzzer.c b/fuzz/c/std/pixel_swizzler_fuzzer.c
index 21b449d0280dabad040f2d32085ff771800aee60..a784a0273a75dec09ae343dc8288182d0ebd5bbe 100644
--- a/fuzz/c/std/pixel_swizzler_fuzzer.c
+++ b/fuzz/c/std/pixel_swizzler_fuzzer.c
@@ -237,11 +237,11 @@ fuzz_swizzle_ycck(wuffs_base__io_buffer* src, uint64_t hash) {
   uint32_t height_in_mcus = (3 & (hash >> 34)) + 1;
 
   uint32_t allow_hv3 = 1 & (hash >> 36);
-  bool is_rgb_or_cmyk = 1 & (hash >> 37);
-  bool triangle_filter_for_2to1 = 1 & (hash >> 38);
+  uint8_t ycc_model = (1 & (hash >> 37)) ? 0x80 : 0x00;
+  uint8_t ycc_upsampling = 1 & (hash >> 38);
   bool use_4th_component = 1 & (hash >> 39);
 
-  if (triangle_filter_for_2to1) {
+  if (ycc_upsampling) {
     x_min_incl = 0;
     y_min_incl = 0;
   } else {
@@ -411,8 +411,8 @@ fuzz_swizzle_ycck(wuffs_base__io_buffer* src, uint64_t hash) {
       width0, width1, width2, width3,       //
       h0, h1, h2, h3,                       //
       v0, v1, v2, v3,                       //
-      is_rgb_or_cmyk,                       //
-      triangle_filter_for_2to1,             //
+      ycc_model,                            //
+      ycc_upsampling,                       //
       wuffs_base__make_slice_u8(scratch_buffer, sizeof(scratch_buffer)));
   if (status.repr) {
     return wuffs_base__status__message(&status);
