diff --git a/internal/cgen/base/image-public.h b/internal/cgen/base/image-public.h
index 6aa5084c5161a7067f88b3669f121f21e47e6406..62ac91469f05d2783fd20875b5030e1a61960024 100644
--- a/internal/cgen/base/image-public.h
+++ b/internal/cgen/base/image-public.h
@@ -10,6 +10,21 @@
 
 // ---------------- Images
 
+// WUFFS_BASE__IMAGE__DIMENSION_MAX_INCL is the largest width or height
+// (measured in pixels) supported by Wuffs' standard library's image codecs.
+//
+// Any non-negative integer N less than or equal to ((1 << 24) - 1) can be
+// losslessly converted to and from single-precision floating point (what C
+// calls the "float" type and what Go calls "float32").
+//
+// A Wuffs image can have up to 4 channels and up to 8 bytes per channel (8 is
+// also known as sizeof(double)), combining for up to 32 bytes per pixel. When
+// calculating memory requirements, the 0xFFFFFF upper bound also means that:
+//   - (32 * N)     will not overflow an int32_t or uint32_t.
+//   - (32 * N * N) will not overflow an int64_t or uint64_t, provided that N
+//     already has a 64-bit type.
+//
+// 0xFFFFFF in decimal is 16777215.
 #define WUFFS_BASE__IMAGE__DIMENSION_MAX_INCL 0xFFFFFF
 
 // wuffs_base__color_u32_argb_premul is an 8 bit per channel premultiplied
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index e2c3b7a256872c843db5d430b521f22e577f7a64..3755bfb20edc0df43b8b2f943b3ab9f79b288b8d 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -4064,6 +4064,21 @@ wuffs_base__malloc_slice_u64(void* (*malloc_func)(size_t), uint64_t num_u64) {
 
 // ---------------- Images
 
+// WUFFS_BASE__IMAGE__DIMENSION_MAX_INCL is the largest width or height
+// (measured in pixels) supported by Wuffs' standard library's image codecs.
+//
+// Any non-negative integer N less than or equal to ((1 << 24) - 1) can be
+// losslessly converted to and from single-precision floating point (what C
+// calls the "float" type and what Go calls "float32").
+//
+// A Wuffs image can have up to 4 channels and up to 8 bytes per channel (8 is
+// also known as sizeof(double)), combining for up to 32 bytes per pixel. When
+// calculating memory requirements, the 0xFFFFFF upper bound also means that:
+//   - (32 * N)     will not overflow an int32_t or uint32_t.
+//   - (32 * N * N) will not overflow an int64_t or uint64_t, provided that N
+//     already has a 64-bit type.
+//
+// 0xFFFFFF in decimal is 16777215.
 #define WUFFS_BASE__IMAGE__DIMENSION_MAX_INCL 0xFFFFFF
 
 // wuffs_base__color_u32_argb_premul is an 8 bit per channel premultiplied
