diff --git a/internal/cgen/base/image-public.h b/internal/cgen/base/image-public.h
index 37fc80615ec9652eec3470113bfe88625f1cc463..a890e5e69ebddd9444d558f9bdc26be8493fe9c2 100644
--- a/internal/cgen/base/image-public.h
+++ b/internal/cgen/base/image-public.h
@@ -651,7 +651,7 @@ wuffs_base__pixel_config__set(wuffs_base__pixel_config* c,
   }
   if (pixfmt_repr) {
     do {
-#if SIZE_MAX < UINT64_MAX
+#if SIZE_MAX < 0xFFFFFFFFFFFFFFFFull
       uint64_t wh = ((uint64_t)width) * ((uint64_t)height);
       // TODO: handle things other than 1 byte per pixel.
       if (wh > ((uint64_t)SIZE_MAX)) {
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index b3c4abb276c15b6e8ed752c818ac4e08743ec96f..8ca2471e0b13ecae84a3f05d3ae850db003c9bb8 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -4664,7 +4664,7 @@ wuffs_base__pixel_config__set(wuffs_base__pixel_config* c,
   }
   if (pixfmt_repr) {
     do {
-#if SIZE_MAX < UINT64_MAX
+#if SIZE_MAX < 0xFFFFFFFFFFFFFFFFull
       uint64_t wh = ((uint64_t)width) * ((uint64_t)height);
       // TODO: handle things other than 1 byte per pixel.
       if (wh > ((uint64_t)SIZE_MAX)) {
