diff --git a/internal/cgen/auxiliary/base.cc b/internal/cgen/auxiliary/base.cc
index 3622285b0a53177baa95e79423d9098838a9c6d8..a4e84724f5dd9437da1c693a5e2e8a0b66365029 100644
--- a/internal/cgen/auxiliary/base.cc
+++ b/internal/cgen/auxiliary/base.cc
@@ -25,16 +25,12 @@ DynIOBuffer::DynIOBuffer(uint64_t max_incl)
     : m_buf(wuffs_base__empty_io_buffer()), m_max_incl(max_incl) {}
 
 DynIOBuffer::~DynIOBuffer() {
-  if (allocated()) {
-    free(m_buf.data.ptr);
-  }
+  free(m_buf.data.ptr);
 }
 
 void  //
 DynIOBuffer::drop() {
-  if (allocated()) {
-    free(m_buf.data.ptr);
-  }
+  free(m_buf.data.ptr);
   m_buf = wuffs_base__empty_io_buffer();
 }
 
@@ -48,9 +44,8 @@ DynIOBuffer::grow(uint64_t min_incl) {
   } else if (n > SIZE_MAX) {
     return DynIOBuffer::GrowResult::FailedOutOfMemory;
   } else if (n > m_buf.data.len) {
-    uint8_t* ptr = static_cast<uint8_t*>(
-        allocated() ? realloc(m_buf.data.ptr, static_cast<size_t>(n))
-                    : malloc(static_cast<size_t>(n)));
+    uint8_t* ptr =
+        static_cast<uint8_t*>(realloc(m_buf.data.ptr, static_cast<size_t>(n)));
     if (!ptr) {
       return DynIOBuffer::GrowResult::FailedOutOfMemory;
     }
@@ -60,12 +55,6 @@ DynIOBuffer::grow(uint64_t min_incl) {
   return DynIOBuffer::GrowResult::OK;
 }
 
-bool  //
-DynIOBuffer::allocated() {
-  IOBuffer buf = wuffs_base__empty_io_buffer();
-  return buf.data.ptr != m_buf.data.ptr;
-}
-
 // round_up rounds min_incl up, returning the smallest value x satisfying
 // (min_incl <= x) and (x <= max_incl) and some other constraints. It returns 0
 // if there is no such x.
diff --git a/internal/cgen/auxiliary/base.hh b/internal/cgen/auxiliary/base.hh
index 28f9ccde7ba2729d58979b07af3133c315dc0c24..03a49cb2e0e3499b1581e8f3d118b67c327db1d6 100644
--- a/internal/cgen/auxiliary/base.hh
+++ b/internal/cgen/auxiliary/base.hh
@@ -84,8 +84,6 @@ class DynIOBuffer {
   DynIOBuffer(const DynIOBuffer&) = delete;
   DynIOBuffer& operator=(const DynIOBuffer&) = delete;
 
-  bool allocated();
-
   static uint64_t round_up(uint64_t min_incl, uint64_t max_incl);
 };
 
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index 4c406ebd92957d4ccad8e09a1dc1a4fabe4b7c9d..ceb4ff17819f4d14ec1c1e42be8ee00abc88a18f 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -14440,8 +14440,6 @@ class DynIOBuffer {
   DynIOBuffer(const DynIOBuffer&) = delete;
   DynIOBuffer& operator=(const DynIOBuffer&) = delete;
 
-  bool allocated();
-
   static uint64_t round_up(uint64_t min_incl, uint64_t max_incl);
 };
 
@@ -72077,16 +72075,12 @@ DynIOBuffer::DynIOBuffer(uint64_t max_incl)
     : m_buf(wuffs_base__empty_io_buffer()), m_max_incl(max_incl) {}
 
 DynIOBuffer::~DynIOBuffer() {
-  if (allocated()) {
-    free(m_buf.data.ptr);
-  }
+  free(m_buf.data.ptr);
 }
 
 void  //
 DynIOBuffer::drop() {
-  if (allocated()) {
-    free(m_buf.data.ptr);
-  }
+  free(m_buf.data.ptr);
   m_buf = wuffs_base__empty_io_buffer();
 }
 
@@ -72100,9 +72094,8 @@ DynIOBuffer::grow(uint64_t min_incl) {
   } else if (n > SIZE_MAX) {
     return DynIOBuffer::GrowResult::FailedOutOfMemory;
   } else if (n > m_buf.data.len) {
-    uint8_t* ptr = static_cast<uint8_t*>(
-        allocated() ? realloc(m_buf.data.ptr, static_cast<size_t>(n))
-                    : malloc(static_cast<size_t>(n)));
+    uint8_t* ptr =
+        static_cast<uint8_t*>(realloc(m_buf.data.ptr, static_cast<size_t>(n)));
     if (!ptr) {
       return DynIOBuffer::GrowResult::FailedOutOfMemory;
     }
@@ -72112,12 +72105,6 @@ DynIOBuffer::grow(uint64_t min_incl) {
   return DynIOBuffer::GrowResult::OK;
 }
 
-bool  //
-DynIOBuffer::allocated() {
-  IOBuffer buf = wuffs_base__empty_io_buffer();
-  return buf.data.ptr != m_buf.data.ptr;
-}
-
 // round_up rounds min_incl up, returning the smallest value x satisfying
 // (min_incl <= x) and (x <= max_incl) and some other constraints. It returns 0
 // if there is no such x.
