diff --git a/internal/cgen/base/fundamental-private.h b/internal/cgen/base/fundamental-private.h
index 8f8ee473cea8776c64a76d9415499aedc9b99709..49689a56d2913863ab106826ac78791b997c16eb 100644
--- a/internal/cgen/base/fundamental-private.h
+++ b/internal/cgen/base/fundamental-private.h
@@ -269,7 +269,7 @@ wuffs_private_impl__bulk_save_host_endian(void* ptr,
 
 static inline wuffs_base__slice_u8  //
 wuffs_private_impl__table_u8__row_u32(wuffs_base__table_u8 t, uint32_t y) {
-  if (y < t.height) {
+  if (t.ptr && (y < t.height)) {
     return wuffs_base__make_slice_u8(t.ptr + (t.stride * y), t.width);
   }
   return wuffs_base__empty_slice_u8();
diff --git a/internal/cgen/base/fundamental-public.h b/internal/cgen/base/fundamental-public.h
index 1db9d0d9258780f637bec428bddaed2e01a38f4d..3bc9bd84dca03b048b1b6c6dc3eb35578981c198 100644
--- a/internal/cgen/base/fundamental-public.h
+++ b/internal/cgen/base/fundamental-public.h
@@ -1384,7 +1384,7 @@ wuffs_base__make_slice_u64(uint64_t* ptr, size_t len) {
 static inline wuffs_base__slice_u8  //
 wuffs_base__make_slice_u8_ij(uint8_t* ptr, size_t i, size_t j) {
   wuffs_base__slice_u8 ret;
-  ret.ptr = ptr + i;
+  ret.ptr = ptr ? (ptr + i) : NULL;
   ret.len = (j >= i) ? (j - i) : 0;
   return ret;
 }
@@ -1392,7 +1392,7 @@ wuffs_base__make_slice_u8_ij(uint8_t* ptr, size_t i, size_t j) {
 static inline wuffs_base__slice_u16  //
 wuffs_base__make_slice_u16_ij(uint16_t* ptr, size_t i, size_t j) {
   wuffs_base__slice_u16 ret;
-  ret.ptr = ptr + i;
+  ret.ptr = ptr ? (ptr + i) : NULL;
   ret.len = (j >= i) ? (j - i) : 0;
   return ret;
 }
@@ -1400,7 +1400,7 @@ wuffs_base__make_slice_u16_ij(uint16_t* ptr, size_t i, size_t j) {
 static inline wuffs_base__slice_u32  //
 wuffs_base__make_slice_u32_ij(uint32_t* ptr, size_t i, size_t j) {
   wuffs_base__slice_u32 ret;
-  ret.ptr = ptr + i;
+  ret.ptr = ptr ? (ptr + i) : NULL;
   ret.len = (j >= i) ? (j - i) : 0;
   return ret;
 }
@@ -1408,7 +1408,7 @@ wuffs_base__make_slice_u32_ij(uint32_t* ptr, size_t i, size_t j) {
 static inline wuffs_base__slice_u64  //
 wuffs_base__make_slice_u64_ij(uint64_t* ptr, size_t i, size_t j) {
   wuffs_base__slice_u64 ret;
-  ret.ptr = ptr + i;
+  ret.ptr = ptr ? (ptr + i) : NULL;
   ret.len = (j >= i) ? (j - i) : 0;
   return ret;
 }
diff --git a/internal/cgen/base/io-public.h b/internal/cgen/base/io-public.h
index 12de6feb11083dd93b5790be77f76b78e482c225..c879f3bf16e20816dca8cb6a670f97bb1c65b4e8 100644
--- a/internal/cgen/base/io-public.h
+++ b/internal/cgen/base/io-public.h
@@ -214,7 +214,7 @@ wuffs_base__io_buffer__reader_length(const wuffs_base__io_buffer* buf) {
 
 static inline uint8_t*  //
 wuffs_base__io_buffer__reader_pointer(const wuffs_base__io_buffer* buf) {
-  return buf ? (buf->data.ptr + buf->meta.ri) : NULL;
+  return (buf && buf->data.ptr) ? (buf->data.ptr + buf->meta.ri) : NULL;
 }
 
 static inline uint64_t  //
@@ -224,7 +224,8 @@ wuffs_base__io_buffer__reader_position(const wuffs_base__io_buffer* buf) {
 
 static inline wuffs_base__slice_u8  //
 wuffs_base__io_buffer__reader_slice(const wuffs_base__io_buffer* buf) {
-  return buf ? wuffs_base__make_slice_u8(buf->data.ptr + buf->meta.ri,
+  return (buf && buf->data.ptr)
+             ? wuffs_base__make_slice_u8(buf->data.ptr + buf->meta.ri,
                                          buf->meta.wi - buf->meta.ri)
              : wuffs_base__empty_slice_u8();
 }
@@ -236,7 +237,7 @@ wuffs_base__io_buffer__writer_length(const wuffs_base__io_buffer* buf) {
 
 static inline uint8_t*  //
 wuffs_base__io_buffer__writer_pointer(const wuffs_base__io_buffer* buf) {
-  return buf ? (buf->data.ptr + buf->meta.wi) : NULL;
+  return (buf && buf->data.ptr) ? (buf->data.ptr + buf->meta.wi) : NULL;
 }
 
 static inline uint64_t  //
@@ -246,7 +247,8 @@ wuffs_base__io_buffer__writer_position(const wuffs_base__io_buffer* buf) {
 
 static inline wuffs_base__slice_u8  //
 wuffs_base__io_buffer__writer_slice(const wuffs_base__io_buffer* buf) {
-  return buf ? wuffs_base__make_slice_u8(buf->data.ptr + buf->meta.wi,
+  return (buf && buf->data.ptr)
+             ? wuffs_base__make_slice_u8(buf->data.ptr + buf->meta.wi,
                                          buf->data.len - buf->meta.wi)
              : wuffs_base__empty_slice_u8();
 }
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index 7242567d993c86695c92bda83f366f2be9a49ca4..4c406ebd92957d4ccad8e09a1dc1a4fabe4b7c9d 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -1661,7 +1661,7 @@ wuffs_base__make_slice_u64(uint64_t* ptr, size_t len) {
 static inline wuffs_base__slice_u8  //
 wuffs_base__make_slice_u8_ij(uint8_t* ptr, size_t i, size_t j) {
   wuffs_base__slice_u8 ret;
-  ret.ptr = ptr + i;
+  ret.ptr = ptr ? (ptr + i) : NULL;
   ret.len = (j >= i) ? (j - i) : 0;
   return ret;
 }
@@ -1669,7 +1669,7 @@ wuffs_base__make_slice_u8_ij(uint8_t* ptr, size_t i, size_t j) {
 static inline wuffs_base__slice_u16  //
 wuffs_base__make_slice_u16_ij(uint16_t* ptr, size_t i, size_t j) {
   wuffs_base__slice_u16 ret;
-  ret.ptr = ptr + i;
+  ret.ptr = ptr ? (ptr + i) : NULL;
   ret.len = (j >= i) ? (j - i) : 0;
   return ret;
 }
@@ -1677,7 +1677,7 @@ wuffs_base__make_slice_u16_ij(uint16_t* ptr, size_t i, size_t j) {
 static inline wuffs_base__slice_u32  //
 wuffs_base__make_slice_u32_ij(uint32_t* ptr, size_t i, size_t j) {
   wuffs_base__slice_u32 ret;
-  ret.ptr = ptr + i;
+  ret.ptr = ptr ? (ptr + i) : NULL;
   ret.len = (j >= i) ? (j - i) : 0;
   return ret;
 }
@@ -1685,7 +1685,7 @@ wuffs_base__make_slice_u32_ij(uint32_t* ptr, size_t i, size_t j) {
 static inline wuffs_base__slice_u64  //
 wuffs_base__make_slice_u64_ij(uint64_t* ptr, size_t i, size_t j) {
   wuffs_base__slice_u64 ret;
-  ret.ptr = ptr + i;
+  ret.ptr = ptr ? (ptr + i) : NULL;
   ret.len = (j >= i) ? (j - i) : 0;
   return ret;
 }
@@ -3189,7 +3189,7 @@ wuffs_base__io_buffer__reader_length(const wuffs_base__io_buffer* buf) {
 
 static inline uint8_t*  //
 wuffs_base__io_buffer__reader_pointer(const wuffs_base__io_buffer* buf) {
-  return buf ? (buf->data.ptr + buf->meta.ri) : NULL;
+  return (buf && buf->data.ptr) ? (buf->data.ptr + buf->meta.ri) : NULL;
 }
 
 static inline uint64_t  //
@@ -3199,7 +3199,8 @@ wuffs_base__io_buffer__reader_position(const wuffs_base__io_buffer* buf) {
 
 static inline wuffs_base__slice_u8  //
 wuffs_base__io_buffer__reader_slice(const wuffs_base__io_buffer* buf) {
-  return buf ? wuffs_base__make_slice_u8(buf->data.ptr + buf->meta.ri,
+  return (buf && buf->data.ptr)
+             ? wuffs_base__make_slice_u8(buf->data.ptr + buf->meta.ri,
                                          buf->meta.wi - buf->meta.ri)
              : wuffs_base__empty_slice_u8();
 }
@@ -3211,7 +3212,7 @@ wuffs_base__io_buffer__writer_length(const wuffs_base__io_buffer* buf) {
 
 static inline uint8_t*  //
 wuffs_base__io_buffer__writer_pointer(const wuffs_base__io_buffer* buf) {
-  return buf ? (buf->data.ptr + buf->meta.wi) : NULL;
+  return (buf && buf->data.ptr) ? (buf->data.ptr + buf->meta.wi) : NULL;
 }
 
 static inline uint64_t  //
@@ -3221,7 +3222,8 @@ wuffs_base__io_buffer__writer_position(const wuffs_base__io_buffer* buf) {
 
 static inline wuffs_base__slice_u8  //
 wuffs_base__io_buffer__writer_slice(const wuffs_base__io_buffer* buf) {
-  return buf ? wuffs_base__make_slice_u8(buf->data.ptr + buf->meta.wi,
+  return (buf && buf->data.ptr)
+             ? wuffs_base__make_slice_u8(buf->data.ptr + buf->meta.wi,
                                          buf->data.len - buf->meta.wi)
              : wuffs_base__empty_slice_u8();
 }
@@ -15309,7 +15311,7 @@ wuffs_private_impl__bulk_save_host_endian(void* ptr,
 
 static inline wuffs_base__slice_u8  //
 wuffs_private_impl__table_u8__row_u32(wuffs_base__table_u8 t, uint32_t y) {
-  if (y < t.height) {
+  if (t.ptr && (y < t.height)) {
     return wuffs_base__make_slice_u8(t.ptr + (t.stride * y), t.width);
   }
   return wuffs_base__empty_slice_u8();
