diff --git a/internal/cgen/base/pixconv-submodule-regular.c b/internal/cgen/base/pixconv-submodule-regular.c
index e0f8fa795ed99e8c2e5e54f509d6bcb462fca206..bac14e02534f2192b7314cbefb8245d232287bed 100644
--- a/internal/cgen/base/pixconv-submodule-regular.c
+++ b/internal/cgen/base/pixconv-submodule-regular.c
@@ -4752,10 +4752,9 @@ wuffs_private_impl__swizzle_ya_nonpremul__ya_nonpremul__src_over(
   while (n >= 1) {
     uint32_t d0 = ((uint32_t)(d[1]) << 24) | ((uint32_t)(d[0]) * 0x010101);
     uint32_t s0 = ((uint32_t)(s[1]) << 24) | ((uint32_t)(s[0]) * 0x010101);
-    wuffs_base__poke_u16le__no_bounds_check(
-        d + (0 * 2),
-        wuffs_private_impl__composite_nonpremul_nonpremul_u32_axxx(d0, s0) >>
-            16);
+    uint32_t c0 =
+        wuffs_private_impl__composite_nonpremul_nonpremul_u32_axxx(d0, s0);
+    wuffs_base__poke_u16le__no_bounds_check(d + (0 * 2), (uint16_t)(c0 >> 16));
 
     s += 1 * 2;
     d += 1 * 2;
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index a764d846a1ad5267954fff073c3263cd474ab0f3..689e0507e31405c177da56e02afb1721c16a13f7 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -25717,10 +25717,9 @@ wuffs_private_impl__swizzle_ya_nonpremul__ya_nonpremul__src_over(
   while (n >= 1) {
     uint32_t d0 = ((uint32_t)(d[1]) << 24) | ((uint32_t)(d[0]) * 0x010101);
     uint32_t s0 = ((uint32_t)(s[1]) << 24) | ((uint32_t)(s[0]) * 0x010101);
-    wuffs_base__poke_u16le__no_bounds_check(
-        d + (0 * 2),
-        wuffs_private_impl__composite_nonpremul_nonpremul_u32_axxx(d0, s0) >>
-            16);
+    uint32_t c0 =
+        wuffs_private_impl__composite_nonpremul_nonpremul_u32_axxx(d0, s0);
+    wuffs_base__poke_u16le__no_bounds_check(d + (0 * 2), (uint16_t)(c0 >> 16));
 
     s += 1 * 2;
     d += 1 * 2;
