diff --git a/internal/cgen/base/fundamental-public.h b/internal/cgen/base/fundamental-public.h
index 6cfe4ce4d9f85e70d1520b46eaa793c73c3a501e..91a3fffed4b7cfe1b6081c980e0909dc7f2602ea 100644
--- a/internal/cgen/base/fundamental-public.h
+++ b/internal/cgen/base/fundamental-public.h
@@ -893,7 +893,7 @@ wuffs_base__make_bitvec256(uint64_t e00,
 }
 
 static inline uint64_t  //
-wuffs_base__bitvec256__get(const wuffs_base__bitvec256* b, uint32_t i) {
+wuffs_base__bitvec256__get_u64(const wuffs_base__bitvec256* b, uint32_t i) {
   return b->elements_u64[i & 3];
 }
 
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index 641c20025c44c6eb0b728dd6d3c2152dbfe9afc7..0a7ba242ece2057a80f059c6baf1ae9731450023 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -1165,7 +1165,7 @@ wuffs_base__make_bitvec256(uint64_t e00,
 }
 
 static inline uint64_t  //
-wuffs_base__bitvec256__get(const wuffs_base__bitvec256* b, uint32_t i) {
+wuffs_base__bitvec256__get_u64(const wuffs_base__bitvec256* b, uint32_t i) {
   return b->elements_u64[i & 3];
 }
 
@@ -67230,7 +67230,7 @@ wuffs_xz__decoder__do_transform_io(
           }
           v_checksum64_want = t_8;
         }
-        if (wuffs_base__bitvec256__get(&v_checksum256_have, 3u) != v_checksum64_want) {
+        if (wuffs_base__bitvec256__get_u64(&v_checksum256_have, 3u) != v_checksum64_want) {
           status = wuffs_base__make_status(wuffs_xz__error__bad_checksum);
           goto exit;
         }
@@ -67263,7 +67263,7 @@ wuffs_xz__decoder__do_transform_io(
           }
           v_checksum64_want = t_9;
         }
-        if (wuffs_base__bitvec256__get(&v_checksum256_have, 2u) != v_checksum64_want) {
+        if (wuffs_base__bitvec256__get_u64(&v_checksum256_have, 2u) != v_checksum64_want) {
           status = wuffs_base__make_status(wuffs_xz__error__bad_checksum);
           goto exit;
         }
@@ -67296,7 +67296,7 @@ wuffs_xz__decoder__do_transform_io(
           }
           v_checksum64_want = t_10;
         }
-        if (wuffs_base__bitvec256__get(&v_checksum256_have, 1u) != v_checksum64_want) {
+        if (wuffs_base__bitvec256__get_u64(&v_checksum256_have, 1u) != v_checksum64_want) {
           status = wuffs_base__make_status(wuffs_xz__error__bad_checksum);
           goto exit;
         }
@@ -67329,7 +67329,7 @@ wuffs_xz__decoder__do_transform_io(
           }
           v_checksum64_want = t_11;
         }
-        if (wuffs_base__bitvec256__get(&v_checksum256_have, 0u) != v_checksum64_want) {
+        if (wuffs_base__bitvec256__get_u64(&v_checksum256_have, 0u) != v_checksum64_want) {
           status = wuffs_base__make_status(wuffs_xz__error__bad_checksum);
           goto exit;
         }
diff --git a/std/xz/decode_xz.wuffs b/std/xz/decode_xz.wuffs
index 411493a263823dac690ef28fa85609c220cacde1..1cbfd6a94c511a589f1c0867a452b4efd1ccb2d9 100644
--- a/std/xz/decode_xz.wuffs
+++ b/std/xz/decode_xz.wuffs
@@ -225,19 +225,19 @@ pri func decoder.do_transform_io?(dst: base.io_writer, src: base.io_reader, work
         } else if this.checksummer == 3 {
             checksum256_have = this.sha256.checksum_bitvec256()
             checksum64_want = args.src.read_u64be?()
-            if checksum256_have.get(i: 3) <> checksum64_want {
+            if checksum256_have.get_u64(i: 3) <> checksum64_want {
                 return "#bad checksum"
             }
             checksum64_want = args.src.read_u64be?()
-            if checksum256_have.get(i: 2) <> checksum64_want {
+            if checksum256_have.get_u64(i: 2) <> checksum64_want {
                 return "#bad checksum"
             }
             checksum64_want = args.src.read_u64be?()
-            if checksum256_have.get(i: 1) <> checksum64_want {
+            if checksum256_have.get_u64(i: 1) <> checksum64_want {
                 return "#bad checksum"
             }
             checksum64_want = args.src.read_u64be?()
-            if checksum256_have.get(i: 0) <> checksum64_want {
+            if checksum256_have.get_u64(i: 0) <> checksum64_want {
                 return "#bad checksum"
             }
         }
