diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index cda9f73406d12e32d7bf86d3ad8ce14bf6214a82..667c0e24587a387a74bc62c736dc248992c61f35 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -55945,7 +55945,6 @@ const char wuffs_png__error__unsupported_png_compression_method[] = "#png: unsup
 const char wuffs_png__error__unsupported_png_file[] = "#png: unsupported PNG file";
 const char wuffs_png__error__internal_error_inconsistent_i_o[] = "#png: internal error: inconsistent I/O";
 const char wuffs_png__error__internal_error_inconsistent_chunk_type[] = "#png: internal error: inconsistent chunk type";
-const char wuffs_png__error__internal_error_inconsistent_frame_bounds[] = "#png: internal error: inconsistent frame bounds";
 const char wuffs_png__error__internal_error_inconsistent_workbuf_length[] = "#png: internal error: inconsistent workbuf length";
 const char wuffs_png__error__internal_error_zlib_decoder_did_not_exhaust_its_input[] = "#png: internal error: zlib decoder did not exhaust its input";
 
diff --git a/std/png/common_consts.wuffs b/std/png/common_consts.wuffs
index 35cdb2d52494529f6c0ef2a46ee7e700d78bdcc7..6b3fc109e399672f1f68190f535bdab58bec801d 100644
--- a/std/png/common_consts.wuffs
+++ b/std/png/common_consts.wuffs
@@ -22,7 +22,6 @@ pub status "#unsupported PNG file"
 
 pri status "#internal error: inconsistent I/O"
 pri status "#internal error: inconsistent chunk type"
-pri status "#internal error: inconsistent frame bounds"
 pri status "#internal error: inconsistent workbuf length"
 pri status "#internal error: zlib decoder did not exhaust its input"
 
