diff --git a/internal/cgen/base/floatconv-submodule-code.c b/internal/cgen/base/floatconv-submodule-code.c
index 663914d0b9fbd1f3d72e63a2d8195fd555d03f22..675c25c2c0c066125896a8644dbbf1308fa283c9 100644
--- a/internal/cgen/base/floatconv-submodule-code.c
+++ b/internal/cgen/base/floatconv-submodule-code.c
@@ -1336,7 +1336,7 @@ wuffs_private_impl__high_prec_dec__to_f64(wuffs_private_impl__high_prec_dec* h,
       shift = (n < num_powers)
                   // The +1 is per "when targeting h->decimal_point being +0 or
                   // +1... when left shifting" in the powers comment above.
-                  ? (powers[n] + 1)
+                  ? (powers[n] + 1u)
                   : WUFFS_PRIVATE_IMPL__HPD__SHIFT__MAX_INCL;
 
       wuffs_private_impl__high_prec_dec__small_lshift(h, shift);
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index fefa4927ee13101f172873995978c180a4eb57d3..283b0db6b3b2e18e9c789eb25de65891803e6d29 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -18774,7 +18774,7 @@ wuffs_private_impl__high_prec_dec__to_f64(wuffs_private_impl__high_prec_dec* h,
       shift = (n < num_powers)
                   // The +1 is per "when targeting h->decimal_point being +0 or
                   // +1... when left shifting" in the powers comment above.
-                  ? (powers[n] + 1)
+                  ? (powers[n] + 1u)
                   : WUFFS_PRIVATE_IMPL__HPD__SHIFT__MAX_INCL;
 
       wuffs_private_impl__high_prec_dec__small_lshift(h, shift);
