diff --git a/internal/cgen/auxiliary/cbor.cc b/internal/cgen/auxiliary/cbor.cc
index fb55a5d71bb89f97338f978a505f8be623d20e32..b834c339fc9524432c5803f191d3022bd15905ea 100644
--- a/internal/cgen/auxiliary/cbor.cc
+++ b/internal/cgen/auxiliary/cbor.cc
@@ -197,7 +197,7 @@ DecodeCbor(DecodeCborCallbacks& callbacks,
               goto done;
             }
             depth++;
-            if (depth > WUFFS_CBOR__DECODER_DEPTH_MAX_INCL) {
+            if (depth > (int32_t)WUFFS_CBOR__DECODER_DEPTH_MAX_INCL) {
               ret_error_message =
                   "wuffs_aux::DecodeCbor: internal error: bad depth";
               goto done;
diff --git a/internal/cgen/auxiliary/json.cc b/internal/cgen/auxiliary/json.cc
index ffef6124c2afed840521f1479463ad4296a100b1..33153bf7847b5404117e2e176caf60478b5e9f3a 100644
--- a/internal/cgen/auxiliary/json.cc
+++ b/internal/cgen/auxiliary/json.cc
@@ -446,7 +446,7 @@ DecodeJson(DecodeJsonCallbacks& callbacks,
               goto done;
             }
             depth++;
-            if (depth > WUFFS_JSON__DECODER_DEPTH_MAX_INCL) {
+            if (depth > (int32_t)WUFFS_JSON__DECODER_DEPTH_MAX_INCL) {
               ret_error_message =
                   "wuffs_aux::DecodeJson: internal error: bad depth";
               goto done;
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index 76b770f35434500a045e43b7fc540a51c1419bb4..b8f60cd381af58dce3296c82e17aeb32e1aa9069 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -64623,7 +64623,7 @@ DecodeCbor(DecodeCborCallbacks& callbacks,
               goto done;
             }
             depth++;
-            if (depth > WUFFS_CBOR__DECODER_DEPTH_MAX_INCL) {
+            if (depth > (int32_t)WUFFS_CBOR__DECODER_DEPTH_MAX_INCL) {
               ret_error_message =
                   "wuffs_aux::DecodeCbor: internal error: bad depth";
               goto done;
@@ -65852,7 +65852,7 @@ DecodeJson(DecodeJsonCallbacks& callbacks,
               goto done;
             }
             depth++;
-            if (depth > WUFFS_JSON__DECODER_DEPTH_MAX_INCL) {
+            if (depth > (int32_t)WUFFS_JSON__DECODER_DEPTH_MAX_INCL) {
               ret_error_message =
                   "wuffs_aux::DecodeJson: internal error: bad depth";
               goto done;
