diff --git a/std/bzip2/decode_bzip2.wuffs b/std/bzip2/decode_bzip2.wuffs
index c06e0733ec6aefbbb680aea2ac92fab344368522..3c18b91208aaf42a94105498a4c815210f7c07c9 100644
--- a/std/bzip2/decode_bzip2.wuffs
+++ b/std/bzip2/decode_bzip2.wuffs
@@ -95,7 +95,8 @@ pub struct decoder? implements base.io_transformer(
 )
 
 pub func decoder.set_quirk_enabled!(quirk: base.u32, enabled: base.bool) {
-	// TODO: support base.QUIRK_IGNORE_CHECKSUM.
+	// We do not support base.QUIRK_IGNORE_CHECKSUM. Benchmarks suggest that
+	// checksum calculation has an insignificant effect on performance.
 }
 
 pub func decoder.workbuf_len() base.range_ii_u64 {
