diff --git a/internal/cgen/base/pixconv-submodule-ycck.c b/internal/cgen/base/pixconv-submodule-ycck.c
index d528a03f21196485387e59dc54fd9b837268b448..374aacb3dc7a1b2a5825a7c9534a55933cfff3df 100644
--- a/internal/cgen/base/pixconv-submodule-ycck.c
+++ b/internal/cgen/base/pixconv-submodule-ycck.c
@@ -1054,6 +1054,10 @@ wuffs_base__pixel_swizzler__swizzle_ycck(
     //
     // See commits 51bc60ef9298cb2efc1b29a9681191f66d49820d and
     // cd769a0cdf1b5affee13f6089b995f3d39569cb4 for benchmark numbers.
+    //
+    // See also https://godbolt.org/z/MbhbPGEz4 for Debian Bullseye's clang 11
+    // versus gcc 10, where only gcc auto-vectorizes, although later clang
+    // versions will also auto-vectorize.
 #else
     if (wuffs_base__cpu_arch__have_x86_avx2()) {
       upfuncs[1][1] =
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index a062a4f018705a05370fb029538179eec3b07f89..ef658b326a1a6042c8404bbf1a750f49096c7c8d 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -24873,6 +24873,10 @@ wuffs_base__pixel_swizzler__swizzle_ycck(
     //
     // See commits 51bc60ef9298cb2efc1b29a9681191f66d49820d and
     // cd769a0cdf1b5affee13f6089b995f3d39569cb4 for benchmark numbers.
+    //
+    // See also https://godbolt.org/z/MbhbPGEz4 for Debian Bullseye's clang 11
+    // versus gcc 10, where only gcc auto-vectorizes, although later clang
+    // versions will also auto-vectorize.
 #else
     if (wuffs_base__cpu_arch__have_x86_avx2()) {
       upfuncs[1][1] =
