diff --git a/internal/cgen/base/pixconv-submodule-ycck.c b/internal/cgen/base/pixconv-submodule-ycck.c
index 936bb5178b0eb11b100315569a582e1177dfaac3..1278c46be38ace78850a4c8b6b3478e7687da1ee 100644
--- a/internal/cgen/base/pixconv-submodule-ycck.c
+++ b/internal/cgen/base/pixconv-submodule-ycck.c
@@ -1054,6 +1054,9 @@ wuffs_base__pixel_swizzler__swizzle_ycck(
     // some unknown reason, GCC performs noticably better on the non-SIMD
     // version. Possibly because GCC's auto-vectorizer is smarter (just with
     // SSE2, not AVX2) than our hand-written code, but that's just a guess.
+    //
+    // See commits 51bc60ef9298cb2efc1b29a9681191f66d49820d and
+    // cd769a0cdf1b5affee13f6089b995f3d39569cb4 for benchmark numbers.
 #else
     if (wuffs_base__cpu_arch__have_x86_avx2()) {
       upfuncs[1][1] =
diff --git a/release/c/wuffs-unsupported-snapshot.c b/release/c/wuffs-unsupported-snapshot.c
index ecbef92d16a223e68428678fc01063cd08ff2c31..be9a439c353725c8399e092dd8b19ca09d0ed613 100644
--- a/release/c/wuffs-unsupported-snapshot.c
+++ b/release/c/wuffs-unsupported-snapshot.c
@@ -24876,6 +24876,9 @@ wuffs_base__pixel_swizzler__swizzle_ycck(
     // some unknown reason, GCC performs noticably better on the non-SIMD
     // version. Possibly because GCC's auto-vectorizer is smarter (just with
     // SSE2, not AVX2) than our hand-written code, but that's just a guess.
+    //
+    // See commits 51bc60ef9298cb2efc1b29a9681191f66d49820d and
+    // cd769a0cdf1b5affee13f6089b995f3d39569cb4 for benchmark numbers.
 #else
     if (wuffs_base__cpu_arch__have_x86_avx2()) {
       upfuncs[1][1] =
