diff --git a/internal/cgen/base/pixconv-submodule-ycck.c b/internal/cgen/base/pixconv-submodule-ycck.c
index 20af35b961eb5f041e16d1e3cc921ead6a8456cf..05bec489c55b0182febed565605448570772687e 100644
--- a/internal/cgen/base/pixconv-submodule-ycck.c
+++ b/internal/cgen/base/pixconv-submodule-ycck.c
@@ -530,9 +530,7 @@ wuffs_base__pixel_swizzler__swizzle_ycc__general__triangle_filter(
         uint32_t color =                           //
             wuffs_base__color_ycc__as__color_u32(  //
                 *up0++, *up1++, *up2++);
-        // ¡ BEGIN set_color_u32_at
         wuffs_base__pixel_buffer__set_color_u32_at(dst, x, y, color);
-        // ¡ END   set_color_u32_at
         // ¡ dst_iter += 4
       }
     }
@@ -602,9 +600,7 @@ wuffs_base__pixel_swizzler__swizzle_ycc__general__box_filter(
       uint32_t color =                           //
           wuffs_base__color_ycc__as__color_u32(  //
               *src_iter0, *src_iter1, *src_iter2);
-      // ¡ BEGIN set_color_u32_at
       wuffs_base__pixel_buffer__set_color_u32_at(dst, x, y, color);
-      // ¡ END   set_color_u32_at
       // ¡ dst_iter += 4
 
       if ((x + 1u) == width) {
