diff --git a/std/xxhash32/common_xxhash32.wuffs b/std/xxhash32/common_xxhash32.wuffs
index 4739843099a9967089c0dcd56801922fe3ab8949..1c515c958460a4cb01387e40230ca46c0f23d400 100644
--- a/std/xxhash32/common_xxhash32.wuffs
+++ b/std/xxhash32/common_xxhash32.wuffs
@@ -49,8 +49,14 @@ pub func hasher.update!(x: roslice base.u8) {
         this.v1 = INITIAL_V1
         this.v2 = INITIAL_V2
         this.v3 = INITIAL_V3
+
+        // In theory, we could "choose up" here to pick SIMD versions. In
+        // practice, non-SIMD versions are faster (and simpler). For example,
+        // see commit b9b28dbe "std/xxhash32: add hasher.up_x86_sse42".
     }
 
+    // Slicing at 0x100_0000 is arbitrary but ensures that the subslice's
+    // length (used in hasher.up) doesn't overflow a u32.
     while args.x.length() > 0 {
         remaining = args.x[.. 0]
         if args.x.length() > 0x100_0000 {
diff --git a/std/xxhash64/common_xxhash64.wuffs b/std/xxhash64/common_xxhash64.wuffs
index 7047fcc4c0ffdadc63c55f77f95bbe103f946cb7..0d2c0d0dd81017d7753c8c6e2eb9c2f9d0f05d34 100644
--- a/std/xxhash64/common_xxhash64.wuffs
+++ b/std/xxhash64/common_xxhash64.wuffs
@@ -48,6 +48,10 @@ pub func hasher.update!(x: roslice base.u8) {
         this.v1 = INITIAL_V1
         this.v2 = INITIAL_V2
         this.v3 = INITIAL_V3
+
+        // In theory, we could "choose up" here to pick SIMD versions. In
+        // practice, non-SIMD versions are faster (and simpler). For example,
+        // see commit b9b28dbe "std/xxhash32: add hasher.up_x86_sse42".
     }
 
     this.up!(x: args.x)
