diff --git a/std/jpeg/decode_mcu_default.wuffs b/std/jpeg/decode_mcu_default.wuffs
index 1dac5211fd79046fcc8590aafba23d324be73267..2b7cd2fe0d6d74d2fabdb312e65e1ca7cb0502ac 100644
--- a/std/jpeg/decode_mcu_default.wuffs
+++ b/std/jpeg/decode_mcu_default.wuffs
@@ -75,7 +75,7 @@ pri func decoder.decode_mcu!(workbuf: slice base.u8, mx: base.u32[..= 0x1FFF], m
             {
                 this.mcu_blocks[.. 1].bulk_memset!(byte_value: 0)
 
-                // Load at least 31 bits, by keeping 56 .. 64 loaded.
+                // Load at least 47 bits, by keeping 56 .. 64 loaded.
                 if r.length() < 8 {
                     ret = 1  // Request another fill_bitstream call.
                     break.goto_done
@@ -149,8 +149,16 @@ pri func decoder.decode_mcu!(workbuf: slice base.u8, mx: base.u32[..= 0x1FFF], m
                 break.goto_done
             }
 
+            // Ensure the informal "the lower bound on the number of loaded
+            // bits is at least 16" loop pre-condition below.
             if n_bits < 16 {
                 bits |= r.peek_u64be() >> (n_bits & 63)
+                // Skip these lines that are normally part of "load more bits".
+                // It's unnecessary, just for the informal (NB > 16) below, and
+                // skipping it avoids a small but negative performance impact.
+                //
+                // r.skip_u32_fast!( etc )
+                // n_bits |= 56
             }
 
             z = 1
@@ -161,9 +169,32 @@ pri func decoder.decode_mcu!(workbuf: slice base.u8, mx: base.u32[..= 0x1FFF], m
                     inv this.mcu_current_block < 10,
                     inv ac_huff_table_fast <> nullptr,
             {
+                // Use the high bits of the bits variable to look up the
+                // Huffman table. Conceptually, we should do this *after*
+                // "Load at least 47 bits" below (at "LOOK-UP" further below),
+                // but it is noticably faster to do it beforehand:
+                // https://github.com/google/wuffs/commit/d9c7740ea6c4a450401c393787838436895b87bb
+                //
+                // Earlier look-up is OK because there is a informal loop
+                // pre-condition (informal means not enforced by the Wuffs
+                // compiler) that (NB >= 16) here, where NB is "the number of
+                // loaded bits in the bits variable". It is literally n_bits
+                // for almost all loop iterations, but it can be higher on the
+                // first iteration, set up by "Ensure the informal" above.
+                //
+                // For subsequent iterations, the "Load at least 47 bits" below
+                // does what it says. The Huffman code consumes up to 16 bits
+                // and processing the Huffman symbol's ssss nibble consumes up
+                // to 15 bits. (47 - 16 - 15) = 16 so at the end of the loop
+                // body, (n_bits >= 16).
+                //
+                // This 16 bound has some slack, as (NB >= (64 - 56)) is all we
+                // need, but a bound of 16 (the longest JPEG Huffman code bit
+                // length) won't need to change even if, in the future, we
+                // enlarge the look-up table size.
                 ac_ht_fast = ac_huff_table_fast[bits >> 56] as base.u32
 
-                // Load at least 31 bits, by keeping 56 .. 64 loaded.
+                // Load at least 47 bits, by keeping 56 .. 64 loaded.
                 //
                 // TODO: eliminate this bounds check entirely.
                 if r.length() < 8 {
@@ -175,6 +206,9 @@ pri func decoder.decode_mcu!(workbuf: slice base.u8, mx: base.u32[..= 0x1FFF], m
                 n_bits |= 56
 
                 // Read the Huffman-encoded ac_symbol, up to 16 bits long.
+                //
+                // LOOK-UP: "ac_ht_fast = etc" conceptually happens here, just
+                // before deriving ac_bl from ac_ht_fast.
                 ac_bl = ac_ht_fast >> 8
                 if n_bits >= ac_bl {
                     ac_symbol = 0xFF & ac_ht_fast
diff --git a/std/jpeg/decode_mcu_progressive_ac_high_bits.wuffs b/std/jpeg/decode_mcu_progressive_ac_high_bits.wuffs
index 2d41c436696767a0ded13779ce79126f34506def..cb96998303d8d31efb713a2989c35929fe1e5ad2 100644
--- a/std/jpeg/decode_mcu_progressive_ac_high_bits.wuffs
+++ b/std/jpeg/decode_mcu_progressive_ac_high_bits.wuffs
@@ -65,8 +65,16 @@ pri func decoder.decode_mcu_progressive_ac_high_bits!(workbuf: slice base.u8, mx
                 break.goto_done
             }
 
+            // Ensure the informal "the lower bound on the number of loaded
+            // bits is at least 16" loop pre-condition below.
             if n_bits < 16 {
                 bits |= r.peek_u64be() >> (n_bits & 63)
+                // Skip these lines that are normally part of "load more bits".
+                // It's unnecessary, just for the informal (NB > 16) below, and
+                // skipping it avoids a small but negative performance impact.
+                //
+                // r.skip_u32_fast!( etc )
+                // n_bits |= 56
             }
 
             z = this.mcu_zig_index
@@ -78,9 +86,32 @@ pri func decoder.decode_mcu_progressive_ac_high_bits!(workbuf: slice base.u8, mx
             {
                 assert z < 64 via "a < b: a <= c; c < b"(c: (this.scan_se as base.u32))
 
+                // Use the high bits of the bits variable to look up the
+                // Huffman table. Conceptually, we should do this *after*
+                // "Load at least 47 bits" below (at "LOOK-UP" further below),
+                // but it is noticably faster to do it beforehand:
+                // https://github.com/google/wuffs/commit/d9c7740ea6c4a450401c393787838436895b87bb
+                //
+                // Earlier look-up is OK because there is a informal loop
+                // pre-condition (informal means not enforced by the Wuffs
+                // compiler) that (NB >= 16) here, where NB is "the number of
+                // loaded bits in the bits variable". It is literally n_bits
+                // for almost all loop iterations, but it can be higher on the
+                // first iteration, set up by "Ensure the informal" above.
+                //
+                // For subsequent iterations, the "Load at least 47 bits" below
+                // does what it says. The Huffman code consumes up to 16 bits
+                // and processing the Huffman symbol's ssss nibble consumes up
+                // to 15 bits. (47 - 16 - 15) = 16 so at the end of the loop
+                // body, (n_bits >= 16).
+                //
+                // This 16 bound has some slack, as (NB >= (64 - 56)) is all we
+                // need, but a bound of 16 (the longest JPEG Huffman code bit
+                // length) won't need to change even if, in the future, we
+                // enlarge the look-up table size.
                 ac_ht_fast = ac_huff_table_fast[bits >> 56] as base.u32
 
-                // Load at least 31 bits, by keeping 56 .. 64 loaded.
+                // Load at least 47 bits, by keeping 56 .. 64 loaded.
                 //
                 // TODO: eliminate this bounds check entirely.
                 if r.length() < 8 {
@@ -92,6 +123,9 @@ pri func decoder.decode_mcu_progressive_ac_high_bits!(workbuf: slice base.u8, mx
                 n_bits |= 56
 
                 // Read the Huffman-encoded ac_symbol, up to 16 bits long.
+                //
+                // LOOK-UP: "ac_ht_fast = etc" conceptually happens here, just
+                // before deriving ac_bl from ac_ht_fast.
                 ac_bl = ac_ht_fast >> 8
                 if n_bits >= ac_bl {
                     ac_symbol = 0xFF & ac_ht_fast
diff --git a/std/jpeg/decode_mcu_progressive_ac_low_bit.wuffs b/std/jpeg/decode_mcu_progressive_ac_low_bit.wuffs
index e7c62a5dca83b15424cdb860fd67746370b45d8b..429180b077805f6aab12a7bdf47ca27ed20b975e 100644
--- a/std/jpeg/decode_mcu_progressive_ac_low_bit.wuffs
+++ b/std/jpeg/decode_mcu_progressive_ac_low_bit.wuffs
@@ -62,7 +62,7 @@ pri func decoder.decode_mcu_progressive_ac_low_bit!(workbuf: slice base.u8, mx:
             while.ac_components true,
                     inv ac_huff_table_fast <> nullptr,
             {
-                // Load at least 31 bits, by keeping 56 .. 64 loaded (call it 8
+                // Load at least 47 bits, by keeping 56 .. 64 loaded (call it 8
                 // bytes). Just like the other decode_mcu_etc functions, this
                 // is enough for the Huffman-encoded ac_symbol (up to 16 bits)
                 // and processing the ac_value (up to 15 bits).
diff --git a/std/jpeg/decode_mcu_progressive_dc_high_bits.wuffs b/std/jpeg/decode_mcu_progressive_dc_high_bits.wuffs
index c309154a737a74a41964ff73a7ff939da5035ba1..73a324cfc7290313ee7244f81fd200278c9ec62a 100644
--- a/std/jpeg/decode_mcu_progressive_dc_high_bits.wuffs
+++ b/std/jpeg/decode_mcu_progressive_dc_high_bits.wuffs
@@ -47,7 +47,7 @@ pri func decoder.decode_mcu_progressive_dc_high_bits!(workbuf: slice base.u8, mx
             while.dc_component true,
                     inv this.mcu_current_block < 10,
             {
-                // Load at least 31 bits, by keeping 56 .. 64 loaded.
+                // Load at least 47 bits, by keeping 56 .. 64 loaded.
                 if r.length() < 8 {
                     ret = 1  // Request another fill_bitstream call.
                     break.goto_done
diff --git a/std/jpeg/decode_mcu_progressive_dc_low_bit.wuffs b/std/jpeg/decode_mcu_progressive_dc_low_bit.wuffs
index f9cb54125310f37f9679ec94cae278f6cf9b04a8..29a16784b7d8f90c361f05808e4d37580a575165 100644
--- a/std/jpeg/decode_mcu_progressive_dc_low_bit.wuffs
+++ b/std/jpeg/decode_mcu_progressive_dc_low_bit.wuffs
@@ -39,7 +39,7 @@ pri func decoder.decode_mcu_progressive_dc_low_bit!(workbuf: slice base.u8, mx:
             while.dc_component true,
                     inv this.mcu_current_block < 10,
             {
-                // Load at least 31 bits, by keeping 56 .. 64 loaded.
+                // Load at least 47 bits, by keeping 56 .. 64 loaded.
                 if r.length() < 8 {
                     ret = 1  // Request another fill_bitstream call.
                     break.goto_done
