diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 1b908730..733caa89 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -47,7 +47,7 @@ const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
   // Use a {} code block so that "goto exit" doesn't trigger "jump bypasses
   // variable initialization" warnings.
   {
-    wuffs_gif__status s = 0;
+    wuffs_base__status s = 0;
     wuffs_gif__decoder dec = ((wuffs_gif__decoder){});
     wuffs_gif__decoder__check_wuffs_version(&dec, sizeof dec, WUFFS_VERSION);
 
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index 64deaf61..5890618b 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -43,7 +43,7 @@ It should print "PASS", amongst other information, and exit(0).
 
 const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
   const char* ret = NULL;
-  wuffs_zlib__status s = 0;
+  wuffs_base__status s = 0;
   wuffs_zlib__decoder dec = ((wuffs_zlib__decoder){});
   wuffs_zlib__decoder__check_wuffs_version(&dec, sizeof dec, WUFFS_VERSION);
