diff --git a/src/readbarcode.c b/src/readbarcode.c
index 77c6774..71257be 100644
--- a/src/readbarcode.c
+++ b/src/readbarcode.c
@@ -858,7 +858,7 @@ numaQuantizeCrossingsByWidth(NUMA       *nas,
                              NUMA      **pnaohist,
                              l_int32     debugflag)
 {
-l_int32    i, n, ned, nod, iw, width;
+l_int32    i, n, ret, ned, nod, iw, width;
 l_float32  val, minsize, maxsize, factor;
 GPLOT     *gplot;
 NUMA      *naedist, *naodist, *naehist, *naohist, *naecent, *naocent;
@@ -875,7 +875,9 @@ NUMA      *naerange, *naorange, *naelut, *naolut, *nad;
         return (NUMA *)ERROR_PTR("binfract <= 0.0", procName, NULL);
 
         /* Get even and odd crossing distances */
-    numaGetCrossingDistances(nas, &naedist, &naodist, &minsize, &maxsize);
+    ret = numaGetCrossingDistances(nas, &naedist, &naodist, &minsize, &maxsize);
+    if (ret)
+        return (NUMA *)ERROR_PTR("crossing data not found", procName, NULL);
 
         /* Bin the spans in units of binfract * minsize.  These
          * units are convenient because they scale to make at least
diff --git a/src/readfile.c b/src/readfile.c
index f37bf12..51e58bf 100644
--- a/src/readfile.c
+++ b/src/readfile.c
@@ -633,7 +633,7 @@ l_int32  format;
 
     if (fread(&firstbytes, 1, 12, fp) != 12)
         return ERROR_INT("failed to read first 12 bytes of file", procName, 1);
-    firstbytes[12] = '\0';
+    firstbytes[12] = 0;
     rewind(fp);
 
     findFileFormatBuffer(firstbytes, &format);
