diff --git a/ares_parse_a_reply.c b/ares_parse_a_reply.c
--- a/ares_parse_a_reply.c
+++ b/ares_parse_a_reply.c
@@ -122,12 +122,17 @@ int ares_parse_a_reply(const unsigned char *abuf, int alen,
   hostent->h_addr_list = ares_malloc((naddrs + 1) * sizeof(char *));
   if (!hostent->h_addr_list)
     {
       goto enomem;
     }
 
+  for (i = 0; i < naddrs + 1; ++i)
+    {
+      hostent->h_addr_list[i] = NULL;
+    }
+
   if (ai.cnames)
     {
       hostent->h_name = strdup(ai.cnames->name);
       ares_free(question_hostname);
     }
   else
@@ -144,32 +149,39 @@ int ares_parse_a_reply(const unsigned char *abuf, int alen,
       addrs = ares_malloc(naddrs * sizeof(struct in_addr));
       if (!addrs)
         {
           goto enomem;
         }
 
+      i = 0;
       next = ai.nodes;
-      for (i = 0; i < naddrs; i++)
+      while (next)
         {
-          hostent->h_addr_list[i] = (char*)&addrs[i];
-          memcpy(hostent->h_addr_list[i], &(((struct sockaddr_in *)next->ai_addr)->sin_addr), sizeof(struct in_addr));
-          if (naddrttls)
+          if (next->ai_family == AF_INET)
             {
-                if(next->ai_ttl > cname_ttl)
-                  addrttls[i].ttl = cname_ttl;
-                else
-                  addrttls[i].ttl = next->ai_ttl;
-
-                memcpy(&addrttls[i].ipaddr, &(((struct sockaddr_in *)next->ai_addr)->sin_addr), sizeof(struct in_addr));
+              hostent->h_addr_list[i] = (char *)&addrs[i];
+              memcpy(hostent->h_addr_list[i],
+                     &(((struct sockaddr_in *)next->ai_addr)->sin_addr),
+                     sizeof(struct in_addr));
+              if (naddrttls)
+                {
+                  if (next->ai_ttl > cname_ttl)
+                    addrttls[i].ttl = cname_ttl;
+                  else
+                    addrttls[i].ttl = next->ai_ttl;
+
+                  memcpy(&addrttls[i].ipaddr,
+                         &(((struct sockaddr_in *)next->ai_addr)->sin_addr),
+                         sizeof(struct in_addr));
+                }
+              ++i;
             }
           next = next->ai_next;
         }
     }
 
-  hostent->h_addr_list[naddrs] = NULL;
-
   if (host)
     {
       *host = hostent;
     }
   else
     {
diff --git a/ares_parse_aaaa_reply.c b/ares_parse_aaaa_reply.c
--- a/ares_parse_aaaa_reply.c
+++ b/ares_parse_aaaa_reply.c
@@ -124,12 +124,17 @@ int ares_parse_aaaa_reply(const unsigned char *abuf, int alen,
   hostent->h_addr_list = ares_malloc((naddrs + 1) * sizeof(char *));
   if (!hostent->h_addr_list)
     {
       goto enomem;
     }
 
+  for (i = 0; i < naddrs + 1; ++i)
+    {
+      hostent->h_addr_list[i] = NULL;
+    }
+
   if (ai.cnames)
     {
       hostent->h_name = strdup(ai.cnames->name);
       ares_free(question_hostname);
     }
   else
@@ -146,32 +151,39 @@ int ares_parse_aaaa_reply(const unsigned char *abuf, int alen,
       addrs = ares_malloc(naddrs * sizeof(struct ares_in6_addr));
       if (!addrs)
         {
           goto enomem;
         }
 
+      i = 0;
       next = ai.nodes;
-      for (i = 0; i < naddrs; i++)
+      while (next)
         {
-          hostent->h_addr_list[i] = (char*)&addrs[i];
-          memcpy(hostent->h_addr_list[i], &(((struct sockaddr_in6 *)next->ai_addr)->sin6_addr), sizeof(struct ares_in6_addr));
-          if (naddrttls)
+          if(next->ai_family == AF_INET6)
             {
-                if(next->ai_ttl > cname_ttl)
-                  addrttls[i].ttl = cname_ttl;
-                else
-                  addrttls[i].ttl = next->ai_ttl;
-
-                memcpy(&addrttls[i].ip6addr, &(((struct sockaddr_in6 *)next->ai_addr)->sin6_addr), sizeof(struct ares_in6_addr));
+              hostent->h_addr_list[i] = (char*)&addrs[i];
+              memcpy(hostent->h_addr_list[i],
+                     &(((struct sockaddr_in6 *)next->ai_addr)->sin6_addr),
+                     sizeof(struct ares_in6_addr));
+              if (naddrttls)
+                {
+                    if(next->ai_ttl > cname_ttl)
+                      addrttls[i].ttl = cname_ttl;
+                    else
+                      addrttls[i].ttl = next->ai_ttl;
+
+                    memcpy(&addrttls[i].ip6addr,
+                           &(((struct sockaddr_in6 *)next->ai_addr)->sin6_addr),
+                           sizeof(struct ares_in6_addr));
+                }
+              ++i;
             }
           next = next->ai_next;
         }
     }
 
-  hostent->h_addr_list[naddrs] = NULL;
-
   if (host)
     {
       *host = hostent;
     }
   else
     {
