diff --git a/prog/fuzzing/dewarp_fuzzer.cc b/prog/fuzzing/dewarp_fuzzer.cc
index 9ea815a..37098a9 100644
--- a/prog/fuzzing/dewarp_fuzzer.cc
+++ b/prog/fuzzing/dewarp_fuzzer.cc
@@ -15,7 +15,8 @@ LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
     pixs = pixReadMemSpix(data, size);
     if(pixs==NULL) return 0;
     
-    dewarpSinglePage(pixs, 0, 1, 1, 0, &pixd, NULL, 1);
+    // Don't use debug, because it requires writing to /tmp
+    dewarpSinglePage(pixs, 0, 1, 1, 0, &pixd, NULL, 0);
 	
     pixac = pixacompReadMem(data, size);
     dewa1 = dewarpaCreateFromPixacomp(pixac, 1, 0, 10, -1);
