diff --git a/src/dump.c b/src/dump.c
--- a/src/dump.c
+++ b/src/dump.c
@@ -293,14 +293,13 @@ static int do_dump(const json_t *json, size_t flags, int depth,
 
         case JSON_OBJECT:
         {
             void *iter;
             const char *separator;
             int separator_length;
-            /* Space for "0x", double the sizeof a pointer for the hex and a terminator. */
-            char loop_key[2 + (sizeof(json) * 2) + 1];
+            char loop_key[LOOP_KEY_LEN];
 
             if(flags & JSON_COMPACT) {
                 separator = ":";
                 separator_length = 1;
             }
             else {
diff --git a/src/jansson_private.h b/src/jansson_private.h
--- a/src/jansson_private.h
+++ b/src/jansson_private.h
@@ -88,12 +88,14 @@ void* jsonp_malloc(size_t size) JANSSON_ATTRS(warn_unused_result);
 void jsonp_free(void *ptr);
 char *jsonp_strndup(const char *str, size_t length) JANSSON_ATTRS(warn_unused_result);
 char *jsonp_strdup(const char *str) JANSSON_ATTRS(warn_unused_result);
 char *jsonp_strndup(const char *str, size_t len) JANSSON_ATTRS(warn_unused_result);
 
 /* Circular reference check*/
+/* Space for "0x", double the sizeof a pointer for the hex and a terminator. */
+#define LOOP_KEY_LEN (2 + (sizeof(json_t *) * 2) + 1)
 int jsonp_loop_check(hashtable_t *parents, const json_t *json, char *key, size_t key_size);
 
 
 /* Windows compatibility */
 #if defined(_WIN32) || defined(WIN32)
 #  if defined(_MSC_VER)  /* MS compiller */
diff --git a/src/value.c b/src/value.c
--- a/src/value.c
+++ b/src/value.c
@@ -319,16 +319,16 @@ static json_t *json_object_copy(json_t *object)
 }
 
 static json_t *json_object_deep_copy(const json_t *object, hashtable_t *parents)
 {
     json_t *result;
     void *iter;
-    char loop_key[2 + (sizeof(object) * 2) + 1];
+    char loop_key[LOOP_KEY_LEN];
 
     if (jsonp_loop_check(parents, object, loop_key, sizeof(loop_key)))
-            return NULL;
+        return NULL;
 
     result = json_object();
     if(!result)
         return NULL;
 
     /* Cannot use json_object_foreach because object has to be cast
@@ -638,13 +638,13 @@ static json_t *json_array_copy(json_t *array)
 }
 
 static json_t *json_array_deep_copy(const json_t *array, hashtable_t *parents)
 {
     json_t *result;
     size_t i;
-    char loop_key[2 + (sizeof(array) * 2) + 1];
+    char loop_key[LOOP_KEY_LEN];
 
     if (jsonp_loop_check(parents, array, loop_key, sizeof(loop_key)))
         return NULL;
 
     result = json_array();
     if(!result)
