diff --git a/src/tiffio.c b/src/tiffio.c
--- a/src/tiffio.c
+++ b/src/tiffio.c
@@ -1760,14 +1760,14 @@ l_float32  fxres, fyres;
         return ERROR_INT("fxres and/or fyres values are invalid", procName, 1);
 
     if (resunit == RESUNIT_CENTIMETER) {  /* convert to ppi */
         *pxres = (l_int32)(2.54 * fxres + 0.5);
         *pyres = (l_int32)(2.54 * fyres + 0.5);
     } else {
-        *pxres = (l_int32)fxres;
-        *pyres = (l_int32)fyres;
+        *pxres = (l_int32)(fxres + 0.5);
+        *pyres = (l_int32)(fyres + 0.5);
     }
 
     return 0;
 }
