diff --git a/test/ares-fuzz.c b/test/ares-fuzz.c
index 82bbf35..bdaa4d6 100644
--- a/test/ares-fuzz.c
+++ b/test/ares-fuzz.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) Brad House
+ * Copyright (C) The c-ares project
  *
  * Permission to use, copy, modify, and distribute this
  * software and its documentation for any purpose and without
@@ -31,6 +31,8 @@
 #include <unistd.h>
 #endif
 
+#include "ares.h"
+
 #define kMaxAflInputSize (1 << 20)
 static unsigned char afl_buffer[kMaxAflInputSize];
 
@@ -46,15 +48,17 @@ static unsigned char afl_buffer[kMaxAflInputSize];
 int LLVMFuzzerTestOneInput(const unsigned char *data, unsigned long size);
 
 static void ProcessFile(int fd) {
-  int count = read(fd, afl_buffer, kMaxAflInputSize);
+  ares_ssize_t count = read(fd, afl_buffer, kMaxAflInputSize);
   /*
    * Make a copy of the data so that it's not part of a larger
    * buffer (where buffer overflows would go unnoticed).
    */
-  unsigned char *copied_data = (unsigned char *)malloc(count);
-  memcpy(copied_data, afl_buffer, count);
-  LLVMFuzzerTestOneInput(copied_data, count);
-  free(copied_data);
+  if (count > 0) {
+    unsigned char *copied_data = (unsigned char *)malloc((size_t)count);
+    memcpy(copied_data, afl_buffer, (size_t)count);
+    LLVMFuzzerTestOneInput(copied_data, (size_t)count);
+    free(copied_data);
+  }
 }
 
 int main(int argc, char *argv[]) {
diff --git a/test/ares-test-fuzz-name.c b/test/ares-test-fuzz-name.c
index 378ecb4..3c4f17b 100644
--- a/test/ares-test-fuzz-name.c
+++ b/test/ares-test-fuzz-name.c
@@ -23,16 +23,18 @@
 // Include ares internal file for DNS protocol constants
 #include "ares_nameser.h"
 
+int LLVMFuzzerTestOneInput(const unsigned char *data, unsigned long size);
+
 // Entrypoint for Clang's libfuzzer, exercising query creation.
 int LLVMFuzzerTestOneInput(const unsigned char *data,
                            unsigned long size) {
   // Null terminate the data.
   char *name = malloc(size + 1);
+  unsigned char *buf = NULL;
+  int buflen = 0;
   name[size] = '\0';
   memcpy(name, data, size);
 
-  unsigned char *buf = NULL;
-  int buflen = 0;
   ares_create_query(name, C_IN, T_AAAA, 1234, 0, &buf, &buflen, 1024);
   free(buf);
   free(name);
diff --git a/test/ares-test-fuzz.c b/test/ares-test-fuzz.c
index 1c0f179..a57f28c 100644
--- a/test/ares-test-fuzz.c
+++ b/test/ares-test-fuzz.c
@@ -19,57 +19,60 @@
 
 #include "ares.h"
 
+int LLVMFuzzerTestOneInput(const unsigned char *data, unsigned long size);
+
+
 // Entrypoint for Clang's libfuzzer
-int LLVMFuzzerTestOneInput(const unsigned char *data,
-                           unsigned long size) {
+int LLVMFuzzerTestOneInput(const unsigned char *data, unsigned long size)
+{
   // Feed the data into each of the ares_parse_*_reply functions.
   struct hostent *host = NULL;
   struct ares_addrttl info[5];
+  struct ares_addr6ttl info6[5];
+  unsigned char addrv4[4] = {0x10, 0x20, 0x30, 0x40};
+  struct ares_srv_reply* srv = NULL;
+  struct ares_mx_reply* mx = NULL;
+  struct ares_txt_reply* txt = NULL;
+  struct ares_soa_reply* soa = NULL;
+  struct ares_naptr_reply* naptr = NULL;
+  struct ares_caa_reply* caa = NULL;
+  struct ares_uri_reply* uri = NULL;
   int count = 5;
-  ares_parse_a_reply(data, size, &host, info, &count);
+  ares_parse_a_reply(data, (int)size, &host, info, &count);
   if (host) ares_free_hostent(host);
 
   host = NULL;
-  struct ares_addr6ttl info6[5];
   count = 5;
-  ares_parse_aaaa_reply(data, size, &host, info6, &count);
+  ares_parse_aaaa_reply(data, (int)size, &host, info6, &count);
   if (host) ares_free_hostent(host);
 
   host = NULL;
-  unsigned char addrv4[4] = {0x10, 0x20, 0x30, 0x40};
-  ares_parse_ptr_reply(data, size, addrv4, sizeof(addrv4), AF_INET, &host);
+  ares_parse_ptr_reply(data, (int)size, addrv4, sizeof(addrv4), AF_INET, &host);
   if (host) ares_free_hostent(host);
 
   host = NULL;
-  ares_parse_ns_reply(data, size, &host);
+  ares_parse_ns_reply(data, (int)size, &host);
   if (host) ares_free_hostent(host);
 
-  struct ares_srv_reply* srv = NULL;
-  ares_parse_srv_reply(data, size, &srv);
+  ares_parse_srv_reply(data, (int)size, &srv);
   if (srv) ares_free_data(srv);
 
-  struct ares_mx_reply* mx = NULL;
-  ares_parse_mx_reply(data, size, &mx);
+  ares_parse_mx_reply(data, (int)size, &mx);
   if (mx) ares_free_data(mx);
 
-  struct ares_txt_reply* txt = NULL;
-  ares_parse_txt_reply(data, size, &txt);
+  ares_parse_txt_reply(data, (int)size, &txt);
   if (txt) ares_free_data(txt);
 
-  struct ares_soa_reply* soa = NULL;
-  ares_parse_soa_reply(data, size, &soa);
+  ares_parse_soa_reply(data, (int)size, &soa);
   if (soa) ares_free_data(soa);
 
-  struct ares_naptr_reply* naptr = NULL;
-  ares_parse_naptr_reply(data, size, &naptr);
+  ares_parse_naptr_reply(data, (int)size, &naptr);
   if (naptr) ares_free_data(naptr);
 
-  struct ares_caa_reply* caa = NULL;
-  ares_parse_caa_reply(data, size, &caa);
+  ares_parse_caa_reply(data, (int)size, &caa);
   if (caa) ares_free_data(caa);
 
-  struct ares_uri_reply* uri = NULL;
-  ares_parse_uri_reply(data, size, &uri);
+  ares_parse_uri_reply(data, (int)size, &uri);
   if (uri) ares_free_data(uri);
 
   return 0;
