diff --git a/src/config.c b/src/config.c
--- a/src/config.c
+++ b/src/config.c
@@ -1288,12 +1288,18 @@ void TY_(AdjustConfig)( TidyDocImpl* doc )
         if ( enc == UTF16LE || enc == UTF16BE || enc == UTF16 )
             TY_(SetOptionInt)( doc, TidyOutputBOM, yes );
         
         TY_(SetOptionBool)( doc, TidyQuoteAmpersand, yes );
         TY_(SetOptionBool)( doc, TidyOmitOptionalTags, no );
     }
+    
+    /* Setup the indent character. */
+    if cfgBool(doc, TidyPPrintTabs) 
+        doc->indent_char = '\t';
+    else
+        doc->indent_char = ' ';
 }
 
 
 /* A service to ParseList(), keeps option values nicely formatted and
    coordinates additions to the internal lists. Within Tidy, this function
    might be used to programmatically add individual values to items that use
diff --git a/src/tidylib.c b/src/tidylib.c
--- a/src/tidylib.c
+++ b/src/tidylib.c
@@ -2250,22 +2250,17 @@ int         tidyDocSaveStream( TidyDocImpl* doc, StreamOut* out )
 
     Bool dropComments = cfgBool(doc, TidyHideComments);
     Bool makeClean    = cfgBool(doc, TidyMakeClean);
     Bool asciiChars   = cfgBool(doc, TidyAsciiChars);
     Bool makeBare     = cfgBool(doc, TidyMakeBare);
     Bool escapeCDATA  = cfgBool(doc, TidyEscapeCdata);
-    Bool ppWithTabs   = cfgBool(doc, TidyPPrintTabs);
     TidyAttrSortStrategy sortAttrStrat = cfg(doc, TidySortAttributes);
     TidyConfigChangeCallback callback = doc->pConfigChangeCallback;
     doc->pConfigChangeCallback = NULL;
 
-    if (ppWithTabs)
-        doc->indent_char = '\t';
-    else
-        doc->indent_char = ' ';
-
+    
     if (escapeCDATA)
         TY_(ConvertCDATANodes)(doc, &doc->root);
 
     if (dropComments)
         TY_(DropComments)(doc, &doc->root);
