diff --git a/fuzz/c/fuzzlib/fuzzlib.c b/fuzz/c/fuzzlib/fuzzlib.c
index 761c289d..a74c55ce 100644
--- a/fuzz/c/fuzzlib/fuzzlib.c
+++ b/fuzz/c/fuzzlib/fuzzlib.c
@@ -41,11 +41,18 @@ static const char* llvmFuzzerTestOneInput(const uint8_t* data, size_t size) {
   hash ^= hash >> 11;
   hash += hash << 15;
 
-  wuffs_base__io_buffer src = {.ptr = (uint8_t*)(data),
-                               .len = size,
-                               .wi = size,
-                               .ri = 0,
-                               .closed = true};
+  wuffs_base__io_buffer src = ((wuffs_base__io_buffer){
+      .data = ((wuffs_base__slice_u8){
+          .ptr = (uint8_t*)(data),
+          .len = size,
+      }),
+      .meta = ((wuffs_base__io_buffer_meta){
+          .wi = size,
+          .ri = 0,
+          .pos = 0,
+          .closed = true,
+      }),
+  });
   return fuzz(wuffs_base__io_buffer__reader(&src), hash);
 }
 
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index 2e34a166..bb8e8d40 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -61,16 +61,21 @@ const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
 
   const size_t dstbuf_size = 65536;
   uint8_t dstbuf[dstbuf_size];
-  wuffs_base__io_buffer dst = {.ptr = dstbuf, .len = dstbuf_size};
+  wuffs_base__io_buffer dst = ((wuffs_base__io_buffer){
+      .data = ((wuffs_base__slice_u8){
+          .ptr = dstbuf,
+          .len = dstbuf_size,
+      }),
+  });
   wuffs_base__io_writer dst_writer = wuffs_base__io_buffer__writer(&dst);
 
   while (true) {
-    dst.wi = 0;
+    dst.meta.wi = 0;
     z = wuffs_zlib__decoder__decode(&dec, dst_writer, src_reader);
     if (z != wuffs_base__suspension__short_write) {
       break;
     }
-    if (dst.wi == 0) {
+    if (dst.meta.wi == 0) {
       fprintf(stderr, "wuffs_zlib__decoder__decode made no progress\n");
       intentional_segfault();
     }
