diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 8d7bb06f..18790876 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -47,7 +47,7 @@ const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
   {
     wuffs_gif__status s = 0;
     wuffs_gif__decoder dec = ((wuffs_gif__decoder){});
-    wuffs_gif__decoder__check_wuffs_version(&dec, WUFFS_VERSION, sizeof dec);
+    wuffs_gif__decoder__check_wuffs_version(&dec, sizeof dec, WUFFS_VERSION);
 
     wuffs_base__image_buffer ib = ((wuffs_base__image_buffer){});
     wuffs_base__image_config ic = ((wuffs_base__image_config){});
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index ba871516..e6ab0926 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -44,7 +44,7 @@ const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
   const char* ret = NULL;
   wuffs_zlib__status s = 0;
   wuffs_zlib__decoder dec = ((wuffs_zlib__decoder){});
-  wuffs_zlib__decoder__check_wuffs_version(&dec, WUFFS_VERSION, sizeof dec);
+  wuffs_zlib__decoder__check_wuffs_version(&dec, sizeof dec, WUFFS_VERSION);
 
   // Ignore the checksum for 99.99%-ish of all input. When fuzzers generate
   // random input, the checkum is very unlikely to match. Still, it's useful to
