diff --git a/prog/fuzzing/adaptmap_fuzzer.cc b/prog/fuzzing/adaptmap_fuzzer.cc
index 85f5014..f1ba9ff 100644
--- a/prog/fuzzing/adaptmap_fuzzer.cc
+++ b/prog/fuzzing/adaptmap_fuzzer.cc
@@ -19,8 +19,6 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	pixDestroy(&pix2);
 	pixDestroy(&payload_copy);
 
-
-
 	pix1 = pixRead("../test8.jpg");
 	payload_copy = pixCopy(NULL, pixs_payload);
 	pixBackgroundNormGrayArrayMorph(payload_copy, pix1, 6, 5, 256, &pix2);
@@ -28,7 +26,6 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	pixDestroy(&pix2);
 	pixDestroy(&payload_copy);
 
-
 	pix1 = pixRead("../test8.jpg");
 	payload_copy = pixCopy(NULL, pixs_payload);
 	return_pix1 = pixBackgroundNormMorph(payload_copy, pix1, 6, 5, 256);
@@ -38,7 +35,6 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 		pixDestroy(&return_pix1);
 	}
 
-
 	pix1 = pixRead("../test8.jpg");
 	pix2 = pixRead("../test8.jpg");
 	payload_copy = pixCopy(NULL, pixs_payload);
@@ -50,7 +46,6 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	pixDestroy(&pix5);
 	pixDestroy(&payload_copy);
 
-
 	pix1 = pixRead("../test8.jpg");
 	payload_copy = pixCopy(NULL, pixs_payload);
 	pixBackgroundNormRGBArraysMorph(payload_copy, pix1, 6, 33, 130, &pix2, &pix3, &pix4);
@@ -60,14 +55,10 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	pixDestroy(&pix4);
 	pixDestroy(&payload_copy);
 
-
-	pix1 = pixRead("../test8.jpg");
 	payload_copy = pixCopy(NULL, pixs_payload);
 	pixContrastNorm(payload_copy, payload_copy, 10, 10, 3, 0, 0);
-	pixDestroy(&pix1);
 	pixDestroy(&payload_copy);
 
-
 	pix1 = pixRead("../test8.jpg");
 	payload_copy = pixCopy(NULL, pixs_payload);
 	return_pix1 = pixGlobalNormNoSatRGB(payload_copy, pix1, 3, 3, 3, 2, 0.9);
@@ -77,7 +68,6 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 		pixDestroy(&return_pix1);
 	}
 
-
 	payload_copy = pixCopy(NULL, pixs_payload);
 	pixThresholdSpreadNorm(payload_copy, L_SOBEL_EDGE, 10, 0, 0, 0.7, -25, 255, 10, &pix1, &pix2, &pix3);
 	pixDestroy(&pix1);
@@ -86,7 +76,5 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	pixDestroy(&payload_copy);
 
 	pixDestroy(&pixs_payload);
-
-
 	return 0;
 }
