diff --git a/test/ares-fuzz.c b/test/ares-fuzz.c
index bdaa4d6..66d6300 100644
--- a/test/ares-fuzz.c
+++ b/test/ares-fuzz.c
@@ -26,9 +26,9 @@
 #include <stdlib.h>
 #include <string.h>
 #ifdef WIN32
-#include <io.h>
+#  include <io.h>
 #else
-#include <unistd.h>
+#  include <unistd.h>
 #endif
 
 #include "ares.h"
@@ -38,16 +38,17 @@ static unsigned char afl_buffer[kMaxAflInputSize];
 
 #ifdef __AFL_LOOP
 /* If we are built with afl-clang-fast, use persistent mode */
-#define KEEP_FUZZING(count)  __AFL_LOOP(1000)
+#  define KEEP_FUZZING(count) __AFL_LOOP(1000)
 #else
 /* If we are built with afl-clang, execute each input once */
-#define KEEP_FUZZING(count) ((count) < 1)
+#  define KEEP_FUZZING(count) ((count) < 1)
 #endif
 
 /* In ares-test-fuzz.c and ares-test-fuzz-name.c: */
 int LLVMFuzzerTestOneInput(const unsigned char *data, unsigned long size);
 
-static void ProcessFile(int fd) {
+static void ProcessFile(int fd)
+{
   ares_ssize_t count = read(fd, afl_buffer, kMaxAflInputSize);
   /*
    * Make a copy of the data so that it's not part of a larger
@@ -61,7 +62,8 @@ static void ProcessFile(int fd) {
   }
 }
 
-int main(int argc, char *argv[]) {
+int main(int argc, char *argv[])
+{
   if (argc == 1) {
     int count = 0;
     while (KEEP_FUZZING(count)) {
diff --git a/test/ares-test-fuzz-name.c b/test/ares-test-fuzz-name.c
index 3c4f17b..cc7763f 100644
--- a/test/ares-test-fuzz-name.c
+++ b/test/ares-test-fuzz-name.c
@@ -26,13 +26,13 @@
 int LLVMFuzzerTestOneInput(const unsigned char *data, unsigned long size);
 
 // Entrypoint for Clang's libfuzzer, exercising query creation.
-int LLVMFuzzerTestOneInput(const unsigned char *data,
-                           unsigned long size) {
+int LLVMFuzzerTestOneInput(const unsigned char *data, unsigned long size)
+{
   // Null terminate the data.
-  char *name = malloc(size + 1);
-  unsigned char *buf = NULL;
-  int buflen = 0;
-  name[size] = '\0';
+  char          *name   = malloc(size + 1);
+  unsigned char *buf    = NULL;
+  int            buflen = 0;
+  name[size]            = '\0';
   memcpy(name, data, size);
 
   ares_create_query(name, C_IN, T_AAAA, 1234, 0, &buf, &buflen, 1024);
diff --git a/test/ares-test-fuzz.c b/test/ares-test-fuzz.c
index a57f28c..f104e34 100644
--- a/test/ares-test-fuzz.c
+++ b/test/ares-test-fuzz.c
@@ -21,59 +21,80 @@
 
 int LLVMFuzzerTestOneInput(const unsigned char *data, unsigned long size);
 
-
 // Entrypoint for Clang's libfuzzer
 int LLVMFuzzerTestOneInput(const unsigned char *data, unsigned long size)
 {
   // Feed the data into each of the ares_parse_*_reply functions.
-  struct hostent *host = NULL;
-  struct ares_addrttl info[5];
-  struct ares_addr6ttl info6[5];
-  unsigned char addrv4[4] = {0x10, 0x20, 0x30, 0x40};
-  struct ares_srv_reply* srv = NULL;
-  struct ares_mx_reply* mx = NULL;
-  struct ares_txt_reply* txt = NULL;
-  struct ares_soa_reply* soa = NULL;
-  struct ares_naptr_reply* naptr = NULL;
-  struct ares_caa_reply* caa = NULL;
-  struct ares_uri_reply* uri = NULL;
-  int count = 5;
+  struct hostent          *host = NULL;
+  struct ares_addrttl      info[5];
+  struct ares_addr6ttl     info6[5];
+  unsigned char            addrv4[4] = { 0x10, 0x20, 0x30, 0x40 };
+  struct ares_srv_reply   *srv       = NULL;
+  struct ares_mx_reply    *mx        = NULL;
+  struct ares_txt_reply   *txt       = NULL;
+  struct ares_soa_reply   *soa       = NULL;
+  struct ares_naptr_reply *naptr     = NULL;
+  struct ares_caa_reply   *caa       = NULL;
+  struct ares_uri_reply   *uri       = NULL;
+  int                      count     = 5;
   ares_parse_a_reply(data, (int)size, &host, info, &count);
-  if (host) ares_free_hostent(host);
+  if (host) {
+    ares_free_hostent(host);
+  }
 
-  host = NULL;
+  host  = NULL;
   count = 5;
   ares_parse_aaaa_reply(data, (int)size, &host, info6, &count);
-  if (host) ares_free_hostent(host);
+  if (host) {
+    ares_free_hostent(host);
+  }
 
   host = NULL;
   ares_parse_ptr_reply(data, (int)size, addrv4, sizeof(addrv4), AF_INET, &host);
-  if (host) ares_free_hostent(host);
+  if (host) {
+    ares_free_hostent(host);
+  }
 
   host = NULL;
   ares_parse_ns_reply(data, (int)size, &host);
-  if (host) ares_free_hostent(host);
+  if (host) {
+    ares_free_hostent(host);
+  }
 
   ares_parse_srv_reply(data, (int)size, &srv);
-  if (srv) ares_free_data(srv);
+  if (srv) {
+    ares_free_data(srv);
+  }
 
   ares_parse_mx_reply(data, (int)size, &mx);
-  if (mx) ares_free_data(mx);
+  if (mx) {
+    ares_free_data(mx);
+  }
 
   ares_parse_txt_reply(data, (int)size, &txt);
-  if (txt) ares_free_data(txt);
+  if (txt) {
+    ares_free_data(txt);
+  }
 
   ares_parse_soa_reply(data, (int)size, &soa);
-  if (soa) ares_free_data(soa);
+  if (soa) {
+    ares_free_data(soa);
+  }
 
   ares_parse_naptr_reply(data, (int)size, &naptr);
-  if (naptr) ares_free_data(naptr);
+  if (naptr) {
+    ares_free_data(naptr);
+  }
 
   ares_parse_caa_reply(data, (int)size, &caa);
-  if (caa) ares_free_data(caa);
+  if (caa) {
+    ares_free_data(caa);
+  }
 
   ares_parse_uri_reply(data, (int)size, &uri);
-  if (uri) ares_free_data(uri);
+  if (uri) {
+    ares_free_data(uri);
+  }
 
   return 0;
 }
