diff --git a/src/vertexcodec.cpp b/src/vertexcodec.cpp
--- a/src/vertexcodec.cpp
+++ b/src/vertexcodec.cpp
@@ -74,12 +74,14 @@
 #else
 #include <arm_neon.h>
 #endif
 #endif
 
 #ifdef SIMD_WASM
+#undef __DEPRECATED
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 #include <wasm_simd128.h>
 #endif
 
 #ifdef SIMD_WASM
 #define wasmx_splat_v32x4(v, i) wasm_v32x4_shuffle(v, v, i, i, i, i)
 #define wasmx_unpacklo_v8x16(a, b) wasm_v8x16_shuffle(a, b, 0, 16, 1, 17, 2, 18, 3, 19, 4, 20, 5, 21, 6, 22, 7, 23)
diff --git a/src/vertexfilter.cpp b/src/vertexfilter.cpp
--- a/src/vertexfilter.cpp
+++ b/src/vertexfilter.cpp
@@ -49,12 +49,13 @@
 #else
 #include <arm_neon.h>
 #endif
 #endif
 
 #ifdef SIMD_WASM
+#undef __DEPRECATED
 #include <wasm_simd128.h>
 #endif
 
 #ifdef SIMD_WASM
 #define wasmx_unpacklo_v16x8(a, b) wasm_v16x8_shuffle(a, b, 0, 8, 1, 9, 2, 10, 3, 11)
 #define wasmx_unpackhi_v16x8(a, b) wasm_v16x8_shuffle(a, b, 4, 12, 5, 13, 6, 14, 7, 15)
