diff --git a/prog/fuzzing/flipdetect_fuzzer.cc b/prog/fuzzing/flipdetect_fuzzer.cc
index bad7a6e..e1bafeb 100644
--- a/prog/fuzzing/flipdetect_fuzzer.cc
+++ b/prog/fuzzing/flipdetect_fuzzer.cc
@@ -9,7 +9,7 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	PIX *pixs_payload = pixReadMemSpix(data, size);
 	if(pixs_payload == NULL) return 0;
 	
-	l_float32 minupconf, minratio, conf1 upconf1, leftconf1;
+	l_float32 minupconf, minratio, conf1, upconf1, leftconf1;
 	PIX *pix_pointer_payload, *return_pix;
 	
 	pix_pointer_payload = pixCopy(NULL, pixs_payload);
@@ -17,7 +17,8 @@ LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 	pixDestroy(&pix_pointer_payload);
 
 	pix_pointer_payload = pixCopy(NULL, pixs_payload);
-	return_pix = pixOrientCorrect(pix_pointer_payload, minupconf, minratio, NULL, NULL, NULL, 1);
+	return_pix = pixOrientCorrect(pix_pointer_payload, minupconf,
+                                      minratio, NULL, NULL, NULL, 1);
 	pixDestroy(&pix_pointer_payload);	
 	pixDestroy(&return_pix);
