diff --git a/include/h2o.h b/include/h2o.h
--- a/include/h2o.h
+++ b/include/h2o.h
@@ -1136,12 +1136,16 @@ struct st_h2o_req_t {
      */
     unsigned char reprocess_if_too_early : 1;
     /**
      * set by the prxy handler if the http2 upstream refused the stream so the client can retry the request
      */
     unsigned char upstream_refused : 1;
+    /**
+     * if h2o_process_request has been called
+     */
+    unsigned char process_called : 1;
 
     /**
      * whether if the response should include server-timing header. Logical OR of H2O_SEND_SERVER_TIMING_*
      */
     unsigned send_server_timing;
 
diff --git a/lib/core/request.c b/lib/core/request.c
--- a/lib/core/request.c
+++ b/lib/core/request.c
@@ -331,12 +331,15 @@ h2o_handler_t *h2o_get_first_handler(h2o_req_t *req)
     setup_pathconf(req, hostconf);
     return req->pathconf->handlers.size != 0 ? req->pathconf->handlers.entries[0] : NULL;
 }
 
 void h2o_process_request(h2o_req_t *req)
 {
+    assert(!req->process_called);
+    req->process_called = 1;
+
     if (req->pathconf == NULL) {
         h2o_hostconf_t *hostconf = h2o_req_setup(req);
         setup_pathconf(req, hostconf);
     }
     call_handlers(req, req->pathconf->handlers.entries);
 }
