diff --git a/fuzz/c/std/gif_fuzzer.c b/fuzz/c/std/gif_fuzzer.c
index 4220344e..68ccf7ca 100644
--- a/fuzz/c/std/gif_fuzzer.c
+++ b/fuzz/c/std/gif_fuzzer.c
@@ -56,9 +56,11 @@ const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
   // Use a {} code block so that "goto exit" doesn't trigger "jump bypasses
   // variable initialization" warnings.
   {
-    wuffs_gif__decoder dec = ((wuffs_gif__decoder){});
-    const char* status = wuffs_gif__decoder__check_wuffs_version(
-        &dec, sizeof dec, WUFFS_VERSION);
+    wuffs_gif__decoder dec;
+    const char* status = wuffs_gif__decoder__initialize(
+        &dec, sizeof dec, WUFFS_VERSION,
+        (hash & 1) ? WUFFS_INITIALIZE__LEAVE_INTERNAL_BUFFERS_UNINITIALIZED
+                   : 0);
     if (status) {
       ret = status;
       goto exit;
diff --git a/fuzz/c/std/zlib_fuzzer.c b/fuzz/c/std/zlib_fuzzer.c
index 063e376d..ffded01f 100644
--- a/fuzz/c/std/zlib_fuzzer.c
+++ b/fuzz/c/std/zlib_fuzzer.c
@@ -61,9 +61,10 @@ uint8_t work_buffer[1];
 #endif
 
 const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
-  wuffs_zlib__decoder dec = ((wuffs_zlib__decoder){});
-  const char* status =
-      wuffs_zlib__decoder__check_wuffs_version(&dec, sizeof dec, WUFFS_VERSION);
+  wuffs_zlib__decoder dec;
+  const char* status = wuffs_zlib__decoder__initialize(
+      &dec, sizeof dec, WUFFS_VERSION,
+      (hash & 1) ? WUFFS_INITIALIZE__LEAVE_INTERNAL_BUFFERS_UNINITIALIZED : 0);
   if (status) {
     return status;
   }
@@ -71,7 +72,7 @@ const char* fuzz(wuffs_base__io_reader src_reader, uint32_t hash) {
   // Ignore the checksum for 99.99%-ish of all input. When fuzzers generate
   // random input, the checkum is very unlikely to match. Still, it's useful to
   // verify that checksumming does not lead to e.g. buffer overflows.
-  wuffs_zlib__decoder__set_ignore_checksum(&dec, hash & 0xFFFF);
+  wuffs_zlib__decoder__set_ignore_checksum(&dec, hash & 0xFFFE);
 
   uint8_t dst_buffer[DST_BUFFER_SIZE];
   wuffs_base__io_buffer dst = ((wuffs_base__io_buffer){
