diff --git a/src/xplist.c b/src/xplist.c
--- a/src/xplist.c
+++ b/src/xplist.c
@@ -110,13 +110,13 @@ static size_t dtostr(char *buf, size_t bufsize, double realval)
         len = snprintf(buf, bufsize, "%cinfinity", (realval > 0.0) ? '+' : '-');
     } else if (realval == 0.0f) {
         len = snprintf(buf, bufsize, "0.0");
     } else {
         size_t i = 0;
         len = snprintf(buf, bufsize, "%.*g", 17, realval);
-        for (i = 0; i < len; i++) {
+        for (i = 0; buf && i < len; i++) {
             if (buf[i] == ',') {
                 buf[i] = '.';
                 break;
             } else if (buf[i] == '.') {
                 break;
             }
@@ -476,13 +476,13 @@ static void node_estimate_size(node_t *node, uint64_t *size, uint32_t depth)
             } else {
                 *size += num_digits_i((int64_t)data->intval);
             }
             *size += (XPLIST_INT_LEN << 1) + 6;
             break;
         case PLIST_REAL:
-            *size += num_digits_i((int64_t)data->realval) + 7;
+            *size += dtostr(NULL, 0, data->realval);
             *size += (XPLIST_REAL_LEN << 1) + 6;
             break;
         case PLIST_DATE:
             *size += 20; /* YYYY-MM-DDThh:mm:ssZ */
             *size += (XPLIST_DATE_LEN << 1) + 6;
             break;
