diff --git a/test/ares-test-fuzz.c b/test/ares-test-fuzz.c
index d7f54fc..ddd6583 100644
--- a/test/ares-test-fuzz.c
+++ b/test/ares-test-fuzz.c
@@ -51,5 +51,9 @@ int LLVMFuzzerTestOneInput(const unsigned char *data,
   ares_parse_caa_reply(data, size, &caa);
   if (caa) ares_free_data(caa);
 
+  struct ares_uri_reply* uri = NULL;
+  ares_parse_uri_reply(data, size, &uri);
+  if (uri) ares_free_data(uri);
+
   return 0;
 }
