diff --git a/src/parser.c b/src/parser.c
--- a/src/parser.c
+++ b/src/parser.c
@@ -1112,12 +1112,17 @@ Node* TY_(ParseBlock)( TidyDocImpl* doc, Node *element, GetTokenMode mode )
         if ( element->tag->model & CM_EMPTY )
         {
             DEBUG_LOG(SPRTF("<<<Exit ParseBlock 1 %d...\n", --in_parse_block));
             return NULL;
         }
 
+        if ( nodeIsDIV(element) && nodeIsDL(element->parent) && TY_(IsHTML5Mode)(doc) )
+        {
+            return TY_(ParseDefList)(doc, element, mode);
+        }
+        
         if ( nodeIsFORM(element) &&
              DescendantOf(element, TidyTag_FORM) )
             TY_(Report)(doc, element, NULL, ILLEGAL_NESTING );
 
         /*
          InlineDup() asks the lexer to insert inline emphasis tags
@@ -1329,13 +1334,13 @@ Node* TY_(ParseBlock)( TidyDocImpl* doc, Node *element, GetTokenMode mode )
                     BadForm( doc );
 
                 TY_(Report)(doc, element, node, DISCARDING_UNEXPECTED );
                 TY_(FreeNode)( doc, node );
                 continue;
             }
-
+            
             /* #427671 - Fix by Randy Waki - 10 Aug 00 */
             /*
              If an LI contains an illegal FRAME, FRAMESET, OPTGROUP, or OPTION
              start tag, discard the start tag and let the subsequent content get
              parsed as content of the enclosing LI.  This seems to mimic IE and
              Netscape, and avoids an infinite loop: without this check,
@@ -2308,13 +2313,13 @@ Node* TY_(ParseDefList)( TidyDocImpl* doc, Node *list, GetTokenMode mode )
                         TY_(pushMemory)( doc, memory );
                         DEBUG_LOG(SPRTF("<<<Exiting ParseDefList 3 with a node to parse: %s\n", node->element));
                         return node;
                     }
                 }
 
-                if ( !(nodeIsDT(node) || nodeIsDD(node)) )
+                if ( !( nodeIsDT(node) || nodeIsDD(node) || ( nodeIsDIV(node) && TY_(IsHTML5Mode)(doc) ) ) )
                 {
                     TY_(UngetToken)( doc );
 
                     if (!(node->tag->model & (CM_BLOCK | CM_INLINE)))
                     {
                         TY_(Report)(doc, list, node, TAG_NOT_ALLOWED_IN);
@@ -2337,13 +2342,13 @@ Node* TY_(ParseDefList)( TidyDocImpl* doc, Node *list, GetTokenMode mode )
                 {
                     TY_(Report)(doc, list, node, DISCARDING_UNEXPECTED);
                     TY_(FreeNode)( doc, node);
                     continue;
                 }
 
-                /* node should be <DT> or <DD>*/
+                /* node should be <DT> or <DD> or <DIV>*/
                 TY_(InsertNodeAtEnd)(list, node);
                 {
                     TidyParserMemory memory = {0};
                     memory.identity = TY_(ParseDefList);
                     memory.original_node = list;
                     memory.reentry_node = node;
diff --git a/src/parser.h b/src/parser.h
--- a/src/parser.h
+++ b/src/parser.h
@@ -53,13 +53,13 @@ typedef struct _TidyParserMemory
     Node         *original_node; /**< Originally provided node at entry. */
     Node         *reentry_node;  /**< The node with which to re-enter. */
     GetTokenMode reentry_mode;   /**< The token mode to use when re-entering. */
     int          reentry_state;  /**< State to set during re-entry. Defined locally in each parser. */
     GetTokenMode mode;           /**< The caller will peek at this value to get the correct mode. */
     int          register_1;     /**< Local variable storage. */
-    int          register_2;   /**< Local variable storage. */
+    int          register_2;     /**< Local variable storage. */
 } TidyParserMemory;
 
 
 /**
  *  This typedef represents a stack of parserState. The Tidy document has its
  *  own instance of this.
